---
title: "MultiCompMicrobiome"
author: "GK"
date: "3.25.24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown



```{r load libraries}
# load important libraries
library(dplyr)
library(readxl)
library(ggplot2)
library(psych)
library(lubridate)
library(stargazer)
library(ggbeeswarm)
library(epitools)
library(vegan)
library(tidyr)
library(gdata)
library(stringr)
library(ggpubr)
library(mice)
library(networkD3)
library(data.table)
library("Partiallyoverlapping")
library("ggsignif")
library("permute")
library("haven")
library(ggcorrplot)
library(tableone)
library(nlme)
library(openxlsx)
library("survival")
library(ggsci)
library("survminer")
library(compositions)
library(joineR)
library(phyloseq)
library(mia)
library(miaViz)
library(scater)
library(patchwork)
library(tidyverse)
library(picante)
require(cowplot)
library(ecodist)
library(pheatmap)
library(dendextend)
library(tidySummarizedExperiment)
library(ANCOMBC)
library(ALDEx2)
library(knitr)
library(palmerpenguins)
library(ggforce)
library(DESeq2)
library(scran)
library(ggordiplots)
library(bluster)
library(DirichletMultinomial)
library(parallel)
library(GGally)
library(htmlwidgets)
library(jaccard)
library(rCausalMGM)
library(bnstruct)
library(rstatix)
library(metan)
library(ggsankey)
library(caret)
library(grid)
library(shadowtext)
library(limma)
library(gcookbook)
library(scales)
library(tibble)


```

```{r import datasets}
# Set working directory and dataset folder
setwd(getwd())
# import the taxonomic table for 16s-Seq from Illumina Miseq at genus level
all_counts <- read_csv("data/0102_ALIR_2023.taxa.genus.cmF.cln.SampleID.summary_table.csv") 
row.names(all_counts)<-all_counts$sample_id
### import de-ifentified metadata file for Pitt cohorts 
MetaData <- read_csv("data/MetaData_all.csv")
row.names(MetaData) <- MetaData$sample_id
identical(row.names(MetaData), MetaData$sample_id)


##### these files contain the main data required for the descriptive analyses of diversity/composition in clinical and experimental samples. 
#### From this point on will follow a series of R scripts to generate the main figures and tables data for the descriptive analyses.
### In later chunks of this Rmarkdown document, we will conduct separate analyses of clinical samples in the UPMC-ARF cohort and perform all quantitative analyses for Figures 2 onwards. 
#### In the last chunks of this document, we will import the taxonomic/metadata files for the MGH cohort to conduct the validation analyses of Figure 5. 

```



```{r Rarefaction curves - Shannon Index}
#### obtain information on soiling of rectal swabs
table(MetaData$rectalswab_soiled)
gut.indx1 = which(MetaData$SampleType =="stool")
gut.indx2 = which(MetaData$SampleType == "rectal" & MetaData$rectalswab_soiled == "yes")
gut.indx3 = which(MetaData$SampleType == "rectal" & is.na(MetaData$rectalswab_soiled))
gut.indx = unique(c(gut.indx1,gut.indx2,gut.indx3))
#### obtain information on ETA samples and whether depletion was applied in the DNA extraction process, and then select undepleted samples with higher sequencing yield
meta_ETA = MetaData[which(MetaData$SampleType=="ETA"),] #| MetaData$SampleType=="BAL"
table(meta_ETA$SampleType)
ETA_subjects = unique(meta_ETA$SubjectID)
indx = list()

for(sub in ETA_subjects){
  x = MetaData$RespSample_Depletion[which(MetaData$SubjectID==sub)]
  
  undep_indx = which(x=="Undep")
  dep_indx =  which(x=="Dep")
  if(length(undep_indx)>0){
    indx = c(indx,which(MetaData$SubjectID==sub)[undep_indx])
  }else{
    indx = c(indx,which(MetaData$SubjectID==sub )[dep_indx])
  }
  
}

ETA.indx = unlist(indx)
### obtain information on oral samples. 
oral.indx = which( MetaData$SampleType=="oral")
MetaData$compartment = MetaData$SampleType
MetaData$compartment[oral.indx] = "oral"
MetaData$compartment[ETA.indx] = "lung"
MetaData$compartment[gut.indx] = "gut"
MetaData = MetaData[which(MetaData$compartment %in% c("oral","lung","gut")),]
table(MetaData$studydayfollowup, MetaData$compartment)

# remove samples without clinical information
counts = all_counts[MetaData$sample_id,-c(1,2)]
identical(row.names(MetaData), row.names(counts))

# Suppl Figure 1a : Rarefaction Curves
metadata_rare = MetaData[which(MetaData$compartment%in%c("oral","lung","gut")),]
counts_rare = counts[metadata_rare$sample_id,]

metadata_rare$compartment =  as.factor(metadata_rare$compartment)
metadata_rare$compartment = ordered(metadata_rare$compartment,levels = c("oral","lung","gut"))


df <- data.frame(x = names(counts_rare))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order",
                                    "Family", "Genus"), sep = "\\.")

row.names(taxa) = names(counts_rare)

theme_set(theme_bw())

OTU = otu_table(t(counts_rare), taxa_are_rows = TRUE)
TAX = tax_table(as.matrix(taxa))
sampledata = sample_data(data.frame(
  SampleID = metadata_rare$sample_id,
  metadata_rare$sample_id,
  Compartment = metadata_rare$compartment,
  followup = metadata_rare$studydayfollowup,
  row.names=metadata_rare$sample_id,
  stringsAsFactors=FALSE
))

ps_OTU_table <- OTU
ps_sample_mapping <- sampledata
ps_exp <- merge_phyloseq(ps_OTU_table, ps_sample_mapping)
ps_exp_trim <- prune_taxa(taxa_sums(ps_exp) > 1, ps_exp)

# For Rarefaction
#combining sample info and OTU_table, this is creating the phyloseq data object
ps_exp <- merge_phyloseq(ps_OTU_table, ps_sample_mapping)
rare_curve_ps1 <- ps_exp
rare_curve_ps1
sample_sums(rare_curve_ps1)
set.seed(42)

calculate_rarefaction_curves <- function(rare_curve_ps1, measures, depths) {
  require('plyr') # ldply
  require('reshape2') # melt
  
  estimate_rarified_richness <- function(rare_curve_ps1, measures, depth) {
    if(max(sample_sums(rare_curve_ps1)) < depth) return()
    rare_curve_ps1 <- prune_samples(sample_sums(rare_curve_ps1) >= depth, rare_curve_ps1)
    
    rarified_rare_curve_ps1 <- rarefy_even_depth(rare_curve_ps1, depth, verbose = FALSE)
    
    alpha_diversity <- estimate_richness(rarified_rare_curve_ps1, measures = measures)
    
    # as.matrix forces the use of melt.array, which includes the Sample names (rownames)
    molten_alpha_diversity <- melt(as.matrix(alpha_diversity), varnames = c('Sample', 'Measure'),value.name =   = 'Alpha_diversity')
    
    molten_alpha_diversity$Compartment = rarified_rare_curve_ps1@sam_data$Compartment[molten_alpha_diversity$Sample] 
    molten_alpha_diversity$Followup = rarified_rare_curve_ps1@sam_data$followup[molten_alpha_diversity$Sample] 
    molten_alpha_diversity
  }
  
  names(depths) <- depths # this enables automatic addition of the Depth to the output by ldply
  rarefaction_curve_data <- ldply(depths, estimate_rarified_richness, rare_curve_ps1 = rare_curve_ps1, measures = measures, .id = 'Depth', .progress = ifelse(interactive(), 'text', 'none'))
  
  # convert Depth from factor to numeric
  
  rarefaction_curve_data$Depth <- as.numeric(levels(rarefaction_curve_data$Depth))[rarefaction_curve_data$Depth]
  
  rarefaction_curve_data
}

rarefaction_curve_data <- calculate_rarefaction_curves(rare_curve_ps1, c('Shannon'), seq(1, 20000, 200))
summary(rarefaction_curve_data)

rarefaction_curve_data_summary <- ddply(rarefaction_curve_data, c('Depth', 'Sample', 'Compartment'), summarise, Alpha_diversity = Alpha_diversity, Alpha_diversity_mean = mean(Alpha_diversity), Alpha_diversity_sd = sd(Alpha_diversity))
rarefaction_curve_data_summary_verbose <- merge(rarefaction_curve_data_summary, data.frame(sample_data(rare_curve_ps1)), by.x = 'Sample', by.y = 'row.names')


ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity,
    ymin = -0.1,
    ymax = 4,
    colour = Sample,
    group = Sample
  )
) + geom_point(size = 0.001) + theme_bw() + 
  theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(color = "black"),
        axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))+ 
  facet_wrap(
    facets = ~ Compartment.x,
    scales = 'free_y',
    ncol = 1
  )+ geom_line() + geom_vline(xintercept=1000, color= "black", linetype='dashed')


ggsave("Rarefaction_Compartment.jpeg", width = 10, height = 15)


```



```{r - Figure1}
# Intra-compartment comparisons of healthy and ICU patients. 

MetaData <- read_csv("data/MetaData_all.csv")

#### define the healthy control samples as corresponding compartments. 

lhmp_OW_indx = which(MetaData$StudyName=="lhmp" & MetaData$SampleType == "oral")
lhmp_BAL_indx = which((MetaData$StudyName=="lhmp" & MetaData$SampleType == "IS") | 
                        (MetaData$StudyName=="lhmp" & MetaData$SampleType == "BAL"))
fmt_indx = which(MetaData$StudyName=="fmt" & MetaData$SampleType == "stool")
MetaData$compartment = NA
MetaData$group[lhmp_OW_indx] = "healthyCtrl"
MetaData$group[lhmp_BAL_indx] = "healthyCtrl"
MetaData$group[fmt_indx] = "healthyCtrl"
MetaData$SampleType[lhmp_OW_indx] = "OW"
MetaData$SampleType[lhmp_BAL_indx] = "BAL/IS"
MetaData$SampleType[fmt_indx] = "FMT"
MetaData$compartment[lhmp_OW_indx] = "oral"
MetaData$compartment[lhmp_BAL_indx] = "lung"
MetaData$compartment[fmt_indx] = "gut"
MetaData$studydayfollowup[lhmp_OW_indx] = "healthyCtrl"
MetaData$studydayfollowup[lhmp_BAL_indx] = "healthyCtrl"
MetaData$studydayfollowup[fmt_indx] = "healthyCtrl"



############## rarefaction prior to diversity comparisons

counts_filtered = counts[-which(rowSums(counts)<=1000),]

df <- data.frame(x = names(counts_filtered))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                    "Family", "Genus"), sep = "\\.")

row.names(taxa) = names(counts_filtered)
MetaData = MetaData[row.names(counts_filtered),]

# creat a single cell experiment object
se_filtered <- SingleCellExperiment(assays = list(counts = t(counts_filtered)),
                                  colData = MetaData,
                                  rowData = taxa)
dim(se_filtered)

ShannonIndexMat = matrix(, nrow = 1640, ncol = 100)
MDS1_oral = matrix(, nrow = 585, ncol = 100)
MDS2_oral = matrix(, nrow = 585, ncol = 100)
MDS3_oral = matrix(, nrow = 585, ncol = 100)

MDS1_lung = matrix(, nrow = 513, ncol = 100)
MDS2_lung = matrix(, nrow = 513, ncol = 100)
MDS3_lung = matrix(, nrow = 513, ncol = 100)

MDS1_gut = matrix(, nrow = 342, ncol = 100)
MDS2_gut = matrix(, nrow = 342, ncol = 100)
MDS3_gut = matrix(, nrow = 342, ncol = 100)

MDS1_all = matrix(, nrow = 1470, ncol = 100)
MDS2_all = matrix(, nrow = 1470, ncol = 100)
MDS3_all = matrix(, nrow = 1470, ncol = 100)

for (i in 1:100){
  se_rarefy <- subsampleCounts(se_filtered, min_size = 1000, name = "subsampled_counts", 
                               seed = 123)
  
  dim(se_rarefy)
  
  counts.rare = as.data.frame(t(assays(se_rarefy)$subsampled_counts))
  MetaData.rare = as.data.frame(colData(se_rarefy))
  
  rowSums(counts.rare)
  
  table(MetaData$studydayfollowup,MetaData$compartment)
  
  

  ###### transformations
  
  df <- data.frame(x = names(counts.rare))
  taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                      "Family", "Genus"), sep = "\\.")
  
  # use Genus as row names. If Genus is repeated, use the full name
  taxa$Genus[which(duplicated(taxa$Genus))] = names(counts.rare)[which(duplicated(taxa$Genus))]
  row.names(taxa) <- taxa$Genus
  
  names(counts.rare) = row.names(taxa)
  # creat a single cell experiment object
  se <- SingleCellExperiment(assays = list(counts = t(counts.rare)),
                             colData = MetaData.rare,
                             rowData = taxa)
  
  
  ### Transformations
  se <- mia::transformSamples(se,  method = "relabundance")
  se <- mia::transformSamples(x = se, abund_values = "counts", method = "clr", 
                              pseudocount = 1, name = "clr")
  
  # keep taxa that exceed a specific prevalence and detection thresholds
  se <- mia::subsetByPrevalentTaxa(se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                   prevalence = 5/100)
  dim(se)
  # Calculate Alpha Diversity
  se <- mia::estimateDiversity(se, 
                               abund_values = "relabundance",
                               index = "shannon", 
                               name = "ShannonIndex")
  
  
  df =  as.data.frame(colData(se))
  ShannonIndexMat[,i] = df$ShannonIndex
  # This for loop takes a day to run
  for(comp in c("oral","lung","gut","all")){
    if(comp !="all"){
      df_compartment = df[which(df$compartment==comp),]
      df_compartment$ICU_healthy = "NA"
      df_compartment$ICU_healthy[which(df_compartment$group=="healthyCtrl")] = "healthyCtrl" 
      df_compartment$ICU_healthy[which(df_compartment$group!="healthyCtrl")] = "ICU" 
      table(df_compartment$ICU_healthy,df_compartment$studydayfollowup)
      if(length(which(df_compartment$ICU_healthy=="NA"))>0){
        df_compartment = df_compartment[-which(df_compartment$ICU_healthy=="NA"),]
      }
      
      
      clr_assay <- t(assays(se)$clr)
      clr_assay = clr_assay[df_compartment$sample_id,]
    }else{
      df$ICU_healthy = df$compartment
      df_compartment = df[which(df$compartment=="oral" | df$compartment=="lung"|df$compartment=="gut"),]
      
      
    
      clr_assay <- t(assays(se)$clr)
      clr_assay = clr_assay[df_compartment$sample_id,]
    }
    
    
    identical(row.names(clr_assay),row.names(df_compartment))
    
    set.seed(1000)
    betaLRT <- vegan::metaMDS(clr_assay, k=2, distance = "bray",autotransform = FALSE) # stress = 0.11
    NMDS = data.frame(MDS1=betaLRT$points[,1], MDS2=betaLRT$points[,2], cluster=df_compartment$ICU_healthy)
    
    if(comp == "oral"){
      MDS1_oral[,i] = NMDS$MDS1
      MDS2_oral[,i] = NMDS$MDS2
      MDS3_oral[,i] = NMDS$cluster
    }else if(comp == "lung"){
      MDS1_lung[,i] = NMDS$MDS1
      MDS2_lung[,i] = NMDS$MDS2
      MDS3_lung[,i] = NMDS$cluster
    }else if(comp == "gut"){
      MDS1_gut[,i] = NMDS$MDS1
      MDS2_gut[,i] = NMDS$MDS2
      MDS3_gut[,i] = NMDS$cluster
    }else if(comp == "all"){
      MDS1_all[,i] = NMDS$MDS1
      MDS2_all[,i] = NMDS$MDS2
      MDS3_all[,i] = NMDS$cluster
    }
    
    permanova_cluster <- vegan::adonis2(clr_assay ~ ICU_healthy,
                                        data = df_compartment,method = "euclidean",
                                        permutations = 999,na.action=na.exclude)
    print(paste0("p-value:",as.data.frame(permanova_cluster)["ICU_healthy", "Pr(>F)"]))
  }
}

colData(se)$ShannonIndex = rowMeans(ShannonIndexMat)
table(df$studydayfollowup, df$compartment)

n_distinct(df$SubjectID)


#### Figure 1
df =  as.data.frame(colData(se))
df = df[-which(is.na(df$compartment)),]
df$studydayfollowup[which(df$group=="healthyCtrl")] = "healthyCtrl"
df$studydayfollowup = as.factor(df$studydayfollowup)
df$studydayfollowup = ordered(df$studydayfollowup, levels = c("healthyCtrl","baseline","middle","late"))

df$compartment = as.factor(df$compartment)
df$compartment = ordered(df$compartment, levels = c("oral","lung","gut"))

stat.test <- df %>%
  group_by(compartment) %>%
  wilcox_test(ShannonIndex ~ studydayfollowup) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance("p.adj")

stat.test = stat.test[which((stat.test$group1=="baseline" & stat.test$group2=="late") | 
                              stat.test$group1=="healthyCtrl" & stat.test$group2=="baseline" ),]

p <- ggplot(df, aes(studydayfollowup, y=ShannonIndex, color = compartment, stat = "stat_count")) + 
  geom_boxplot() + scale_color_jama()+ geom_beeswarm(size = 0.5)+
  # geom_jitter(width = 0.1)+
  facet_wrap(~compartment) +theme_bw() + 
  theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(color = "black"),
        axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))

stat.test <- stat.test %>% add_xy_position(x = "studydayfollowup")
p + stat_pvalue_manual(stat.test,  label = "p", tip.length = 0.01,
                       y.position = c(3.5, 3.5))
ggsave("figure1a_alpha_compartment_followupInterval.pdf", width = 9, height = 5)


# Figure 1 b
for(comp in c("oral","lung","gut","all")){
  if(comp == "all"){
    df$ICU_healthy = df$compartment
    df_compartment = df[which(df$compartment=="oral" | df$compartment=="lung"|df$compartment=="gut"),]
  }else{
    df_compartment = df[which(df$compartment==comp),]
    df_compartment$ICU_healthy = "NA"
    df_compartment$ICU_healthy[which(df_compartment$group=="healthyCtrl")] = "healthyCtrl"
    df_compartment$ICU_healthy[which(df_compartment$group!="healthyCtrl")] = "ICU"
    table(df_compartment$ICU_healthy,df_compartment$studydayfollowup)
    if(length(which(df_compartment$ICU_healthy=="NA"))>0){
      df_compartment = df_compartment[-which(df_compartment$ICU_healthy=="NA"),]
    }
  }
  
 
  if(comp == "oral"){
    NMDS = matrix(, nrow = 585, ncol = 3)
    NMDS = data.frame(matrix = NMDS)
    NMDS[,1] = rowMeans(MDS1_oral)
    NMDS[,2]= rowMeans(MDS2_oral)
    NMDS[,3] = df_compartment$ICU_healthy
  }else if(comp == "lung"){
    NMDS = matrix(, nrow = 513, ncol = 3)
    NMDS = data.frame(matrix = NMDS)
    NMDS[,1] = rowMeans(MDS1_lung)
    NMDS[,2]= rowMeans(MDS2_lung)
    NMDS[,3] = df_compartment$ICU_healthy
  }else if(comp == "gut"){
    NMDS = matrix(, nrow = 342, ncol = 3)
    NMDS = data.frame(matrix = NMDS)
    NMDS[,1] = rowMeans(MDS1_gut)
    NMDS[,2]= rowMeans(MDS2_gut)
    NMDS[,3] = df_compartment$ICU_healthy
  }else if(comp == "all"){
    NMDS = matrix(, nrow = 1470, ncol = 3)
    NMDS = data.frame(matrix = NMDS)
    NMDS[,1] = rowMeans(MDS1_all)
    NMDS[,2] = rowMeans(MDS2_all)
    NMDS[,3] = df_compartment$ICU_healthy
  }
  
  names(NMDS) = c("MDS1","MDS2", "cluster")
  row.names(NMDS) = row.names(df_compartment)
  # NMDS = NMDS[-which(NMDS$MDS1< -1.5),]
  NMDS$cluster <- as.factor(NMDS$cluster)
  NMDS$MDS1 <- as.numeric(NMDS$MDS1)
  NMDS$MDS2 <- as.numeric(NMDS$MDS2)
  
  NMDS.mean = aggregate(NMDS[,1:2],list(cluster=NMDS$cluster),"mean")
  cent <- aggregate(cbind(MDS1, MDS2) ~ cluster, data = NMDS, FUN = "mean")
  segs <- merge(NMDS, setNames(cent, c('cluster','oNMDS1','oNMDS2')),
                by = 'cluster', sort = FALSE)
  veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
  {
    theta <- (0:npoints) * 2 * pi/npoints
    Circle <- cbind(cos(theta), sin(theta))
    t(center + scale * t(Circle %*% chol(cov)))
  }
  df_ell <- data.frame()
  for(g in levels(NMDS$cluster)){
    df_ell <- rbind(df_ell, cbind(as.data.frame(with(NMDS[NMDS$cluster==g,],
                                                     veganCovEllipse(cov.wt(cbind(MDS1,MDS2),wt=rep(1/length(MDS1),length(MDS1)))$cov,center=c(mean(MDS1),mean(MDS2)))))
                                  ,cluster=g))
  }
  
  if(comp == "oral"){
    cols = c("#79af97","#374e55")
  }else if(comp == "lung"){
    cols = c("#79af97","#df8f44")
  }else if(comp == "gut"){
    cols = c("#79af97","#00a1d5")
  }else{
    cols = c("#374e55","#df8f44","#00a1d5")
  }
  
  p <- ggplot(data = NMDS, aes(x=MDS1, y=MDS2, colour=cluster)) + 
    geom_point(aes(color = cluster), size = 0.5, alpha = 0.2) +
    geom_path(data=df_ell, aes(x=MDS1, y=MDS2,colour=cluster), size=1, linetype=1) +
    annotate("text",x=NMDS.mean$MDS1,y=NMDS.mean$MDS2,label="") +
    # scale_color_nejm() + 
    scale_colour_manual(values = cols)+
    geom_segment(data = segs,mapping = aes(xend = oNMDS1, yend = oNMDS2), alpha = 0.2) +
    theme_bw() + theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
                       panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
                       axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))
  
  if(comp == "oral"){
    p1 = p
  }else if(comp == "lung"){
    p2 = p
  }else if(comp == "gut"){
    p3 = p
  }else{
    p4 = p
  }
  
  
  
  
}
p1+p2+p3
ggsave(paste0("ICU_healthy_Beta.jpg"), width = 7, height = 5)

p4 # Figure 1 G
ggsave(paste0("Beta_compartments_alltime_R1C11.jpg"), width = 7, height = 7)


# Figure 1: C, D, E: Differential abundance - ICU vs healthy


counts_filtered = counts[-which(rowSums(counts)<=1000),]

df <- data.frame(x = names(counts_filtered))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                    "Family", "Genus"), sep = "\\.")

row.names(taxa) = names(counts_filtered)
MetaDataـfiltered = MetaData[row.names(counts_filtered),]

# creat a single cell experiment object
se <- SingleCellExperiment(assays = list(counts = t(counts_filtered)),
                                    colData = MetaDataـfiltered,
                                    rowData = taxa)

### Transformations
se <- mia::transformSamples(se,  method = "relabundance")
se <- mia::transformSamples(x = se, abund_values = "counts", method = "clr", 
                            pseudocount = 1, name = "clr")

# keep taxa that exceed a specific prevalence and detection thresholds
se <- mia::subsetByPrevalentTaxa(se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                 prevalence = 5/100)
dim(se)

singleton_percent = length(which(colSums(counts_filtered)==0))*100/ncol(counts_filtered)
low_abundance_percent = (ncol(counts_filtered)-dim(se)[1]-length(which(colSums(counts_filtered)==0)))*100/ncol(counts_filtered)
analyzed_taxa_percent = dim(se)[1]*100/1459
singleton_percent+low_abundance_percent+analyzed_taxa_percent

write.csv(row.names(se),"analyzed_taxa.csv")

# Calculate Alpha Diversity
se <- mia::estimateDiversity(se, 
                             abund_values = "relabundance",
                             index = "shannon", 
                             name = "ShannonIndex")

clr = as.data.frame(assays(se)$clr)

df = as.data.frame(colData(se))
df$ICU_healthy = "NA"
df$ICU_healthy[which(df$group=="healthyCtrl")] = "healthyCtrl" 
df$ICU_healthy[which(df$group!="healthyCtrl")] = "ICU" 
table(df$ICU_healthy,df$studydayfollowup)

if(length(which(df$ICU_healthy=="NA"))>0){
  df = df[-which(df$ICU_healthy=="NA"),]
}
df$ICU_healthy[df$ICU_healthy=="ICU"] = "ARF"

comp = "oral"
for(comp in c("oral","lung", "gut")){
  df_comp = df[which(df$compartment==comp),]
  clr_comp = clr[,df_comp$sample_id]
  
  table(df_comp$ICU_healthy)
  
  samples_ = df_comp$ICU_healthy
  mm <- model.matrix(~0 + samples_)
  fit <- lmFit(clr_comp, mm)
  
  head(fit)
  contr <- makeContrasts(samples_ARF - samples_healthyCtrl, levels = colnames(coef(fit)))
  file_name = paste0("diff_abundance_",comp)
  manual_colors = c("#D3D3D3","#F8766D")
  tmp <- contrasts.fit(fit, contr)
  # Empirical Bayes smoothing of standard errors
  tmp <- eBayes(tmp)
  top.table <- topTable(tmp, sort.by = "P", n = Inf)
  write.csv(top.table,file = paste0('MultiCompMbio/results/',file_name,'.csv'))
  
  diff_taxa = top.table
  # row.names(diff_genes) = diff_genes$X
  diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
  names(diff_df) <- c("logFC","adj.P.Val")
  
  diff_df["Genus"]<-row.names(diff_taxa)
  head(diff_df)
  diff_df["Significance"] <- "NotSig"
  diff_df[which(diff_df['adj.P.Val'] < 0.05 & abs(diff_df$logFC)>1.5),"Significance"] <- "Sig & abs(logFC)>2"
  
  top_peaks = diff_df[which(diff_df['adj.P.Val'] < 0.05 & abs(diff_df$logFC)>1.5),]
  
  ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
    geom_point(aes(color = Significance)) +
    scale_color_manual(values = manual_colors) +
    theme_bw(base_size = 16) +
    geom_text_repel(
      data = top_peaks,
      aes(label = Genus),
      size = 5,
      box.padding = unit(0.35, "lines"),
      point.padding = unit(0.3, "lines")
    )+ theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
             panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
             axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))+
    geom_vline(xintercept=0, col="grey", linetype = "dashed" ) +
    geom_hline(yintercept=-log10(0.05), col="grey", linetype = "dashed")
  
  ggsave( paste0('MultiCompMbio/results/',file_name,'.jpg'), width = 7, height = 5)
}


#### INTERCOMPARTMENT COMPARISONS AMONG ICU PATIENTS

# Figure 1 : bacterial load by sample type

df =  as.data.frame(colData(se))
df = df[-which(is.na(df$compartment)),]

df$compartment = as.factor(df$compartment)
df$compartment = ordered(df$compartment, levels = c("oral","lung","gut"))
df$Log10_Nof16scopies = log10(df$Nof16scopies)
stat.test <- df %>%
  wilcox_test(Log10_Nof16scopies ~ compartment) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance("p.adj")

p <- ggplot(df, aes(compartment, y=Log10_Nof16scopies, color = compartment, stat = "stat_count")) + 
  geom_boxplot() + scale_color_jama()+ geom_beeswarm(size = 1)+ theme_bw() + 
  theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(color = "black"),
        axis.text.x  = element_text(siz = 18, color = "black"), axis.text.y  = element_text(siz = 18,color = "black"))

stat.test <- stat.test %>% add_xy_position(x = "compartment")
p + stat_pvalue_manual(stat.test,  label = "p", tip.length = 0.01, 
                       y.position = c(9, 9.5,10) )

ggsave("MultiCompMbio/results/Log10_Nof16scopies_compartment.pdf", width = 5, height = 7)



# Figure 1 H: Differential abundance Oral-Gut & Lung-Gut

clr = as.data.frame(assays(se)$clr)

df = as.data.frame(colData(se))
df$ICU_healthy = "NA"
df$ICU_healthy[which(df$group=="healthyCtrl")] = "healthyCtrl" 
df$ICU_healthy[which(df$group!="healthyCtrl")] = "ICU" 
table(df$ICU_healthy,df$studydayfollowup)

if(length(which(df$ICU_healthy=="NA"))>0){
  df = df[-which(df$ICU_healthy=="NA"),]
}
df = df[which(df$studydayfollowup=="baseline" & df$ICU_healthy=="ICU"),]

df_comp = df[-which(is.na(df$compartment)),]
clr_comp = clr[,df_comp$sample_id]
dim(df_comp)

table(df_comp$compartment)


samples_ = df_comp$compartment
mm <- model.matrix(~0 + samples_)
fit <- lmFit(clr_comp, mm)

head(fit)

contr <- makeContrasts(samples_oral - samples_lung, levels = colnames(coef(fit)))
file_name = paste0("diff_abundance_oral_lung")

tmp <- contrasts.fit(fit, contr)
# Empirical Bayes smoothing of standard errors
tmp <- eBayes(tmp)
top.table <- topTable(tmp, sort.by  = "P", n = Inf)
write.csv(top.table,file = paste0('MultiCompMbio/results/',file_name,'.csv'))

diff_taxa = top.table
diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
names(diff_df) <- c("logFC","adj.P.Val")

diff_df["Genus"]<-row.names(diff_taxa)
head(diff_df)
diff_df["Significance"] <- "NotSig"
diff_df[which(diff_df['adj.P.Val'] < 0.05 & abs(diff_df$logFC)>1.5),"Significance"] <- "Sig & abs(logFC)>1.5"

top_peaks = diff_df[which(diff_df['adj.P.Val'] < 0.05 & abs(diff_df$logFC)>1.5),]

manual_colors = c("#D3D3D3","#F8766D")

ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
  geom_point(aes(color = Significance)) +
  scale_color_manual(values = manual_colors) +
  theme_bw(base_size = 16) +
  
  theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
        axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))+
  geom_vline(xintercept=0, col="grey", linetype = "dashed" ) +
  geom_hline(yintercept=-log10(0.05), col="grey", linetype = "dashed")

ggsave( paste0('MultiCompMbio/results/',file_name,'.jpg'), width = 7, height = 5)


contr <- makeContrasts(samples_lung - samples_gut, levels = colnames(coef(fit)))
file_name = paste0("diff_abundance_lung_gut")

tmp <- contrasts.fit(fit, contr)
# Empirical Bayes smoothing of standard errors
tmp <- eBayes(tmp)
top.table <- topTable(tmp, sort.by  = "P", n = Inf)
write.csv(top.table,file = paste0('MultiCompMbio/results/',file_name,'.csv'))

diff_taxa = top.table
# row.names(diff_genes) = diff_genes$X
diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
names(diff_df) <- c("logFC","adj.P.Val")

diff_df["Genus"]<-row.names(diff_taxa)
head(diff_df)
diff_df["Significance"] <- "NotSig"
diff_df[which(diff_df['adj.P.Val'] < 0.05 & abs(diff_df$logFC)>1.5),"Significance"] <- "Sig & abs(logFC)>1.5"

top_peaks = diff_df[which(diff_df['adj.P.Val'] < 0.05 & abs(diff_df$logFC)>1.5),]

manual_colors = c("#D3D3D3","#F8766D")

ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
  geom_point(aes(color = Significance)) +
  scale_color_manual(values = manual_colors) +
  theme_bw(base_size = 16) +
  geom_text_repel(
    data = top_peaks,
    aes(label = Genus),
    size = 5,
    box.padding = unit(0.35, "lines"),
    point.padding = unit(0.3, "lines")
  )+ theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
           panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
           axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))+
  geom_vline(xintercept=0, col="grey", linetype = "dashed" ) +
  geom_hline(yintercept=-log10(0.05), col="grey", linetype = "dashed")

ggsave( paste0('MultiCompMbio/results/',file_name,'.jpg'), width = 7, height = 5)



# bacterial load by sample type over time

df =  as.data.frame(colData(se))
df = df[-which(is.na(df$compartment)),]
df = df[-which(df$studydayfollowup=="healthyCtrl"),]

df$Log10_Nof16scopies = log10(df$Nof16scopies)

df$studydayfollowup = as.factor(df$studydayfollowup)
df$studydayfollowup = ordered(df$studydayfollowup, levels = c("baseline","middle","late"))

df$compartment = as.factor(df$compartment)
df$compartment = ordered(df$compartment, levels = c("oral","lung","gut"))

table(df$compartment,df$studydayfollowup)

stat.test <- df %>%
  group_by(compartment) %>%
  wilcox_test(Log10_Nof16scopies ~ studydayfollowup) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance("p.adj")

p <- ggplot(df, aes(studydayfollowup, y=Log10_Nof16scopies, color = compartment, stat = "stat_count")) +
  geom_boxplot() + scale_color_jama()+ geom_beeswarm(size = 0.5)+
  facet_wrap(~compartment) +theme_bw() +
  theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(color = "black"),
        axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))

stat.test <- stat.test %>% add_xy_position(x = "studydayfollowup")
p + stat_pvalue_manual(stat.test,  label = "p", tip.length = 0.01,
                       y.position = c(9, 9.5,10))
ggsave("MultiCompMbio/results/Log10_Nof16scopies_compartment_followupInterval.pdf", width = 9, height = 5)


```


```{r FigS1}

MetaData <- read_csv("data/MetaData_all.csv")

#### define the healthy control samples as corresponding compartments. 
lhmp_OW_indx = which(MetaData$StudyName=="lhmp" & MetaData$SampleType == "oral")
lhmp_BAL_indx = which((MetaData$StudyName=="lhmp" & MetaData$SampleType == "IS") | 
                        (MetaData$StudyName=="lhmp" & MetaData$SampleType == "BAL"))
fmt_indx = which(MetaData$StudyName=="fmt" & MetaData$SampleType == "stool")
MetaData$compartment = NA
MetaData$group[lhmp_OW_indx] = "healthyCtrl"
MetaData$group[lhmp_BAL_indx] = "healthyCtrl"
MetaData$group[fmt_indx] = "healthyCtrl"
MetaData$SampleType[lhmp_OW_indx] = "OW"
MetaData$SampleType[lhmp_BAL_indx] = "BAL/IS"
MetaData$SampleType[fmt_indx] = "FMT"
MetaData$compartment[lhmp_OW_indx] = "oral"
MetaData$compartment[lhmp_BAL_indx] = "lung"
MetaData$compartment[fmt_indx] = "gut"
MetaData$studydayfollowup[lhmp_OW_indx] = "healthyCtrl"
MetaData$studydayfollowup[lhmp_BAL_indx] = "healthyCtrl"
MetaData$studydayfollowup[fmt_indx] = "healthyCtrl"




###### transformations

counts_filtered = counts[-which(rowSums(counts)<=1000),]
MetaData_filtered = MetaData[row.names(counts_filtered),]

table(MetaData_filtered$studydayfollowup, MetaData_filtered$compartment)

df <- data.frame(x = names(counts_filtered))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                    "Family", "Genus"), sep = "\\.")

# use Genus as row names. If Genus is repeated, use the full name
taxa$Genus[which(duplicated(taxa$Genus))] = names(counts_filtered)[which(duplicated(taxa$Genus))]
row.names(taxa) <- taxa$Genus

names(counts_filtered) = row.names(taxa)
# creat a single cell experiment object
se <- SingleCellExperiment(assays = list(counts = t(counts_filtered)),
                           colData = MetaData_filtered,
                           rowData = taxa)



### Transformations
se <- mia::transformSamples(se,  method = "relabundance")
se <- mia::transformSamples(x = se, abund_values = "counts", method = "clr", 
                            pseudocount = 1, name = "clr")

# keep taxa that exceed a specific prevalence and detection thresholds
se <- mia::subsetByPrevalentTaxa(se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                 prevalence = 5/100)

# Calculate Alpha Diversity
se <- mia::estimateDiversity(se, 
                             abund_values = "relabundance",
                             index = "shannon", 
                             name = "ShannonIndex")


df =  as.data.frame(colData(se))
table(df$studydayfollowup, df$compartment)

n_distinct(df$SubjectID)


#  QC - Supplement Figure 

df <- as.data.frame(MetaData_filtered)

table(df$SampleType)

df$compartment[df$SampleType=="BAL/IS"] = "BAL/IS"
df$compartment[df$SampleType=="BALCon"] = "BALCon"
df$compartment[df$SampleType=="ExtrNeg"] = "ExtrNeg"
df$compartment[df$SampleType=="FMT"] = "FMT"
df$compartment[df$SampleType=="OW"] = "OW"
df$compartment[df$SampleType=="OWCon"] = "OWCon"
df$compartment[df$SampleType=="PCRNeg"] = "PCRNeg"
df$compartment[df$SampleType=="PCRPosZymo"] = "PCRPosZymo"
df$compartment[df$SampleType=="TCON"] = "TCON"
df = df[-which(is.na(df$compartment)),]
pcoa_meta.data  <- df

pcoa_meta.data$compartment[which(pcoa_meta.data$compartment=="oral" |
                                   pcoa_meta.data$compartment=="lung"|
                                   pcoa_meta.data$compartment=="gut"|
                                   pcoa_meta.data$compartment=="OW"|
                                   pcoa_meta.data$compartment=="BAL/IS"|
                                   pcoa_meta.data$compartment=="FMT")] = "ClinicalSample"

pcoa_meta.data$compartment[which(pcoa_meta.data$compartment=="TCON" |
                                   pcoa_meta.data$compartment=="OWCon"|
                                   pcoa_meta.data$compartment=="BALCon"|
                                   pcoa_meta.data$compartment=="ExtrNeg"|
                                   pcoa_meta.data$compartment=="PCRNeg")] = "NegativeControls"
table(pcoa_meta.data$compartment)

pcoa_meta.data$compartment =  as.factor(pcoa_meta.data$compartment)
pcoa_meta.data$compartment = ordered(pcoa_meta.data$compartment,levels = c("ClinicalSample",
                                                                           "NegativeControls","PCRPosZymo"))
# Beta Diversity
clr_assay <- as.data.frame(t(assays(se)$clr))
clr_assay = clr_assay[pcoa_meta.data$sample_id,]
identical(row.names(clr_assay),row.names(clr_assay ))


set.seed(1000)
betaLRT <- vegan::metaMDS(clr_assay, k=2, distance = "bray",autotransform = FALSE)

NMDS = data.frame(MDS1=betaLRT$points[,1], MDS2=betaLRT$points[,2], cluster=pcoa_meta.data$compartment)
NMDS = NMDS[-which(NMDS$MDS1==max(NMDS$MDS1)),]
NMDS = NMDS[-which(NMDS$MDS1==max(NMDS$MDS1)),]
NMDS$cluster <- as.factor(NMDS$cluster)
NMDS.mean=aggregate(NMDS[,1:2],list(cluster=NMDS$cluster),"mean")
cent <- aggregate(cbind(MDS1, MDS2) ~ cluster, data = NMDS, FUN = "mean")
segs <- merge(NMDS, setNames(cent, c('cluster','oNMDS1','oNMDS2')),
              by = 'cluster', sort = FALSE)
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}
df_ell <- data.frame()
for(g in levels(NMDS$cluster)){
  df_ell <- rbind(df_ell, cbind(as.data.frame(with(NMDS[NMDS$cluster==g,],
                                                   veganCovEllipse(cov.wt(cbind(MDS1,MDS2),wt=rep(1/length(MDS1),length(MDS1)))$cov,center=c(mean(MDS1),mean(MDS2)))))
                                ,cluster=g))
}



ggplot(data = NMDS, aes(x=MDS1, y=MDS2, colour=cluster)) + 
  geom_point(aes(color = cluster), size = 0.5, alpha = 0.1) +
  geom_path(data=df_ell, aes(x=MDS1, y=MDS2,colour=cluster), size=1.3, linetype=1) +
  annotate("text",x=NMDS.mean$MDS1,y=NMDS.mean$MDS2,label="") +
  scale_colour_manual(values = c("#BEAED4", "#BF5B17","#1B9E77"))+
  geom_segment(data = segs,mapping = aes(xend = oNMDS1, yend = oNMDS2), alpha = 0.1) +
  theme_bw() + theme(legend.position = "none",panel.border = element_blank(), panel.grid.major = element_blank(),
                     panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))

ggsave(paste0("MultiCompMbio/results/QC_pcoa_all_samples.jpg"), width = 5, height = 5)
permanova_cluster <- vegan::adonis2(clr_assay ~ compartment,
                                    data = pcoa_meta.data,method = "euclidean",
                                    permutations = 999,na.action=na.exclude)
print(paste0("p-value:",as.data.frame(permanova_cluster)["compartment", "Pr(>F)"]))

########### Gut QC

gut_MetaData = MetaData_healthy_COVID_merged[which(MetaData_healthy_COVID_merged$SampleType=="stool" | 
                                                     MetaData_healthy_COVID_merged$SampleType=="rectal"),]
row.names(gut_MetaData) = gut_MetaData$sample_id
gut_data = all_counts[gut_MetaData$sample_id,-c(1,2)]
row.names(gut_data)
gut_data = gut_data[-which(rowSums(gut_data)<=1000),]

gut_MetaData = gut_MetaData[row.names(gut_data),]

gut_MetaData$gut.sample.source = "Unknown"

stool.indx = which(gut_MetaData$SampleType=="stool")
gut_MetaData$gut.sample.source[stool.indx] = "Stool"

soiled.rs.indx = which(gut_MetaData$SampleType=="rectal" & gut_MetaData$rectalswab_soiled == "yes")
gut_MetaData$gut.sample.source[soiled.rs.indx] = "Soiled_Rectal"

non.soiled.rs.indx = which(gut_MetaData$SampleType=="rectal" & gut_MetaData$rectalswab_soiled == "no")
gut_MetaData$gut.sample.source[non.soiled.rs.indx] = "Unsoiled_Rectal"

table(gut_MetaData$gut.sample.source)


df <- as.data.frame(gut_MetaData)
length(which(is.na(df$total16sreads)))

df$Log10_total16sreads = log10(df$total16sreads)
df$Log10_Nof16scopies = log10(df$Nof16scopies)
df$gut.sample.source = as.factor(df$gut.sample.source)
df$gut.sample.source = ordered(df$gut.sample.source, levels = c("Unsoiled_Rectal","Soiled_Rectal", "Stool", "Unknown"))

comp <- split(t(combn(levels(df$gut.sample.source), 2)),
              seq(nrow(t(combn(levels(df$gut.sample.source), 2)))))
p1 = ggplot(df, aes(gut.sample.source, y=Log10_Nof16scopies, color = gut.sample.source, stat = "stat_count")) + 
  geom_boxplot() + scale_color_nejm()+ 
  geom_beeswarm(size = 0.5, cex = 0.8)+
  stat_compare_means(comparisons = comp)  +  
  theme_bw() + 
  theme(legend.position = "none", panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(color = "black"),
        axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))

p1
ggsave(paste0("MultiCompMbio/results/QC_gutSamples_Log10_Nof16scopies.pdf"), width = 5, height = 5)


df <- data.frame(x = names(gut_data))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                    "Family","Genus"), sep = "\\.")
names(gut_data) <- taxa$Genus

# row.names(gut_MetaData) = gut_MetaData$sample_id
identical(row.names(gut_MetaData), row.names(gut_data))

gut_se <- SingleCellExperiment(assays = list(counts = t(gut_data)),
                               colData = gut_MetaData,
                               rowData = taxa)

gut_se <- mia::transformSamples(gut_se, method = "clr",pseudocount = 1)


# Beta Diversity
clr_assay <- t(assays(gut_se)$clr)

df <- as.data.frame(colData(gut_se))
set.seed(1000)
which(is.na(clr_assay))
betaLRT <- vegan::metaMDS(clr_assay, k=2, distance = "euclidean",autotransform = FALSE) 

NMDS = data.frame(MDS1=betaLRT$points[,1], MDS2=betaLRT$points[,2], cluster=df$gut.sample.source)
NMDS = NMDS[-which(NMDS$MDS1< -1.5),]
NMDS$cluster <- as.factor(NMDS$cluster)
NMDS.mean=aggregate(NMDS[,1:2],list(cluster=NMDS$cluster),"mean")
cent <- aggregate(cbind(MDS1, MDS2) ~ cluster, data = NMDS, FUN = "mean")
segs <- merge(NMDS, setNames(cent, c('cluster','oNMDS1','oNMDS2')),
              by = 'cluster', sort = FALSE)
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}
df_ell <- data.frame()
for(g in levels(NMDS$cluster)){
  df_ell <- rbind(df_ell, cbind(as.data.frame(with(NMDS[NMDS$cluster==g,],
                                                   veganCovEllipse(cov.wt(cbind(MDS1,MDS2),wt=rep(1/length(MDS1),length(MDS1)))$cov,center=c(mean(MDS1),mean(MDS2)))))
                                ,cluster=g))
}


p2 = ggplot(data = NMDS, aes(x=MDS1, y=MDS2, colour=cluster)) + 
  geom_point(aes(color = cluster), size = 0.5, alpha = 0.2) +
  geom_path(data=df_ell, aes(x=MDS1, y=MDS2,colour=cluster), size=1, linetype=1) +
  annotate("text",x=NMDS.mean$MDS1,y=NMDS.mean$MDS2,label="") +
  scale_color_nejm() + 
  geom_segment(data = segs,mapping = aes(xend = oNMDS1, yend = oNMDS2), alpha = 0.2) +
  theme_bw() + theme(legend.position = "none",panel.border = element_blank(), panel.grid.major = element_blank(),
                     panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))

p1+p2
ggsave(paste0("MultiCompMbio/results/QC_gutSamples.pdf"), width = 10, height = 5)


permanova_cluster <- vegan::adonis2(clr_assay ~ gut.sample.source,
                                    data = df,method = "euclidean",
                                    permutations = 999,na.action=na.exclude)
print(paste0("p-value:",as.data.frame(permanova_cluster)["gut.sample.source", "Pr(>F)"]))


##### N of reads by sample type
MetaData <- MetaData %>% mutate(SampleClass = case_when(SampleType=="BAL" ~ "ClinicalSample", 
                                                                                  SampleType=="BALCon" ~ "ProcedureNegativeControl", 
                                                                                  SampleType=="ETA" ~ "ClinicalSample", 
                                                                                  SampleType=="ExtrNeg" ~ "ExperNegativeControl", 
                                                                                  SampleType=="IS" ~ "ClinicalSample", 
                                                                                  SampleType=="oral" ~ "ClinicalSample", 
                                                                                  SampleType=="OWCon" ~ "ProcedureNegativeControl", 
                                                                                  SampleType=="PCRNeg" ~ "ExperNegativeControl", 
                                                                                  SampleType=="PCRPosZymo" ~ "PCRPosZymo", 
                                                                                  SampleType=="rectal" ~ "ClinicalSample", 
                                                                                  SampleType=="stool" ~ "ClinicalSample", 
                                                                                  SampleType=="TCON" ~ "ProcedureNegativeControl"
                                                                                    ))

my_colors <- c("ClinicalSample" = "plum4", 
               "ExperNegativeControl" = "orange", 
               "PCRPosZymo" = "seagreen", 
               "ProcedureNegativeControl" = "brown")

MetaData$SampleClass <- ordered(MetaData$SampleClass, levels = c("ClinicalSample", "ProcedureNegativeControl", "ExperNegativeControl", "PCRPosZymo"))
my_comparisons <- list(c("ClinicalSample", "ProcedureNegativeControl"), c("ClinicalSample", "ExperNegativeControl"),  c("ClinicalSample", "PCRPosZymo"))
MetaData %>% filter(!is.na(SampleClass))  %>% ggplot(aes(x=SampleClass, y = total16sreads, color = SampleClass)) +  
  #geom_jitter(size=0.05, width = 0.1, alpha = 0.2) + 
  geom_boxplot(alpha=0.1) +
  geom_violin() +
  theme_pubr() + theme(legend.position = "none", axis.text.x = element_text(size = 8, angle = 90))  + ylab("N of 16S reads") + xlab("") + ggtitle("") + stat_compare_means(comparisons = my_comparisons) + scale_color_manual(values = my_colors) +  geom_hline(yintercept = 1000, linetype = "dashed", color = "red")
ggsave("TotalReadsbySampleClass.jpeg", width = 4, height = 6)


```


```{r - Figure 2 }

### barplots of abundances grouped by oxygen requirement and plausible pathogenicity

MetaData <- read_csv("data/MetaData_all.csv")
gut.indx1 = which(MetaData$SampleType =="stool")
gut.indx2 = which(MetaData$SampleType == "rectal" & MetaData$rectalswab_soiled == "yes")
gut.indx3 = which(MetaData$SampleType == "rectal" & is.na(MetaData$rectalswab_soiled))
gut.indx = unique(c(gut.indx1,gut.indx2,gut.indx3))
#### obtain information on ETA samples and whether depletion was applied in the DNA extraction process, and then select undepleted samples with higher sequencing yield
meta_ETA = MetaData[which(MetaData$SampleType=="ETA"),] #| MetaData$SampleType=="BAL"
table(meta_ETA$SampleType)
ETA_subjects = unique(meta_ETA$SubjectID)
indx = list()

for(sub in ETA_subjects){
  x = MetaData$RespSample_Depletion[which(MetaData$SubjectID==sub)]
  
  undep_indx = which(x=="Undep")
  dep_indx =  which(x=="Dep")
  if(length(undep_indx)>0){
    indx = c(indx,which(MetaData$SubjectID==sub)[undep_indx])
  }else{
    indx = c(indx,which(MetaData$SubjectID==sub )[dep_indx])
  }
  
}

ETA.indx = unlist(indx)
### obtain information on oral samples. 
oral.indx = which( MetaData$SampleType=="oral")
MetaData$compartment = MetaData$SampleType
MetaData$compartment[oral.indx] = "oral"
MetaData$compartment[ETA.indx] = "lung"
MetaData$compartment[gut.indx] = "gut"
MetaData = MetaData[which(MetaData$compartment %in% c("oral","lung","gut")),]
table(MetaData$studydayfollowup, MetaData$compartment)

# remove samples without clinical information
counts = all_counts[MetaData$sample_id,-c(1,2)]
identical(row.names(MetaData), row.names(counts))


###### transformations

counts_filtered = counts[-which(rowSums(counts)<=1000),]

df <- data.frame(x = names(counts_filtered))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                    "Family", "Genus"), sep = "\\.")

row.names(taxa) = names(counts_filtered)
MetaDataـfiltered = MetaData[row.names(counts_filtered),]
# create a single cell experiment object
se <- SingleCellExperiment(assays = list(counts = t(counts_filtered)),
                           colData = MetaDataـfiltered,
                           rowData = taxa)
### Transformations
se <- mia::transformSamples(se,  method = "relabundance")
se <- mia::transformSamples(x = se, abund_values = "counts", method = "clr", 
                            pseudocount = 1, name = "clr")
# keep taxa that exceed a specific prevalence and detection thresholds
se <- mia::subsetByPrevalentTaxa(se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                 prevalence = 5/100)
# Calculate Alpha Diversity
se <- mia::estimateDiversity(se, 
                             abund_values = "relabundance",
                             index = "shannon", 
                             name = "ShannonIndex")

df =  as.data.frame(colData(se))
table(df$studydayfollowup, df$compartment)
n_distinct(df$SubjectID)

### Figure 2: anaerobe analysis - Oral - lung - gut - COVID==No

for(compart in c("oral","lung","gut")){
  comp_df = as.data.frame(colData(se))
  comp_df = comp_df[which(comp_df$COVID=="No"),]
  comp_df = comp_df[which(comp_df$compartment==compart),]
  
  comp_rel_abund_assay = as.data.frame(assays(se)$relabundance)
  comp_rel_abund_assay_df <- comp_rel_abund_assay[,comp_df$sample_id]
  comp_rel_abund_assay_df$Genus <- rownames(comp_rel_abund_assay_df)
  
  #### import this to create a table of bugs classified
  comp_bugsClassified = read.csv(paste0("MultiCompMbio/data/",compart,"_bugsClassified.csv"))
  comp_bugsClassified = comp_bugsClassified[,-1]
  comp_rel_abund_assay_df <- merge(comp_rel_abund_assay_df, comp_bugsClassified, by = "Genus", all.x = TRUE)
  
  ###### take the df from oral_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
  comp_rel_abund_assay_df$O2Req[is.na(comp_rel_abund_assay_df$O2Req)] <- "Unclassifiable"
  
  anaerobeslist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$O2Req=="Anaerobe"]
  aerobeslist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$O2Req=="Aerobe"]
  facultanaerobeslist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$O2Req=="Facultative anaerobe"]
  Microaerophilelist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$O2Req=="Microaerophile"]
  Unclassifiableslist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$O2Req=="Unclassifiable"]
  Variablelist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$O2Req=="Variable"]
  
  compbugsbyOxygen <- as.data.frame(t(comp_rel_abund_assay_df))
  ### remove last row of pathogens
  compbugsbyOxygen <- compbugsbyOxygen[1:dim(compbugsbyOxygen)[1]-1, ]
  
  #### in the compbugsbyoxygen, make colnames by the genus variable
  colnames(compbugsbyOxygen) <- compbugsbyOxygen[1,]
  #### remove the anaerobe and 02 req rows
  compbugsbyOxygen <- compbugsbyOxygen[2:dim(compbugsbyOxygen)[1]-1, ]
  
  #### calculate rowsums by the list selection criteria. 
  # make all values numeric - include some code to maintain the rownames by sample_id with this conversion
  
  compbugsbyOxygenSample_id  <-as.data.frame(rownames(compbugsbyOxygen))
  compbugsbyOxygen <- as.data.frame(sapply(compbugsbyOxygen,as.numeric))
  
  row.names(compbugsbyOxygen) = compbugsbyOxygenSample_id$`rownames(compbugsbyOxygen)`
  
  # calculate row sums by the desired list
  # https://urldefense.com/v3/__https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums__;!!NHLzug!JjN2W8WciCi2ekGgS4mEpM4YSFxUKZ-wVeDddRb7rQYgxULnGvMrUbn9UcCd2ZAq9r0OBSMWdr914dxawiLm$ 
  compbugsbyOxygen <- compbugsbyOxygen %>% 
    mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
  compbugsbyOxygen$AnaerobeSum
  
  compbugsbyOxygen <- compbugsbyOxygen %>% 
    mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
  compbugsbyOxygen$AnaerobeSum
  
  compbugsbyOxygen <- compbugsbyOxygen %>% 
    mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
  compbugsbyOxygen$AnaerobeSum
  
  compbugsbyOxygen <- compbugsbyOxygen %>% 
    mutate(MicroaerophileSum = rowSums(select(., .dots = all_of(Microaerophilelist))))
  compbugsbyOxygen$AnaerobeSum
  
  compbugsbyOxygen <- compbugsbyOxygen %>% 
    mutate(VariableSum = rowSums(select(., .dots = all_of(Variablelist))))
  compbugsbyOxygen$AnaerobeSum
  
  compbugsbyOxygen <- compbugsbyOxygen %>% 
    mutate(UnclassifiableSum = rowSums(select(., .dots = all_of(Unclassifiableslist))))
  compbugsbyOxygen$AnaerobeSum
  
  compbugsbyOxygen$totalO2sum <- compbugsbyOxygen$AnaerobeSum + compbugsbyOxygen$AerobeSum + compbugsbyOxygen$FacultAnaerobeSum  + compbugsbyOxygen$MicroaerophileSum  + compbugsbyOxygen$VariableSum + compbugsbyOxygen$UnclassifiableSum 
  ###mutate total to 1 and then unclassifiable as difference from total
  compbugsbyOxygen$totalO2sum <- 1
  compbugsbyOxygen$UnclassifiableSum <- 1 - (compbugsbyOxygen$AnaerobeSum + compbugsbyOxygen$AerobeSum + compbugsbyOxygen$FacultAnaerobeSum  + compbugsbyOxygen$MicroaerophileSum  + compbugsbyOxygen$VariableSum )
  
  comp_bugsClassifiedSum <- subset(compbugsbyOxygen, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum,MicroaerophileSum, VariableSum , UnclassifiableSum))
  
  row.names(comp_bugsClassifiedSum)
  ### incorporate with the comp_df 
  comp_df_sum <- merge(comp_df,comp_bugsClassifiedSum, by = 'row.names', all = TRUE)
  comp_df_sum = comp_df_sum[-which(is.na(comp_df_sum$sample_id)),]
  x = cbind(comp_df_sum$Row.names,comp_df_sum$sample_id)
  
  #### plot of comparisons of abundance by clinical groups. 
  
  # comp_df_sum = comp_df_sum[-which(is.na(comp_df_sum$studydayfollowup)),]
  comp_df_sum$studydayfollowup = as.factor(comp_df_sum$studydayfollowup)
  comp_df_sum$studydayfollowup = ordered(comp_df_sum$studydayfollowup, levels = c("baseline", "middle","late"))
  
  comp_abundances = comp_df_sum[,which(names(comp_df_sum)%in%c("sample_id","ShannonIndex","AnaerobeSum","AerobeSum","FacultAnaerobeSum"))]
  
  # Pairwise comparisons.
  mycomparisons <- list(c("baseline", "middle"), c("baseline", "late"), c("middle", "late"))
  anaerobescomp <- ggboxplot(comp_df_sum, x = "studydayfollowup", y = "AnaerobeSum", 
                             color = "#3E0B51",add = "jitter" , palette = "cbbPalette", xlab = "", ylab = "Obligative Anaerobe Abundance", legend = "none" ) + 
    stat_compare_means(comparisons = mycomparisons) 
  
  aerobescomp <- ggboxplot(comp_df_sum, x = "studydayfollowup", y = "AerobeSum", 
                           color = "#404581",add = "jitter" , palette = "cbbPalette", xlab = "", ylab = "Aerobe Abundance", legend = "none" ) + 
    stat_compare_means(comparisons = mycomparisons) 
  
  facultativeanaerobescomp <- ggboxplot(comp_df_sum, x = "studydayfollowup", y = "FacultAnaerobeSum", 
                                        color = "#40768B",add = "jitter" , palette = "cbbPalette", xlab = "", ylab = "Facultative Anaerobe Abundance", legend = "none" ) + 
    stat_compare_means(comparisons = mycomparisons) 
  
  
  ##### save the 3 more common for comp which we will use in a 2x3 graph with OPS
  
  comparAbundO2reqcomp_3common <- ggarrange(anaerobescomp, aerobescomp, facultativeanaerobescomp, ncol = 3, nrow = 1)
  ggsave(paste0("MultiCompMbio/results/comparAbundO2req_",compart,"_3common.jpeg"), width = 17, height =5)
  
  ##### create revised taxa barplot for these categories. 
  
  # O2 req with bar per ID
  dfo <- subset(comp_df_sum, select = c(studydayfollowup, AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum))
  dfo <- dfo %>% arrange(AnaerobeSum) %>%
    arrange(factor(studydayfollowup, levels = c('baseline','middle','late'))) #to order aesthetic by increasing anaerobic presence 
  dfo$Order <- seq(1, length.out=nrow(dfo), by=1)
  o2long <- dfo %>% gather(O2_Requirement, Relative.Abundance, -c(studydayfollowup, Order))
  o2long <- data.frame(o2long)
  o2long$studydayfollowup <- factor(o2long$studydayfollowup, levels = c('baseline','middle','late')) 
  o2long$O2_Requirement <-  factor(o2long$O2_Requirement, levels = c("AnaerobeSum", "AerobeSum",
                                                                     "FacultAnaerobeSum", "MicroaerophileSum", "VariableSum", "UnclassifiableSum"))
  
  ######### Plot for anaerobes by clinical group
  o2plot <- ggplot(o2long, aes(x=Order, y=Relative.Abundance, fill=O2_Requirement)) +
    geom_bar(stat="identity") + 
    geom_col(width=1) +
    scale_fill_viridis(discrete=TRUE, direction=1, labels = c("Anaerobes", "Aerobes", "Facultative Anaerobes", 
                                                              "Microaerophiles", "Variable", "Unclassifiable")) +
    facet_grid(~studydayfollowup, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
    ggtitle("Oxygen requirements of component bacteria by study followup interval") +
    theme(axis.line=element_blank(), axis.text.x=element_blank(),
          axis.ticks=element_blank(),
          axis.title.x=element_blank(), axis.text=element_text(size=12),
          axis.title=element_text(size=18), plot.title=element_text(size=18),
          panel.background=element_blank(), panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(), plot.background=element_blank(),
          legend.title=element_blank(), legend.text=element_text(size=12),
          strip.text.x = element_text(size=12), legend.position = "bottom")
  ggsave(paste0("MultiCompMbio/results/o2plot-",compart,".jpeg"), width = 10, height = 4.5)
  
  
  ### Oral or Respiratory Pathogen analysis - Oral - lung - gut - COVID==No
  
  comp_df = as.data.frame(colData(se))
  comp_df = comp_df[which(comp_df$COVID=="No"),]
  comp_df = comp_df[which(comp_df$compartment==compart),]
  
  comp_rel_abund_assay = as.data.frame(assays(se)$relabundance)
  comp_rel_abund_assay_df <- comp_rel_abund_assay[,comp_df$sample_id]
  comp_rel_abund_assay_df$Genus <- rownames(comp_rel_abund_assay_df)
  
  #### import this to create a table of bugs classified
  comp_bugsClassified = read.csv(paste0("MultiCompMbio/data/",compart,"_bugsClassified.csv"))
  comp_bugsClassified = comp_bugsClassified[,-1]
  
  comp_rel_abund_assay_df <- merge(comp_rel_abund_assay_df, comp_bugsClassified, by = "Genus", all.x = TRUE)
  
  ###### take the df from comp_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
  if(length(is.na(comp_rel_abund_assay_df$OralorRespiratoryPath))>0){
    comp_rel_abund_assay_df$OralorRespiratoryPath[is.na(comp_rel_abund_assay_df$OralorRespiratoryPath)] <- "Other"
  }
  
  comp_rel_abund_assay_df$OralorRespiratoryPath[which(comp_rel_abund_assay_df$OralorRespiratoryPath=="")] <- "Other"
  table(comp_rel_abund_assay_df$OralorRespiratoryPath)
  
  orallist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$OralorRespiratoryPath=="Oral"]
  pathogenlist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$OralorRespiratoryPath=="Pathogenic"]
  otherlist <- comp_rel_abund_assay_df$Genus[comp_rel_abund_assay_df$OralorRespiratoryPath=="Other"]
  
  
  compbugsbyOralPath <- as.data.frame(t(comp_rel_abund_assay_df))
  ### remove O2Req row of
  compbugsbyOralPath <- compbugsbyOralPath[-which(row.names(compbugsbyOralPath)=="O2Req"), ]
  
  
  #### in the compbugsbyOralPath, make colnames by the genus variable
  colnames(compbugsbyOralPath) <- compbugsbyOralPath[1,]
  #### remove the anaerobe and OralPath rows
  compbugsbyOralPath <- compbugsbyOralPath[-c(1,dim(compbugsbyOralPath)[1]), ]
  
  #### calculate rowsums by the list selection criteria. 
  # make all values numeric - include some code to maintain the rownames by sample_id with this conversion
  
  compbugsbyOralPathSample_id  <-as.data.frame(rownames(compbugsbyOralPath))
  compbugsbyOralPath <- as.data.frame(sapply(compbugsbyOralPath,as.numeric))
  row.names(compbugsbyOralPath) = compbugsbyOralPathSample_id$`rownames(compbugsbyOralPath)`
  
  
  # calculate row sums by the desired list
  # https://urldefense.com/v3/__https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums__;!!NHLzug!JjN2W8WciCi2ekGgS4mEpM4YSFxUKZ-wVeDddRb7rQYgxULnGvMrUbn9UcCd2ZAq9r0OBSMWdr914dxawiLm$ 
  compbugsbyOralPath <- compbugsbyOralPath %>% 
    mutate(PathogenSum = rowSums(select(., .dots = all_of(pathogenlist))))
  compbugsbyOralPath$PathogenSum
  
  compbugsbyOralPath <- compbugsbyOralPath %>% 
    mutate(OralSum = rowSums(select(., .dots = all_of(orallist))))
  compbugsbyOralPath$OralSum
  
  compbugsbyOralPath <- compbugsbyOralPath %>% 
    mutate(OtherSum = rowSums(select(., .dots = all_of(otherlist))))
  compbugsbyOralPath$OtherSum
  
  
  compbugsbyOralPath$totalOralsum <- compbugsbyOralPath$OralSum + compbugsbyOralPath$PathogenSum  
  ###mutate total to 1 and then other as difference from total
  compbugsbyOralPath$totalOralsum <- 1
  compbugsbyOralPath$OtherSum <- 1 - (compbugsbyOralPath$OralSum + compbugsbyOralPath$PathogenSum)
  
  comp_bugsClassifiedSum <- subset(compbugsbyOralPath, select = c(OralSum, PathogenSum, OtherSum))
  
  x = cbind(row.names(comp_df),row.names(comp_bugsClassifiedSum))
  identical(row.names(comp_df),row.names(comp_bugsClassifiedSum))
  
  
  ### incorporate with the comp_df 
  comp_df_sum <- merge(comp_df,comp_bugsClassifiedSum, by = 'row.names', all = TRUE)
  
  comp_abundances1 = as.data.frame(comp_df_sum[,which(names(comp_df_sum)%in%c("sample_id","OralSum", "PathogenSum", "OtherSum" ))])
  
  comp_abundances = merge(comp_abundances,comp_abundances1, by = 'sample_id', all = TRUE)
  
  write.csv(comp_abundances, paste0("MultiCompMbio/results/",compart,"_abundances_alltimepoints.csv"))
  
  #### plot of comparisons of abundance by clinical groups. 
  
  comp_df_sum$studydayfollowup = as.factor(comp_df_sum$studydayfollowup)
  comp_df_sum$studydayfollowup = ordered(comp_df_sum$studydayfollowup, levels = c("baseline", "middle","late"))
  # Pairwise comparisons.
  mycomparisons <- list(c("baseline", "middle"), c("baseline", "late"), c("middle", "late"))
  
  oraltaxa <- ggboxplot(comp_df_sum, x = "studydayfollowup", y = "OralSum", 
                        color = "#2494CC",add = "jitter" , 
                        xlab = "", ylab = "Oral Commensal Abundance", legend = "none" ) + 
    stat_compare_means(comparisons = mycomparisons) 
  
  pathogenictaxa <- ggboxplot(comp_df_sum, x = "studydayfollowup", y = "PathogenSum", 
                              color ="#FF5C77",add = "jitter" , palette = "cbbPalette", xlab = "", 
                              ylab = "Respiratory Pathogens Abundance", legend = "none" ) + 
    stat_compare_means(comparisons = mycomparisons) 
  
  othertaxa <- ggboxplot(comp_df_sum, x = "studydayfollowup", y = "OtherSum", 
                         color = "#74737A",add = "jitter" , palette = "cbbPalette", xlab = "",
                         ylab = "Other Taxa Abundance", legend = "none" ) + 
    stat_compare_means(comparisons = mycomparisons) 
  
  comparAbundOralPathogen <- ggarrange(oraltaxa, pathogenictaxa, othertaxa, ncol = 3, nrow = 1)
  ggsave(paste0("MultiCompMbio/results/comparAbundOralPathogen_",compart,".jpeg"), width = 17, height = 5)
  
  
  
  ##### create revised taxa barplot for these categories. 
  
  # Pathogen with bar per ID
  dfo <- subset(comp_df_sum, select = c(studydayfollowup, OralSum, PathogenSum, OtherSum))
  dfo <- dfo %>% arrange(PathogenSum) %>%
    arrange(factor(studydayfollowup, levels = c('baseline','middle','late'))) #to order aesthetic by increasing anaerobic presence 
  dfo$Order <- seq(1, length.out=nrow(dfo), by=1)
  
  o2long <- dfo %>% gather(Oral_Pathogen, Relative.Abundance, -c(studydayfollowup, Order))
  o2long <- data.frame(o2long)
  # o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
  o2long$studydayfollowup <- factor(o2long$studydayfollowup, levels = c('baseline','middle','late')) 
  o2long$Oral_Pathogen <-  factor(o2long$Oral_Pathogen, levels = c("OralSum", "PathogenSum","OtherSum"))
  
  ######### Plot for anaerobes by clinical group
  o2plot <- ggplot(o2long, aes(x=Order, y=Relative.Abundance, fill=Oral_Pathogen)) +
    geom_bar(stat="identity") + 
    geom_col(width=1) +
    scale_fill_manual(values = c("#2494CC", "#FF5C77", "#74737A"),labels = c("Oral Commensal", "Respiratory Pathogen", "Other") )+
    facet_grid(~studydayfollowup, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
    ggtitle("Oral vs respiratory pathogen component bacteria by study followup interval - oral") +
    theme(axis.line=element_blank(), axis.text.x=element_blank(),
          axis.ticks=element_blank(),
          axis.title.x=element_blank(), axis.text=element_text(size=12),
          axis.title=element_text(size=18), plot.title=element_text(size=18),
          panel.background=element_blank(), panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(), plot.background=element_blank(),
          legend.title=element_blank(), legend.text=element_text(size=12),
          strip.text.x = element_text(size=12), legend.position = "bottom")
  o2plot
  ggsave(paste0("MultiCompMbio/results/OralorRespiratoryPath-",compart,".jpeg"), width = 10, height = 4.5)
  
}



```




```{r Linear Regression Covariates - Bubble plots}

##### the following code will generate the bubble plots shown in the supplement, regressing shannon, qpcr, anaerobe and pathogen abundance against different clinical covariates. 
#### obtain antibiotic exposure scores


# Bubble plots - Shannon


tblAntibioticHistory = read.csv("tblAntibioticHistory.csv") 

tblAntibioticHistory <- filter(tblAntibioticHistory, day1Date !="")


tblAntibioticHistory$ZosynStartDate

tblAntibioticHistory <- tblAntibioticHistory %>% rowwise() %>%
  mutate(onZosyn = ifelse(between(day1Date, ZosynStartDate, ZosynEndDate), 1, 0),
         onMetronidazole = ifelse(between(day1Date, MetronidazoleStartDate, MetronidazoleEndDate), 1, 0),
         onMeropenem = ifelse(between(day1Date, MeropenemStartDate, MeropenemEndDate), 1, 0),
         onSulbactam = ifelse(between(day1Date, SulbactamStartDate, SulbactamEndDate), 1, 0))

table(tblAntibioticHistory$onZosyn)
table(tblAntibioticHistory$onMetronidazole)


tblAntibioticHistory <- tblAntibioticHistory %>% rowwise() %>%
  mutate(day1AnaerobicAbx = ifelse(max(onZosyn, onMetronidazole, onMeropenem,onSulbactam) == 1, "yes", "no"))

tblAntibioticHistory$day1AnaerobicAbx
tblAntibioticHistory$SubjectID

AnaerobicAbx = tblAntibioticHistory[,which(names(tblAntibioticHistory)%in%c("SubjectID","day1AnaerobicAbx"))]
NAT.score = read.csv("MultiCompMbio/data/Abx_NAT_score_45days.csv")
NAT.score = NAT.score[,c(1,16)] # day1 = V15

names(NAT.score) = c("SubjectID","NAT.Score")

compartment = "oral"
followuptime = "baseline"
for(followuptime in c("baseline","all")){
  for(compartment in c("oral","lung","gut")){
    df <- as.data.frame(colData(se))
    if(followuptime=="baseline"){
      df = df[which(df$compartment==compartment & df$studydayfollowup=="baseline"),]
    }else{
      df = df[which(df$compartment==compartment),]
    }
    df = df[which(df$COVID=="No"),]
    
    table(df$group)
    df = df[,-which(names(df)%in%c("total_ABx_score_day1_Convex",  "total_ABx_score_day3_Convex",  "total_ABx_score_day7_Convex",
                                   "total_ABx_score_day11_Convex"))]
    
    
    ABx_score_Convex = read.csv("MultiCompMbio/data/total_ABx_score_Convex.csv")
    names(ABx_score_Convex)[1] = "SubjectID"
    ABx_score_Convex$SubjectID = as.character(ABx_score_Convex$SubjectID)
    df$SubjectID = as.character(df$SubjectID)
    
    df =left_join(df,ABx_score_Convex, by = "SubjectID")  
    
    if(compartment == "oral"){
      abundances = read.csv("MultiCompMbio/results/oral_abundances_alltimepoints.csv")
    }else if(compartment == "lung"){
      abundances = read.csv("MultiCompMbio/results/lung_abundances_alltimepoints.csv")
    }else if(compartment == "gut"){
      abundances = read.csv("MultiCompMbio/results/gut_abundances_alltimepoints.csv")
    }
    
    row.names(abundances) = abundances$sample_id
    
    
    comp_df_sum_all = left_join(df,abundances, by = 'sample_id')
    NAT.score$SubjectID = as.character(NAT.score$SubjectID)
    comp_df_sum_all_NAT = left_join(comp_df_sum_all,NAT.score, by = "SubjectID")
    # comp_df_sum_all_NAT = comp_df_sum_all_NAT[-which(is.na(comp_df_sum_all_NAT$sample_id)),]
    AnaerobicAbx$SubjectID = as.character(AnaerobicAbx$SubjectID )
    comp_df_sum_all_NAT_AnaAbx = left_join(comp_df_sum_all_NAT,AnaerobicAbx, by = "SubjectID")
    # comp_df_sum_all_NAT_AnaAbx = comp_df_sum_all_NAT_AnaAbx[-which(is.na(comp_df_sum_all_NAT_AnaAbx$sample_id)),]
    
    comp_df_sum_all_NAT_AnaAbx$PathogenSum
    comp_df_sum_all_NAT_AnaAbx$AnaerobeSum
    
    df1 = comp_df_sum_all_NAT_AnaAbx
    
    df1$Gender[df1$Gender=="Male"] = 0
    df1$Gender[df1$Gender=="Female"] = 1
    df1$Gender <- as.numeric(df1$Gender)
    df1$ShannonIndex.y
    
    names(df1)[which(names(df1)=="ShannonIndex.y")] = "ShannonIndex"
    
    model_vars = df1[,c("ShannonIndex","Age","Gender","BMI",
                        "HistCOPD","HistAlcohol","HistDiabetes","HistImmunosuppression",
                        "group","Pneumonia","Aspiration","PEEP","FiO2",
                        "total_ABx_score_day1_Convex","NAT.Score","day1AnaerobicAbx","studyday3_mgpred")]
    
    model_vars$group[which(model_vars$group=="AcuteOnChronic")] = 0
    model_vars$group[which(model_vars$group=="AEILD")] = 1
    model_vars$group[which(model_vars$group=="airwaycontrol")] = 2
    model_vars$group[which(model_vars$group=="ards")] = 3
    model_vars$group[which(model_vars$group=="atrisk")] = 4
    model_vars$group[which(model_vars$group=="CHF")] = 5
    model_vars$group[which(model_vars$group=="other")] = 6
    model_vars$group = as.numeric(model_vars$group)
    
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="no")]= 0 
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="yes")]= 1
    
    model_vars$day1AnaerobicAbx = as.numeric(model_vars$day1AnaerobicAbx)
    
    shannon.r2 = list()
    pvalue = list()
    coefficients = list()
    for(dependant_var in names(model_vars)[2:length(names(model_vars))]){
      
      # model_data = model_vars[-which(is.na(model_vars[,dependant_var])),]
      
      model <- glm(ShannonIndex ~ model_vars[,dependant_var], data = model_vars,
                   family = "gaussian")
      shannon.r2 = c(shannon.r2,with(summary(model), 1 - deviance/null.deviance))
      
      pvalue = c(pvalue,summary(model)$coefficients[8])
      coefficients = c(coefficients,summary(model)$coefficients[2])
      
      
    }
    
    pvalue = unlist(pvalue)
    shannon.r2 = unlist(shannon.r2)
    coefficients = unlist(coefficients)
    
    
    vars  = c("Age","Sex","BMI","COPD","Alcohol_Use",
              "Diabetes","Immunosuppression","Clinical_Group",
              "Pneumonia","Aspiration","PEEP","FiO2",
              "Antibiotic_Score","NAT_Score","Anaerobic_Abx","Steroids_Score")
    
    order = c("Steroids_Score","Anaerobic_Abx","NAT_Score","Antibiotic_Score","FiO2","PEEP","Aspiration","Pneumonia",
              "Clinical_Group","Immunosuppression","Diabetes","Alcohol_Use","COPD","BMI","Sex","Age")
    
    
    
    dataBubblePlot <- data.frame(
      Clinical.Covariates = vars,
      Shannon.R2 = shannon.r2, 
      pvalue = pvalue,
      coefficient = coefficients
      
    )
    dataBubblePlot$pvalueـlevel = dataBubblePlot$pvalue
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue<0.05] = "<0.05"
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue>=0.05] = ">=0.05"
    dataBubblePlot$pvalueـlevel = factor(dataBubblePlot$pvalueـlevel, levels = c(">=0.05","<0.05"))
    ggplot(dataBubblePlot, aes(x = Shannon.R2, y = Clinical.Covariates,size = pvalueـlevel, color=coefficient)) + 
      geom_point(alpha = 0.7) + scale_colour_gradient2(
        limits=c(-0.25, 0.25),
        low = muted("red"),
        mid = "grey",
        high = muted("blue"),
        midpoint = 0
      )+ scale_y_discrete(limits = order)+theme_bw() + 
      theme(legend.position = "none", panel.border = element_blank(), 
            panel.grid.major = element_line(colour = "grey",linewidth = 0.075, linetype = 2),
            panel.grid.minor = element_line(colour = "grey",linewidth = 0.075, linetype = 2), axis.line = element_line(color = "black"),
            axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))
    
    ggsave(paste0('MultiCompMbio/results/bubble_Shannon_',compartment,'_',followuptime,'.jpg'), width = 4, height = 5)
    
    #### Nof16scopies
    
    model_vars = df1[,c("Nof16scopies","Age","Gender","BMI",
                        "HistCOPD","HistAlcohol","HistDiabetes","HistImmunosuppression",
                        "group","Pneumonia","Aspiration","PEEP","FiO2",
                        "total_ABx_score_day1_Convex","NAT.Score","day1AnaerobicAbx","studyday3_mgpred")]
    
    model_vars$group[which(model_vars$group=="AcuteOnChronic")] = 0
    model_vars$group[which(model_vars$group=="AEILD")] = 1
    model_vars$group[which(model_vars$group=="airwaycontrol")] = 2
    model_vars$group[which(model_vars$group=="ards")] = 3
    model_vars$group[which(model_vars$group=="atrisk")] = 4
    model_vars$group[which(model_vars$group=="CHF")] = 5
    model_vars$group[which(model_vars$group=="other")] = 6
    model_vars$group = as.numeric(model_vars$group)
    
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="no")]= 0 
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="yes")]= 1
    
    model_vars$day1AnaerobicAbx = as.numeric(model_vars$day1AnaerobicAbx)
    
    Nof16scopies.r2 = list()
    pvalue = list()
    coefficients = list()
    for(dependant_var in names(model_vars)[2:length(names(model_vars))]){
      
      # model_data = model_vars[-which(is.na(model_vars[,dependant_var])),]
      
      model <- glm(Nof16scopies ~ model_vars[,dependant_var], data = model_vars,
                   family = "gaussian")
      Nof16scopies.r2 = c(Nof16scopies.r2,with(summary(model), 1 - deviance/null.deviance))
      
      pvalue = c(pvalue,summary(model)$coefficients[8])
      coefficients = c(coefficients,summary(model)$coefficients[2])
      
      
    }
    
    pvalue = unlist(pvalue)
    Nof16scopies.r2 = unlist(Nof16scopies.r2)
    coefficients = unlist(coefficients)
    
    
    vars  = c("Age","Sex","BMI","COPD","Alcohol_Use",
              "Diabetes","Immunosuppression","Clinical_Group",
              "Pneumonia","Aspiration","PEEP","FiO2",
              "Antibiotic_Score","NAT_Score","Anaerobic_Abx","Steroids_Score")
    
    order = c("Steroids_Score","Anaerobic_Abx","NAT_Score","Antibiotic_Score","FiO2","PEEP","Aspiration","Pneumonia",
              "Clinical_Group","Immunosuppression","Diabetes","Alcohol_Use","COPD","BMI","Sex","Age")
    
    
    
    dataBubblePlot <- data.frame(
      Clinical.Covariates = vars,
      Nof16scopies.R2 = Nof16scopies.r2, 
      pvalue = pvalue,
      coefficient = coefficients
      
    )
    dataBubblePlot$pvalueـlevel = dataBubblePlot$pvalue
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue<0.05] = "<0.05"
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue>=0.05] = ">=0.05"
    dataBubblePlot$pvalueـlevel = factor(dataBubblePlot$pvalueـlevel, levels = c(">=0.05","<0.05"))
    ggplot(dataBubblePlot, aes(x = Nof16scopies.R2, y = Clinical.Covariates,size = pvalueـlevel, color=coefficient)) + 
      geom_point(alpha = 0.7) + scale_colour_gradient2(
        limits=c(-0.25, 0.25),
        low = muted("red"),
        mid = "grey",
        high = muted("blue"),
        midpoint = 0
      )+ scale_y_discrete(limits = order)+theme_bw() + 
      theme(legend.position = "none", panel.border = element_blank(), 
            panel.grid.major = element_line(colour = "grey",linewidth = 0.075, linetype = 2),
            panel.grid.minor = element_line(colour = "grey",linewidth = 0.075, linetype = 2), axis.line = element_line(color = "black"),
            axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))
    
    ggsave(paste0('MultiCompMbio/results/bubble_Nof16scopies.R2_',compartment,'_',followuptime,'.jpg'), width = 4, height = 5)
    
  }
}



# bubble plots - clinical covariates impact the abundance of pathogens and obligate anaerobes 



tblAntibioticHistory = read.csv("data/tblAntibioticHistory.csv") 

tblAntibioticHistory <- filter(tblAntibioticHistory, day1Date !="")


tblAntibioticHistory$ZosynStartDate

tblAntibioticHistory <- tblAntibioticHistory %>% rowwise() %>%
  mutate(onZosyn = ifelse(between(day1Date, ZosynStartDate, ZosynEndDate), 1, 0),
         onMetronidazole = ifelse(between(day1Date, MetronidazoleStartDate, MetronidazoleEndDate), 1, 0),
         onMeropenem = ifelse(between(day1Date, MeropenemStartDate, MeropenemEndDate), 1, 0),
         onSulbactam = ifelse(between(day1Date, SulbactamStartDate, SulbactamEndDate), 1, 0))

table(tblAntibioticHistory$onZosyn)
table(tblAntibioticHistory$onMetronidazole)


tblAntibioticHistory <- tblAntibioticHistory %>% rowwise() %>%
  mutate(day1AnaerobicAbx = ifelse(max(onZosyn, onMetronidazole, onMeropenem,onSulbactam) == 1, "yes", "no"))

tblAntibioticHistory$day1AnaerobicAbx
tblAntibioticHistory$SubjectID

AnaerobicAbx = tblAntibioticHistory[,which(names(tblAntibioticHistory)%in%c("SubjectID","day1AnaerobicAbx"))]
NAT.score = read.csv("data/Abx_NAT_score_45days.csv")
NAT.score = NAT.score[,c(1,16)] # day1 = V15

names(NAT.score) = c("SubjectID","NAT.Score")

compartment = "oral"
followup = "all"
for(followuptime in c("baseline","all")){
  for(compartment in c("oral","lung","gut")){
    
    df <- as.data.frame(colData(se))
    if(followuptime=="baseline"){
      df = df[which(df$compartment==compartment & df$studydayfollowup=="baseline"),]
    }else{
      df = df[which(df$compartment==compartment),]
    }
    
    df = df[which(df$COVID=="No"),]
    
    table(df$group)
    df = df[,-which(names(df)%in%c("total_ABx_score_day1_Convex",  "total_ABx_score_day3_Convex",  "total_ABx_score_day7_Convex",
                                   "total_ABx_score_day11_Convex"))]
    
    
    ABx_score_Convex = read.csv("data/total_ABx_score_Convex.csv")
    names(ABx_score_Convex)[1] = "SubjectID"
    ABx_score_Convex$SubjectID = as.character(ABx_score_Convex$SubjectID)
    df$SubjectID = as.character(df$SubjectID)
    
    df =left_join(df,ABx_score_Convex, by = "SubjectID")  
    
    if(compartment == "oral"){
      abundances = read.csv("MultiCompMbio/results/oral_abundances_alltimepoints.csv")
    }else if(compartment == "lung"){
      abundances = read.csv("MultiCompMbio/results/lung_abundances_alltimepoints.csv")
    }else if(compartment == "gut"){
      abundances = read.csv("MultiCompMbio/results/gut_abundances_alltimepoints.csv")
    }
    
    row.names(abundances) = abundances$sample_id
    
    
    comp_df_sum_all = left_join(df,abundances, by = 'sample_id')
    NAT.score$SubjectID = as.character(NAT.score$SubjectID)
    comp_df_sum_all_NAT = left_join(comp_df_sum_all,NAT.score, by = "SubjectID")
    # comp_df_sum_all_NAT = comp_df_sum_all_NAT[-which(is.na(comp_df_sum_all_NAT$sample_id)),]
    AnaerobicAbx$SubjectID = as.character(AnaerobicAbx$SubjectID )
    comp_df_sum_all_NAT_AnaAbx = left_join(comp_df_sum_all_NAT,AnaerobicAbx, by = "SubjectID")
    # comp_df_sum_all_NAT_AnaAbx = comp_df_sum_all_NAT_AnaAbx[-which(is.na(comp_df_sum_all_NAT_AnaAbx$sample_id)),]
    
    comp_df_sum_all_NAT_AnaAbx$PathogenSum
    comp_df_sum_all_NAT_AnaAbx$AnaerobeSum
    
    df1 = comp_df_sum_all_NAT_AnaAbx
    
    df1$Gender[df1$Gender=="Male"] = 0
    df1$Gender[df1$Gender=="Female"] = 1
    df1$Gender <- as.numeric(df1$Gender)
    
    
    model_vars = df1[,c("AnaerobeSum","Age","Gender","BMI",
                        "HistCOPD","HistAlcohol","HistDiabetes","HistImmunosuppression",
                        "group","Pneumonia","Aspiration","PEEP","FiO2",
                        "total_ABx_score_day1_Convex","NAT.Score","day1AnaerobicAbx","studyday3_mgpred")]
    
    
    model_vars$group[which(model_vars$group=="AcuteOnChronic")] = 0
    model_vars$group[which(model_vars$group=="AEILD")] = 1
    model_vars$group[which(model_vars$group=="airwaycontrol")] = 2
    model_vars$group[which(model_vars$group=="ards")] = 3
    model_vars$group[which(model_vars$group=="atrisk")] = 4
    model_vars$group[which(model_vars$group=="CHF")] = 5
    model_vars$group[which(model_vars$group=="other")] = 6
    model_vars$group = as.numeric(model_vars$group)
    
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="no")]= 0 
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="yes")]= 1
    model_vars$day1AnaerobicAbx = as.numeric(model_vars$day1AnaerobicAbx)
    
    AnaerobeSum.r2 = list()
    pvalue = list()
    coefficients = list()
    for(dependant_var in names(model_vars)[2:length(names(model_vars))]){
      # model_data = model_vars[-which(is.na(model_vars[,dependant_var])),]
      
      model <- glm(AnaerobeSum ~ model_vars[,dependant_var], data = model_vars,
                   family = "gaussian")
      AnaerobeSum.r2 = c(AnaerobeSum.r2,with(summary(model), 1 - deviance/null.deviance))
      
      pvalue = c(pvalue,summary(model)$coefficients[8])
      coefficients = c(coefficients,summary(model)$coefficients[2])
      
      
    }
    
    pvalue = unlist(pvalue)
    AnaerobeSum.r2 = unlist(AnaerobeSum.r2)
    coefficients = unlist(coefficients)
    
    
    vars  = c("Age","Sex","BMI","COPD","Alcohol_Use",
              "Diabetes","Immunosuppression","Clinical_Group",
              "Pneumonia","Aspiration","PEEP","FiO2",
              "Antibiotic_Score","NAT_Score","Anaerobic_Abx","Steroids_Score")
    
    order = c("Steroids_Score","Anaerobic_Abx", "NAT_Score","Antibiotic_Score","FiO2","PEEP","Aspiration","Pneumonia",
              "Clinical_Group","Immunosuppression","Diabetes","Alcohol_Use","COPD","BMI","Sex","Age")
    
    dataBubblePlot <- data.frame(
      Clinical.Covariates = vars,
      AnaerobeSum.R2 = AnaerobeSum.r2, 
      pvalue = pvalue,
      coefficient = coefficients
      
    )
    dataBubblePlot$pvalueـlevel = dataBubblePlot$pvalue
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue<0.05] = "<0.05"
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue>=0.05] = ">=0.05"
    dataBubblePlot$pvalueـlevel = factor(dataBubblePlot$pvalueـlevel, levels = c(">=0.05","<0.05"))
    ggplot(dataBubblePlot, aes(x = AnaerobeSum.R2, y = Clinical.Covariates,size = pvalueـlevel, color=coefficient)) + 
      geom_point(alpha = 0.7) + scale_colour_gradient2(
        limits=c(-0.25, 0.25),
        low = muted("red"),
        mid = "grey",
        high = muted("blue"),
        midpoint = 0
      ) + scale_y_discrete(limits = order)+theme_bw() + 
      theme(legend.position = "none", panel.border = element_blank(), 
            panel.grid.major = element_line(colour = "grey",linewidth = 0.075, linetype = 2),
            panel.grid.minor = element_line(colour = "grey",linewidth = 0.075, linetype = 2), axis.line = element_line(color = "black"),
            axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))
    
    ggsave(paste0('MultiCompMbio/results/bubble_AnaerobeSum_',compartment,'_',followuptime,'.jpg'), width = 4, height = 5)
    
    
    df1 = comp_df_sum_all_NAT_AnaAbx
    
    df1$Gender[df1$Gender=="Male"] = 0
    df1$Gender[df1$Gender=="Female"] = 1
    df1$Gender <- as.numeric(df1$Gender)
    
    model_vars = df1[,c("PathogenSum","Age","Gender","BMI",
                        "HistCOPD","HistAlcohol","HistDiabetes","HistImmunosuppression",
                        "group","Pneumonia","Aspiration","PEEP","FiO2",
                        "total_ABx_score_day1_Convex","NAT.Score","day1AnaerobicAbx","studyday3_mgpred")]
    
    model_vars$group[which(model_vars$group=="AcuteOnChronic")] = 0
    model_vars$group[which(model_vars$group=="AEILD")] = 1
    model_vars$group[which(model_vars$group=="airwaycontrol")] = 2
    model_vars$group[which(model_vars$group=="ards")] = 3
    model_vars$group[which(model_vars$group=="atrisk")] = 4
    model_vars$group[which(model_vars$group=="CHF")] = 5
    model_vars$group[which(model_vars$group=="other")] = 6
    model_vars$group = as.numeric(model_vars$group)
    
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="no")]= 0 
    model_vars$day1AnaerobicAbx[which(model_vars$day1AnaerobicAbx=="yes")]= 1
    model_vars$day1AnaerobicAbx = as.numeric(model_vars$day1AnaerobicAbx)
    
    PathogenSum.r2 = list()
    pvalue = list()
    coefficients = list()
    for(dependant_var in names(model_vars)[2:length(names(model_vars))]){
      # model_data = model_vars[-which(is.na(model_vars[,dependant_var])),]
      
      model <- glm(PathogenSum ~ model_vars[,dependant_var], data = model_vars,
                   family = "gaussian")
      PathogenSum.r2 = c(PathogenSum.r2,with(summary(model), 1 - deviance/null.deviance))
      
      pvalue = c(pvalue,summary(model)$coefficients[8])
      coefficients = c(coefficients,summary(model)$coefficients[2])
      
      
    }
    
    pvalue = unlist(pvalue)
    PathogenSum.r2 = unlist(PathogenSum.r2)
    coefficients = unlist(coefficients)
    
    
    dataBubblePlot <- data.frame(
      Clinical.Covariates = vars,
      PathogenSum.R2 = PathogenSum.r2, 
      pvalue = pvalue,
      coefficient = coefficients
      
    )
    dataBubblePlot$pvalueـlevel = dataBubblePlot$pvalue
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue<0.05] = "<0.05"
    dataBubblePlot$pvalueـlevel[dataBubblePlot$pvalue>=0.05] = ">=0.05"
    dataBubblePlot$pvalueـlevel = factor(dataBubblePlot$pvalueـlevel, levels = c(">=0.05","<0.05"))
    ggplot(dataBubblePlot, aes(x = PathogenSum.R2, y = Clinical.Covariates,size = pvalueـlevel, color=coefficient)) + 
      geom_point(alpha = 0.7) + scale_colour_gradient2(
        limits=c(-0.25, 0.25),
        low = muted("red"),
        mid = "grey",
        high = muted("blue"),
        midpoint = 0
      )+ scale_y_discrete(limits = order)+theme_bw() + 
      theme(legend.position = "none", panel.border = element_blank(), 
            panel.grid.major = element_line(colour = "grey",linewidth = 0.075, linetype = 2),
            panel.grid.minor = element_line(colour = "grey",linewidth = 0.075, linetype = 2), axis.line = element_line(color = "black"),
            axis.text.x  = element_text(color = "black"), axis.text.y  = element_text(color = "black"))
    
    ggsave(paste0('MultiCompMbio/results/bubble_PathogenSum_',compartment,'_',followuptime,'.jpg'), width = 4, height = 5)
    
  }
}


```


```{r DMM clusters}

### Perform filtering of the dataset to conduct clustering in each compartment separately

############## SingleCell Experiment

counts_filtered = counts[-which(rowSums(counts)<=1000),]
MetaData_filtered = MetaData[row.names(counts_filtered),]

df <- data.frame(x = names(counts_filtered))
taxa <- df %>% tidyr::separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                                    "Family", "Genus"), sep = "\\.")

# use Genus as row names. If Genus is repeated, use the full name
taxa$Genus[which(duplicated(taxa$Genus))] = names(counts_filtered)[which(duplicated(taxa$Genus))]
row.names(taxa) <- taxa$Genus

names(counts_filtered) = row.names(taxa)
# creat a single cell experiment object
se <- SingleCellExperiment(assays = list(counts = t(counts_filtered)),
                           colData = MetaData_filtered,
                           rowData = taxa)


### Transformations
se <- mia::transformSamples(se,  method = "relabundance")
se <- mia::transformSamples(x = se, abund_values = "counts", method = "clr", 
                            pseudocount = 1, name = "clr")

# keep taxa that exceed a specific prevalence and detection thresholds
se <- mia::subsetByPrevalentTaxa(se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                 prevalence = 5/100)

# Calculate Alpha Diversity
se <- mia::estimateDiversity(se, 
                             abund_values = "relabundance",
                             index = "shannon", 
                             name = "ShannonIndex")

filtered_taxa_table = as.data.frame(assays(se)$counts)
df =  as.data.frame(colData(se))

write.csv(filtered_taxa_table,"MultiCompMbio/results/filtered_taxa_table.csv")
write.csv(df,"MultiCompMbio/results/filtered_taxa_metadata.csv")

table(df$studydayfollowup, df$compartment)

n_distinct(df$SubjectID)


# Clustering 


#  DMM Clustering - baseline

# oral
compartment = "oral"
indx = which(se$compartment== compartment & se$studydayfollowup == "baseline")
se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)

P <- plotDMNFit(se_dmn, type = "laplace")

save_plot(paste0('MultiCompMbio/results/',compartment,'_DMN_modelFit.pdf'), P,
          base_aspect_ratio = 1.5, base_height =7)

RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit

RS.best_fit <- getDMN(se_dmn)[[3]]

# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")

RS.pred$group = se_compartment$group
cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

# check this plot and change the cluster name accordingly 
ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "Intermediate"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "LowDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==3)] <- "HighDiversity"



df = as.data.frame(colData(se_compartment))

df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/',compartment,'_DMM_baseline_clusters.csv'))
# lung 
compartment = "lung"
indx = which(se$compartment== compartment & se$studydayfollowup == "baseline")
se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)

P <- plotDMNFit(se_dmn, type = "laplace")

save_plot(paste0('MultiCompMbio/results/',compartment,'_DMN_modelFit.pdf'), P,
          base_aspect_ratio = 1.5, base_height =7)

RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit

RS.best_fit <- getDMN(se_dmn)[[3]]

# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")

RS.pred$group = se_compartment$group
cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

# check this plot and change the cluster name accordingly 
ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "LowDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "Intermediate"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==3)] <- "HighDiversity"



df = as.data.frame(colData(se_compartment))

df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggviolin(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/',compartment,'_DMM_baseline_clusters.csv'))
# gut #
compartment = "gut"
indx = which(se$compartment== compartment & se$studydayfollowup == "baseline")
se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)

P <- plotDMNFit(se_dmn, type = "laplace")

save_plot(paste0('MultiCompMbio/results/',compartment,'_DMN_modelFit.pdf'), P,
          base_aspect_ratio = 1.5, base_height =7)

RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit

RS.best_fit <- getDMN(se_dmn)[[3]]

# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")

RS.pred$group = se_compartment$group
cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

# check this plot and change the cluster name accordingly 
ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "HighDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "LowDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==3)] <- "Intermediate"


df = as.data.frame(colData(se_compartment))

df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)


meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/',compartment,'_DMM_baseline_clusters.csv'))
#  DMM Clustering - all time points

# oral ##
compartment = "oral"
indx = which(se$compartment== compartment & (se$studydayfollowup == "baseline"|se$studydayfollowup == "middle"|
                                               se$studydayfollowup == "late"))
se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)

P <- plotDMNFit(se_dmn, type = "laplace")

save_plot(paste0('MultiCompMbio/results/',compartment,'_all_DMN_modelFit.pdf'), P,
          base_aspect_ratio = 1.5, base_height =7)

RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit

RS.best_fit <- getDMN(se_dmn)[[3]]

# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")

RS.pred$group = se_compartment$group
cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

# check this plot and change the cluster name accordingly 
ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "Intermediate"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "HighDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==3)] <- "LowDiversity"

df = as.data.frame(colData(se_compartment))

df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/',compartment,'_DMM_alltimepoints_clusters.csv'))
#lung
compartment = "lung"
indx = which(se$compartment== compartment & (se$studydayfollowup == "baseline"|se$studydayfollowup == "middle"|
                                               se$studydayfollowup == "late"))
se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)

P <- plotDMNFit(se_dmn, type = "laplace")

save_plot(paste0('MultiCompMbio/results/',compartment,'_DMN_modelFit.pdf'), P,
          base_aspect_ratio = 1.5, base_height =7)

RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit

RS.best_fit <- getDMN(se_dmn)[[3]]

# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")

RS.pred$group = se_compartment$group
cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

# check this plot and change the cluster name accordingly 
ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "LowDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "Intermediate"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==3)] <- "HighDiversity"



df = as.data.frame(colData(se_compartment))

df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/',compartment,'_DMM_alltimepoints_clusters.csv'))
##gut 
compartment = "gut"
indx = which(se$compartment== compartment & (se$studydayfollowup == "baseline"|se$studydayfollowup == "middle"|
                                               se$studydayfollowup == "late"))
se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)

P <- plotDMNFit(se_dmn, type = "laplace")

save_plot(paste0('MultiCompMbio/results/',compartment,'_DMN_modelFit.pdf'), P,
          base_aspect_ratio = 1.5, base_height =7)

RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit

RS.best_fit <- getDMN(se_dmn)[[3]]

# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")

RS.pred$group = se_compartment$group
cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

# check this plot and change the cluster name accordingly 
ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "HighDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "LowDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==3)] <- "Intermediate"


df = as.data.frame(colData(se_compartment))

df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/',compartment,'_DMM_alltimepoints_clusters.csv'))

# baseline COVID clustering

indx = which(se$COVID == "Yes")

se_compartment = se[,indx]

se_dmn <- runDMN(se_compartment, name = "DMN", k = 1:7)


RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
RS.best_fit
RS.best_fit <- getDMN(se_dmn)[[2]]


# # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id.
# Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")


cluster_number <- RS.pred$cluster
colData(se_compartment)$cluster <- RS.pred$cluster

df = as.data.frame(colData(se_compartment))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==1)] <- "HighDiversity"
colData(se_compartment)$cluster[which(colData(se_compartment)$cluster==2)] <- "LowDiversity"


df = as.data.frame(colData(se_compartment))


df$cluster <- as.factor(df$cluster )
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(se_compartment))
RS.pred$cluster = meta_data$cluster
RS.pred$SubjectID = meta_data$SubjectID
RS.pred$group = meta_data$group
RS.pred$SampleType = meta_data$SampleType
write.csv(RS.pred, paste0('MultiCompMbio/results/covid_DMM_clusters.csv'))


# Combine clusters so far


baseline_oral_clusters = read.csv("MultiCompMbio/results/oral_DMM_baseline_clusters.csv")
baseline_lung_clusters = read.csv("MultiCompMbio/results/lung_DMM_baseline_clusters.csv")
baseline_gut_clusters = read.csv("MultiCompMbio/results/gut_DMM_baseline_clusters.csv")

baseline_clusters = rbind(baseline_oral_clusters,baseline_lung_clusters,baseline_gut_clusters)

names(baseline_clusters)[3] = "baselineonlyanalysis_cluster" 


alltimepoints_oral_clusters = read.csv("MultiCompMbio/results/oral_DMM_alltimepoints_clusters.csv")
alltimepoints_lung_clusters = read.csv("MultiCompMbio/results/lung_DMM_alltimepoints_clusters.csv")
alltimepoints_gut_clusters = read.csv("MultiCompMbio/results/gut_DMM_alltimepoints_clusters.csv")

alltimepoints_clusters = rbind(alltimepoints_oral_clusters,alltimepoints_lung_clusters,alltimepoints_gut_clusters)
names(alltimepoints_clusters)[3] = "alltimepoints_cluster" 

dmm_clusters = merge(baseline_clusters,alltimepoints_clusters, by = "sample_id", all = TRUE)
names(dmm_clusters) 
dmm_clusters = dmm_clusters[,c("sample_id", "baselineonlyanalysis_cluster","alltimepoints_cluster", "SampleType.y" )]
names(dmm_clusters)[4] = "SampleType"

oral_abundances = read.csv("MultiCompMbio/results/oral_abundances_alltimepoints.csv")
lung_abundances = read.csv("MultiCompMbio/results/lung_abundances_alltimepoints.csv")
gut_abundances = read.csv("MultiCompMbio/results/gut_abundances_alltimepoints.csv")
covid_abundances = read.csv("MultiCompMbio/results/covid_abundances_alltimepoints.csv")

abundances = rbind(oral_abundances,lung_abundances,gut_abundances,covid_abundances)
dmm_clusters = merge(dmm_clusters,abundances,by = "sample_id", all = TRUE)
names(dmm_clusters)
dmm_clusters = dmm_clusters[,-which(names(dmm_clusters)=="X")]

covid_baseline_clusters = read.csv("MultiCompMbio/results/covid_DMM_clusters.csv")
covid_baseline_clusters = covid_baseline_clusters[,c(2,3)]
names(covid_baseline_clusters)[2] = "COVID_clusters"
dmm_clusters = merge(dmm_clusters,covid_baseline_clusters, by = "sample_id", all = TRUE)

write.csv(dmm_clusters, "MultiCompMbio/results/MultiCompMbio_all_clusters_withCOVID.csv")



########### MGH data processing for cluster analysis

mgh.lung.counts = read.csv("MultiCompMbio/data/DAT_20230606_JMV_COVID_Sputum_Reads_Filtered.csv", row.names = 1)
mgh.lung.counts = mgh.lung.counts[-which(rowSums(mgh.lung.counts)<1000),]


mgh.lung.meta = read.csv("MultiCompMbio/data/mghsputum_6.25.23.csv", row.names = 1)
common_ids = intersect(row.names(mgh.lung.counts),row.names(mgh.lung.meta))
mgh.lung.counts = mgh.lung.counts[common_ids,]
mgh.lung.meta = mgh.lung.meta[common_ids,]

mgh.gut.counts = read.csv("MultiCompMbio/data/DAT_20230606_JMV_COVID_Stool_Reads_Filtered.csv", row.names = 1)
if(length(which(rowSums(mgh.lung.counts)<1000))>0){
  mgh.gut.counts = mgh.gut.counts[-which(rowSums(mgh.lung.counts)<1000),]
}
mgh.gut.meta = read.csv("MultiCompMbio/data/mghstool_6.25.23.csv", row.names = 1)
common_ids = intersect(row.names(mgh.gut.counts),row.names(mgh.gut.meta))
mgh.gut.counts = mgh.gut.counts[common_ids,]
mgh.gut.meta = mgh.gut.meta[common_ids,]


### lung analysis

mgh.counts = mgh.lung.counts
mgh.meta.data = mgh.lung.meta
cols = "#df8f44"


df <- data.frame(x = names(mgh.counts))
mgh.taxa <- df %>% tidyr::separate(x, c("Genus", "Species"), sep = "\\.")

row.names(mgh.taxa) = names(mgh.counts)



identical(row.names(mgh.meta.data), row.names(mgh.counts))

# creat a single cell experiment object
mgh.se <- SingleCellExperiment(assays = list(counts = t(mgh.counts)),
                               colData = mgh.meta.data,
                               rowData = mgh.taxa)

taxonomyRanks(mgh.se)
nrow(mgh.se)
mgh.se <- agglomerateByRank(mgh.se, rank="Genus")
nrow(mgh.se)


### Transformations
mgh.se <- mia::transformSamples(mgh.se, method = "relabundance")
mgh.se <- mia::transformSamples(x = mgh.se, abund_values = "counts", method = "clr", 
                                pseudocount = 1, name = "clr")
dim(mgh.se)
# keep taxa that exceed a specific prevalence and detection thresholds
mgh.se <- mia::subsetByPrevalentTaxa(mgh.se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                     prevalence = 5/100)
dim(mgh.se)

mgh.se <- mia::estimateDiversity(mgh.se, 
                                 abund_values = "relabundance",
                                 index = "shannon", 
                                 name = "ShannonIndex")



mgh.clr = as.data.frame(assays(mgh.se)$clr)
mgh.clr = as.data.frame(t(mgh.clr[,which(mgh.se$time_SP=="sputum1")]))

mgh.clr$ShannonIndex = mgh.se$ShannonIndex[which(mgh.se$time_SP=="sputum1")]
mgh.clr$survival_time60_fromICUAdmission = mgh.se$survival_time60_fromICUAdmission[which(mgh.se$time_SP=="sputum1")]
mgh.clr$SurvivalTimeFromAdmission = mgh.se$SurvivalTimeFromAdmission[which(mgh.se$time_SP=="sputum1")]
mgh.clr$survival_status60 = mgh.se$survival_status60[which(mgh.se$time_SP=="sputum1")]


write.csv(mgh.clr, "MultiCompMbio/results/MGH_lung_clr_sputum1.csv")


######### clustering at baseline

mgh.se_dmm = mgh.se[,which(mgh.se$time_SP == "sputum1")]

mgh.se_dmm <- runDMN(mgh.se_dmm, name = "DMN", k = 1:7)


RS.best_fit <- getBestDMNFit(mgh.se_dmm, type = "laplace")
RS.best_fit # three clusters

RS.best_fit <- getDMN(mgh.se_dmm)[[3]]
# 
# # # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id. 
# # Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")
# 

cluster_number <- RS.pred$cluster
colData(mgh.se_dmm)$cluster <- RS.pred$cluster

df = as.data.frame(colData(mgh.se_dmm))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(mgh.se_dmm)$cluster[which(colData(mgh.se_dmm)$cluster==1)] <- "LowDiversity"
colData(mgh.se_dmm)$cluster[which(colData(mgh.se_dmm)$cluster==2)] <- "Intermediate"
colData(mgh.se_dmm)$cluster[which(colData(mgh.se_dmm)$cluster==3)] <- "HighDiversity" 



df = as.data.frame(colData(mgh.se_dmm))
df$cluster = as.factor(df$cluster)
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)

meta_data = as.data.frame(colData(mgh.se_dmm))
RS.pred$cluster = meta_data$cluster
RS.pred$sample_id = row.names(meta_data)

write.csv(RS.pred, paste0('MultiCompMbio/results/MGH_lung_DMM_baseline_clusters.csv'))

############## Anaerobes
tracheal_df = as.data.frame(colData(mgh.se))

tracheal_rel_abund_assay = as.data.frame(assays(mgh.se)$relabundance)

tracheal_df = tracheal_df[-which(tracheal_df$time_SP%in%c("sputum5", "sputum6")),]
tracheal_df$time_SP = as.factor(tracheal_df$time_SP)

tracheal_rel_abund_assay_df <- tracheal_rel_abund_assay[,row.names(tracheal_df)]
tracheal_rel_abund_assay_df$Genus <- rownames(tracheal_rel_abund_assay_df)

#### import this to create a table of bugs classified
tracheal_bugsClassified = read.csv("MultiCompMbio/data/tracheal_bugsClassified.csv")
tracheal_bugsClassified = tracheal_bugsClassified[,-1]

tracheal_rel_abund_assay_df <- merge(tracheal_rel_abund_assay_df, tracheal_bugsClassified, by = "Genus", all.x = TRUE)

###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
tracheal_rel_abund_assay_df$O2Req[is.na(tracheal_rel_abund_assay_df$O2Req)] <- "Unclassifiable"

anaerobeslist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$O2Req=="Anaerobe"]
aerobeslist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$O2Req=="Aerobe"]
facultanaerobeslist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$O2Req=="Facultative anaerobe"]
Microaerophilelist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$O2Req=="Microaerophile"]
Unclassifiableslist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$O2Req=="Unclassifiable"]
Variablelist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$O2Req=="Variable"]

trachealbugsbyOxygen <- as.data.frame(t(tracheal_rel_abund_assay_df))
### remove last row of pathogens
trachealbugsbyOxygen <- trachealbugsbyOxygen[1:dim(trachealbugsbyOxygen)[1]-1, ]

#### in the trachealbugsbyoxygen, make colnames by the genus variable
colnames(trachealbugsbyOxygen) <- trachealbugsbyOxygen[1,]
#### remove the anaerobe and 02 req rows
trachealbugsbyOxygen <- trachealbugsbyOxygen[seq(2,dim(trachealbugsbyOxygen)[1]-1,1), ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

trachealbugsbyOxygenSample_id  <-as.data.frame(rownames(trachealbugsbyOxygen))
trachealbugsbyOxygen <- as.data.frame(sapply(trachealbugsbyOxygen,as.numeric))

trachealbugsbyOxygen$x <- rownames(trachealbugsbyOxygen)
trachealbugsbyOxygenSample_id$x <- rownames(trachealbugsbyOxygenSample_id)
trachealbugsbyOxygen <- merge(trachealbugsbyOxygen, trachealbugsbyOxygenSample_id, by = "x")
rownames(trachealbugsbyOxygen) <- trachealbugsbyOxygen$`rownames(trachealbugsbyOxygen)`
trachealbugsbyOxygen$x <- NULL
names(trachealbugsbyOxygen)[dim(trachealbugsbyOxygen)[2]] = "sample_id"

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(MicroaerophileSum = rowSums(select(., .dots = all_of(Microaerophilelist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(VariableSum = rowSums(select(., .dots = all_of(Variablelist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(UnclassifiableSum = rowSums(select(., .dots = all_of(Unclassifiableslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen$totalO2sum <- trachealbugsbyOxygen$AnaerobeSum + trachealbugsbyOxygen$AerobeSum + trachealbugsbyOxygen$FacultAnaerobeSum  + trachealbugsbyOxygen$MicroaerophileSum  + trachealbugsbyOxygen$VariableSum + trachealbugsbyOxygen$UnclassifiableSum 
###mutate total to 1 and then unclassifiable as difference from total
trachealbugsbyOxygen$totalO2sum <- 1
trachealbugsbyOxygen$UnclassifiableSum <- 1 - (trachealbugsbyOxygen$AnaerobeSum + trachealbugsbyOxygen$AerobeSum + trachealbugsbyOxygen$FacultAnaerobeSum  + trachealbugsbyOxygen$MicroaerophileSum  + trachealbugsbyOxygen$VariableSum )

tracheal_bugsClassifiedSum <- subset(trachealbugsbyOxygen, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum,MicroaerophileSum, VariableSum , UnclassifiableSum))

### incorporate with the tracheal_df 
tracheal_df_sum <- merge(tracheal_df,tracheal_bugsClassifiedSum, by = 'row.names', all = TRUE)
tracheal_abundances = tracheal_df_sum[,which(names(tracheal_df_sum)%in%c("Row.names","ShannonIndex","AnaerobeSum", "AerobeSum", "FacultAnaerobeSum" ))]
names(tracheal_abundances)[1] = "sample_id"

### Oral or Respiratory Pathogen analysis - Lung

tracheal_df = as.data.frame(colData(mgh.se))

tracheal_rel_abund_assay = as.data.frame(assays(mgh.se)$relabundance)

tracheal_df = tracheal_df[-which(tracheal_df$time_SP%in%c("sputum5", "sputum6")),]
tracheal_df$time_SP = as.factor(tracheal_df$time_SP)

tracheal_rel_abund_assay_df <- tracheal_rel_abund_assay[,row.names(tracheal_df)]
tracheal_rel_abund_assay_df$Genus <- rownames(tracheal_rel_abund_assay_df)


#### import this to create a table of bugs classified
tracheal_bugsClassified = read.csv("MultiCompMbio/data/tracheal_bugsClassified.csv")
tracheal_bugsClassified = tracheal_bugsClassified[,-1]

tracheal_rel_abund_assay_df <- merge(tracheal_rel_abund_assay_df, tracheal_bugsClassified, by = "Genus", all.x = TRUE)

###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
tracheal_rel_abund_assay_df$OralorRespiratoryPath[is.na(tracheal_rel_abund_assay_df$OralorRespiratoryPath)] <- "Other"
tracheal_rel_abund_assay_df$OralorRespiratoryPath[which(tracheal_rel_abund_assay_df$OralorRespiratoryPath=="")] <- "Other"
table(tracheal_rel_abund_assay_df$OralorRespiratoryPath)

orallist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Oral"]
pathogenlist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Pathogenic"]
otherlist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Other"]


trachealbugsbyOralPath <- as.data.frame(t(tracheal_rel_abund_assay_df))
### remove O2Req row of
trachealbugsbyOralPath <- trachealbugsbyOralPath[-which(row.names(trachealbugsbyOralPath)=="O2Req"), ]


#### in the trachealbugsbyOralPath, make colnames by the genus variable
colnames(trachealbugsbyOralPath) <- trachealbugsbyOralPath[1,]
#### remove the anaerobe and OralPath rows
trachealbugsbyOralPath1 <- trachealbugsbyOralPath[-c(1,dim(trachealbugsbyOralPath)[1]), ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

trachealbugsbyOralPathSample_id  <-as.data.frame(rownames(trachealbugsbyOralPath))
trachealbugsbyOralPath <- as.data.frame(sapply(trachealbugsbyOralPath,as.numeric))

trachealbugsbyOralPath$x <- rownames(trachealbugsbyOralPath)
trachealbugsbyOralPathSample_id$x <- rownames(trachealbugsbyOralPathSample_id)
trachealbugsbyOralPath <- merge(trachealbugsbyOralPath, trachealbugsbyOralPathSample_id, by = "x")
rownames(trachealbugsbyOralPath) <- trachealbugsbyOralPath$`rownames(trachealbugsbyOralPath)`
trachealbugsbyOralPath$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
trachealbugsbyOralPath <- trachealbugsbyOralPath %>% 
  mutate(PathogenSum = rowSums(select(., .dots = all_of(pathogenlist))))
trachealbugsbyOralPath$PathogenSum

trachealbugsbyOralPath <- trachealbugsbyOralPath %>% 
  mutate(OralSum = rowSums(select(., .dots = all_of(orallist))))
trachealbugsbyOralPath$OralSum

trachealbugsbyOralPath <- trachealbugsbyOralPath %>% 
  mutate(OtherSum = rowSums(select(., .dots = all_of(otherlist))))
trachealbugsbyOralPath$OtherSum


trachealbugsbyOralPath$totalOralsum <- trachealbugsbyOralPath$OralSum + trachealbugsbyOralPath$PathogenSum  
###mutate total to 1 and then other as difference from total
trachealbugsbyOralPath$totalOralsum <- 1
trachealbugsbyOralPath$OtherSum <- 1 - (trachealbugsbyOralPath$OralSum + trachealbugsbyOralPath$PathogenSum)

tracheal_bugsClassifiedSum <- subset(trachealbugsbyOralPath, select = c(OralSum, PathogenSum, OtherSum))

row.names(tracheal_df)
### incorporate with the tracheal_df 
tracheal_df_sum <- merge(tracheal_df,tracheal_bugsClassifiedSum, by = 'row.names', all = TRUE)
tracheal_df_sum$OralSum

names(tracheal_df_sum)[1] = "sample_id"
tracheal_abundances1 = tracheal_df_sum[,which(names(tracheal_df_sum)%in%c("sample_id","OralSum", "PathogenSum", "OtherSum" ))]
tracheal_abundances = merge(tracheal_abundances,tracheal_abundances1, by = 'sample_id', all = TRUE)

dmm_clusters = read.csv("MultiCompMbio/results/MGH_lung_DMM_baseline_clusters.csv")
dmm_clusters = dmm_clusters[,-1]
tracheal_abundances = left_join(tracheal_abundances, dmm_clusters, by = "sample_id")
tracheal_abundances = tracheal_abundances[-which(is.na(tracheal_abundances$cluster)),]

write.csv(tracheal_abundances, "MultiCompMbio/results/MGH_lung_clusters.csv")

### gut analysis

mgh.counts = mgh.gut.counts
mgh.meta.data = mgh.gut.meta
mgh.meta.data$totalreads = rowSums(mgh.counts)
mgh.meta.data$sample_id = row.names(mgh.meta.data)
mgh.meta.data$time = mgh.meta.data$time_ST
cols =  "#00a1d5"

df <- data.frame(x = names(mgh.counts))
mgh.taxa <- df %>% tidyr::separate(x, c("Genus", "Species"), sep = "\\.")

row.names(mgh.taxa) = names(mgh.counts)


identical(row.names(mgh.meta.data), row.names(mgh.counts))

# creat a single cell experiment object
mgh.se <- SingleCellExperiment(assays = list(counts = t(mgh.counts)),
                               colData = mgh.meta.data,
                               rowData = mgh.taxa)



taxonomyRanks(mgh.se)
nrow(mgh.se)
mgh.se <- agglomerateByRank(mgh.se, rank="Genus")
nrow(mgh.se)


### Transformations
mgh.se <- mia::transformSamples(mgh.se, method = "relabundance")
mgh.se <- mia::transformSamples(x = mgh.se, abund_values = "counts", method = "clr", 
                                pseudocount = 1, name = "clr")

# keep taxa that exceed a specific prevalence and detection thresholds
mgh.se <- mia::subsetByPrevalentTaxa(mgh.se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                                     prevalence = 5/100)
mgh.se <- mia::estimateDiversity(mgh.se, 
                                 abund_values = "relabundance",
                                 index = "shannon", 
                                 name = "ShannonIndex")


mgh.clr = as.data.frame(assays(mgh.se)$clr)
mgh.clr = as.data.frame(t(mgh.clr[,which(mgh.se$time_ST=="stool1")]))

mgh.clr$ShannonIndex =se$ShannonIndex[which(mgh.se$time_ST=="stool1")]
mgh.clr$survival_time60_fromICUAdmission = mgh.se$survival_time60_fromICUAdmission[which(mgh.se$time_ST=="stool1")]
mgh.clr$survival_status60 = mgh.se$survival_status60[which(mgh.se$time_ST=="stool1")]

write.csv(mgh.clr, "MultiCompMbio/results/MGH_gut_clr_stool1.csv")

##### clustering at baseline
mgh.se_dmm <- mgh.se[,which(mgh.se$time_ST == "stool1")]

mgh.se_dmn <- runDMN(mgh.se_dmm, name = "DMN", k = 1:7)

RS.best_fit <- getBestDMNFit(mgh.se_dmn, type = "laplace")
RS.best_fit # 3 clusters

RS.best_fit <- getDMN(mgh.se_dmn)[[3]]
# 
# # # The function ‘mixture’ with ‘assign = TRUE’ will generate the cluster membership for each sample_id. 
# # Use this result as cluster assignment and continue with downstream analyses.
RS.pred <- mixture(RS.best_fit, assign = TRUE) %>% enframe(name = "sample_id", value = "cluster")
# 

cluster_number <- RS.pred$cluster
colData(mgh.se_dmn)$cluster <- RS.pred$cluster

df = as.data.frame(colData(mgh.se_dmn))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
          color = "cluster",add = "jitter")

colData(mgh.se_dmn)$cluster[which(colData(mgh.se_dmn)$cluster==1)] <- "Intermediate" 
colData(mgh.se_dmn)$cluster[which(colData(mgh.se_dmn)$cluster==2)] <- "HighDiversity"
colData(mgh.se_dmn)$cluster[which(colData(mgh.se_dmn)$cluster==3)] <- "LowDiversity" 



df = as.data.frame(colData(mgh.se_dmn))
df$cluster = as.factor(df$cluster)
comp <- split(t(combn(levels(df$cluster), 2)),
              seq(nrow(t(combn(levels(df$cluster), 2)))))

ggboxplot(df, x = "cluster", y = "ShannonIndex",
         color = "cluster",add = "jitter", palette = "lancet", legend = "none" ) +
  stat_compare_means(comparisons = comp)


meta_data = as.data.frame(colData(mgh.se_dmn))
RS.pred$cluster = meta_data$cluster
RS.pred$sample_id = meta_data$sample_id

write.csv(RS.pred, paste0('MultiCompMbio/results/MGH_gut_DMM_baseline_clusters.csv'))


############## Anaerobes
gut_df = as.data.frame(colData(mgh.se))

gut_rel_abund_assay = as.data.frame(assays(mgh.se)$relabundance)

gut_df = gut_df[-which(gut_df$time%in%c("stool5", "stool6","stool7")),]
gut_df$time = as.factor(gut_df$time)

gut_rel_abund_assay_df <- gut_rel_abund_assay[,row.names(gut_df)]
gut_rel_abund_assay_df$Genus <- rownames(gut_rel_abund_assay_df)


#### import this to create a table of bugs classified
gut_bugsClassified = read.csv("MultiCompMbio/data/gut_bugsClassified.csv")
gut_bugsClassified = gut_bugsClassified[,-1]

gut_rel_abund_assay_df <- merge(gut_rel_abund_assay_df, gut_bugsClassified, by = "Genus", all.x = TRUE)

###### take the df from gut_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
gut_rel_abund_assay_df$O2Req[is.na(gut_rel_abund_assay_df$O2Req)] <- "Unclassifiable"

anaerobeslist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$O2Req=="Anaerobe"]
aerobeslist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$O2Req=="Aerobe"]
facultanaerobeslist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$O2Req=="Facultative anaerobe"]
Microaerophilelist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$O2Req=="Microaerophile"]
Unclassifiableslist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$O2Req=="Unclassifiable"]
Variablelist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$O2Req=="Variable"]

gutbugsbyOxygen <- as.data.frame(t(gut_rel_abund_assay_df))
### remove last row of pathogens
gutbugsbyOxygen <- gutbugsbyOxygen[1:dim(gutbugsbyOxygen)[1]-1, ]

#### in the gutbugsbyoxygen, make colnames by the genus variable
colnames(gutbugsbyOxygen) <- gutbugsbyOxygen[1,]
#### remove the anaerobe and 02 req rows
gutbugsbyOxygen <- gutbugsbyOxygen[2:dim(gutbugsbyOxygen)[1]-1, ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

gutbugsbyOxygenSample_id  <-as.data.frame(rownames(gutbugsbyOxygen))
gutbugsbyOxygen <- as.data.frame(sapply(gutbugsbyOxygen,as.numeric))

gutbugsbyOxygen$x <- rownames(gutbugsbyOxygen)
gutbugsbyOxygenSample_id$x <- rownames(gutbugsbyOxygenSample_id)
gutbugsbyOxygen <- merge(gutbugsbyOxygen, gutbugsbyOxygenSample_id, by = "x")
rownames(gutbugsbyOxygen) <- gutbugsbyOxygen$`rownames(gutbugsbyOxygen)`
gutbugsbyOxygen$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
gutbugsbyOxygen <- gutbugsbyOxygen %>% 
  mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
gutbugsbyOxygen$AnaerobeSum

gutbugsbyOxygen <- gutbugsbyOxygen %>% 
  mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
gutbugsbyOxygen$AnaerobeSum

gutbugsbyOxygen <- gutbugsbyOxygen %>% 
  mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
gutbugsbyOxygen$AnaerobeSum

gutbugsbyOxygen <- gutbugsbyOxygen %>% 
  mutate(MicroaerophileSum = rowSums(select(., .dots = all_of(Microaerophilelist))))
gutbugsbyOxygen$AnaerobeSum

gutbugsbyOxygen <- gutbugsbyOxygen %>% 
  mutate(VariableSum = rowSums(select(., .dots = all_of(Variablelist))))
gutbugsbyOxygen$AnaerobeSum

gutbugsbyOxygen <- gutbugsbyOxygen %>% 
  mutate(UnclassifiableSum = rowSums(select(., .dots = all_of(Unclassifiableslist))))
gutbugsbyOxygen$AnaerobeSum

gutbugsbyOxygen$totalO2sum <- gutbugsbyOxygen$AnaerobeSum + gutbugsbyOxygen$AerobeSum + gutbugsbyOxygen$FacultAnaerobeSum  + gutbugsbyOxygen$MicroaerophileSum  + gutbugsbyOxygen$VariableSum + gutbugsbyOxygen$UnclassifiableSum 
###mutate total to 1 and then unclassifiable as difference from total
gutbugsbyOxygen$totalO2sum <- 1
gutbugsbyOxygen$UnclassifiableSum <- 1 - (gutbugsbyOxygen$AnaerobeSum + gutbugsbyOxygen$AerobeSum + gutbugsbyOxygen$FacultAnaerobeSum  + gutbugsbyOxygen$MicroaerophileSum  + gutbugsbyOxygen$VariableSum )

gut_bugsClassifiedSum <- subset(gutbugsbyOxygen, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum,MicroaerophileSum, VariableSum , UnclassifiableSum))

### incorporate with the gut_df 
gut_df_sum <- merge(gut_df,gut_bugsClassifiedSum, by = 'row.names', all = TRUE)
#### plot of comparisons of abundance by clinical groups. 

gut_df_sum = gut_df_sum[-which(is.na(gut_df_sum$time)),]

gut_df_sum$sample_id
gut_abundances = gut_df_sum[,which(names(gut_df_sum)%in%c("sample_id","ShannonIndex","AnaerobeSum", "AerobeSum", "FacultAnaerobeSum" ))]


### Oral or Respiratory Pathogen analysis - gut


gut_df = as.data.frame(colData(mgh.se))

gut_rel_abund_assay = as.data.frame(assays(mgh.se)$relabundance)

gut_df = gut_df[-which(gut_df$time%in%c("stool5", "stool6", "stool7")),]
gut_df$time = as.factor(gut_df$time)

gut_rel_abund_assay_df <- gut_rel_abund_assay[,row.names(gut_df)]
gut_rel_abund_assay_df$Genus <- rownames(gut_rel_abund_assay_df)


#### import this to create a table of bugs classified
gut_bugsClassified = read.csv("MultiCompMbio/data/gut_bugsClassified.csv")
gut_bugsClassified = gut_bugsClassified[,-1]

gut_rel_abund_assay_df <- merge(gut_rel_abund_assay_df, gut_bugsClassified, by = "Genus", all.x = TRUE)

###### take the df from gut_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
gut_rel_abund_assay_df$OralorRespiratoryPath[is.na(gut_rel_abund_assay_df$OralorRespiratoryPath)] <- "Other"
gut_rel_abund_assay_df$OralorRespiratoryPath[which(gut_rel_abund_assay_df$OralorRespiratoryPath=="")] <- "Other"
table(gut_rel_abund_assay_df$OralorRespiratoryPath)

orallist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$OralorRespiratoryPath=="Oral"]
pathogenlist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$OralorRespiratoryPath=="Pathogenic"]
otherlist <- gut_rel_abund_assay_df$Genus[gut_rel_abund_assay_df$OralorRespiratoryPath=="Other"]


gutbugsbyOralPath <- as.data.frame(t(gut_rel_abund_assay_df))
### remove O2Req row of
gutbugsbyOralPath <- gutbugsbyOralPath[-which(row.names(gutbugsbyOralPath)=="O2Req"), ]


#### in the gutbugsbyOralPath, make colnames by the genus variable
colnames(gutbugsbyOralPath) <- gutbugsbyOralPath[1,]
#### remove the anaerobe and OralPath rows
gutbugsbyOralPath <- gutbugsbyOralPath[-c(1,dim(gutbugsbyOralPath)[1]), ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

gutbugsbyOralPathSample_id  <-as.data.frame(rownames(gutbugsbyOralPath))
gutbugsbyOralPath <- as.data.frame(sapply(gutbugsbyOralPath,as.numeric))

gutbugsbyOralPath$x <- rownames(gutbugsbyOralPath)
gutbugsbyOralPathSample_id$x <- rownames(gutbugsbyOralPathSample_id)
gutbugsbyOralPath <- merge(gutbugsbyOralPath, gutbugsbyOralPathSample_id, by = "x")
rownames(gutbugsbyOralPath) <- gutbugsbyOralPath$`rownames(gutbugsbyOralPath)`
gutbugsbyOralPath$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
gutbugsbyOralPath <- gutbugsbyOralPath %>% 
  mutate(PathogenSum = rowSums(select(., .dots = all_of(pathogenlist))))
gutbugsbyOralPath$PathogenSum

gutbugsbyOralPath <- gutbugsbyOralPath %>% 
  mutate(OralSum = rowSums(select(., .dots = all_of(orallist))))
gutbugsbyOralPath$OralSum

gutbugsbyOralPath <- gutbugsbyOralPath %>% 
  mutate(OtherSum = rowSums(select(., .dots = all_of(otherlist))))
gutbugsbyOralPath$OtherSum


gutbugsbyOralPath$totalOralsum <- gutbugsbyOralPath$OralSum + gutbugsbyOralPath$PathogenSum  
###mutate total to 1 and then other as difference from total
gutbugsbyOralPath$totalOralsum <- 1
gutbugsbyOralPath$OtherSum <- 1 - (gutbugsbyOralPath$OralSum + gutbugsbyOralPath$PathogenSum)

gut_bugsClassifiedSum <- subset(gutbugsbyOralPath, select = c(OralSum, PathogenSum, OtherSum))

row.names(gut_df)
### incorporate with the gut_df 
gut_df_sum <- merge(gut_df,gut_bugsClassifiedSum, by = 'row.names', all = TRUE)
gut_df_sum$OralSum
#### plot of comparisons of abundance by clinical groups. 

# gut_df_sum = gut_df_sum[-which(is.na(gut_df_sum$time)),]
# gut_df_sum$studydayfollowup = as.factor(gut_df_sum$studydayfollowup)
# gut_df_sum$studydayfollowup = ordered(gut_df_sum$studydayfollowup, levels = c("baseline", "middle","late"))

gut_abundances1 = gut_df_sum[,which(names(gut_df_sum)%in%c("sample_id","OralSum", "PathogenSum", "OtherSum" ))]
gut_abundances = merge(gut_abundances,gut_abundances1, by = 'sample_id', all = TRUE)

dmm_clusters = read.csv("MultiCompMbio/results/MGH_gut_DMM_baseline_clusters.csv")
dmm_clusters = dmm_clusters[,-1]
gut_abundances = left_join(gut_abundances, dmm_clusters, by = "sample_id")
gut_abundances = gut_abundances[-which(is.na(gut_abundances$cluster)),]

write.csv(gut_abundances, "MultiCompMbio/results/MGH_gut_clusters.csv")



# cluster assignment prediction

library(rCausalMGM)
library(bnstruct)
library(projpred)
library(BioNet)
library(nnet)
comp = "oral"
for(comp in c("oral","lung","gut")){
  relabundance = as.data.frame(assays(se)$relabundance[,which(se$compartment==comp & se$studydayfollowup=="baseline")])
  relabundance$sum = rowSums(relabundance)
  
  clr = as.data.frame(assays(se)$clr[,which(se$compartment==comp & se$studydayfollowup=="baseline")])
  ordered.clr = as.data.frame(t(clr[order(relabundance$sum,decreasing = TRUE)[1:50],]))
  
  clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_withCOVID.csv")
  row.names(clusters) = clusters$sample_id
  clusters = clusters[row.names(ordered.clr),c("ShannonIndex","baselineonlyanalysis_cluster")]
  names(clusters) = c("ShannonIndex","DMM_cluster")
  clusters$DMM_cluster[which(clusters$DMM_cluster=="HighDiversity")] = 0
  clusters$DMM_cluster[which(clusters$DMM_cluster=="Intermediate")] = 1
  clusters$DMM_cluster[which(clusters$DMM_cluster=="LowDiversity")] = 2
  clusters$DMM_cluster = as.numeric(clusters$DMM_cluster)
  
  pgm.data = merge(ordered.clr, clusters, by = "row.names", keep = TRUE)
  row.names(pgm.data) = pgm.data$Row.names
  pgm.data = pgm.data[,-1]
  
  
  cluster0 = which(pgm.data$DMM_cluster==0)
  cluster1 = which(pgm.data$DMM_cluster==1)
  cluster2 = which(pgm.data$DMM_cluster==2)
  
  cluster0_samples = row.names(pgm.data[cluster0,])
  cluster1_samples = row.names(pgm.data[cluster1,])
  cluster2_samples = row.names(pgm.data[cluster2,])
  
  folds_0 <- cvfolds(length(cluster0), 5, seed = 4000)
  folds_1 <- cvfolds(length(cluster1), 5, seed = 4000)
  folds_2 <- cvfolds(length(cluster2), 5, seed = 4000)
  
  fold = 1
  training.samples = c(cluster0_samples[which(folds_0!=fold)],
                       cluster1_samples[which(folds_1!=fold)],
                       cluster2_samples[which(folds_2!=fold)])
  training.data = pgm.data[training.samples,]
  
  testing.samples = c(cluster0_samples[which(folds_0==fold)],
                      cluster1_samples[which(folds_1==fold)],
                      cluster2_samples[which(folds_2==fold)])
  testing.data = pgm.data[testing.samples,]
  
  training.data$DMM_cluster = as.factor(training.data$DMM_cluster )
  
  g = rCausalMGM::fciMax(training.data, alpha = 0.1, fdr = TRUE)
  rCausalMGM::saveGraph(g, paste0("MultiCompMbio/results/",comp,"_fciMax_0.1.sif"))
}

comp = "lung"
for(comp in c("oral","lung","gut")){
  relabundance = as.data.frame(assays(se)$relabundance[,which(se$compartment==comp & se$studydayfollowup=="baseline")])
  relabundance$sum = rowSums(relabundance)
  
  clr = as.data.frame(assays(se)$clr[,which(se$compartment==comp & se$studydayfollowup=="baseline")])
  ordered.clr = as.data.frame(t(clr[order(relabundance$sum,decreasing = TRUE)[1:50],]))
  
  clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_withCOVID.csv")
  row.names(clusters) = clusters$sample_id
  clusters = clusters[row.names(ordered.clr),c("ShannonIndex","baselineonlyanalysis_cluster")]
  names(clusters) = c("ShannonIndex","DMM_cluster")
  clusters$DMM_cluster[which(clusters$DMM_cluster=="HighDiversity")] = 0
  clusters$DMM_cluster[which(clusters$DMM_cluster=="Intermediate")] = 1
  clusters$DMM_cluster[which(clusters$DMM_cluster=="LowDiversity")] = 2
  clusters$DMM_cluster = as.numeric(clusters$DMM_cluster)
  
  pgm.data = merge(ordered.clr, clusters, by = "row.names", keep = TRUE)
  row.names(pgm.data) = pgm.data$Row.names
  pgm.data = pgm.data[,-1]
  
  
  cluster0 = which(pgm.data$DMM_cluster==0)
  cluster1 = which(pgm.data$DMM_cluster==1)
  cluster2 = which(pgm.data$DMM_cluster==2)
  
  cluster0_samples = row.names(pgm.data[cluster0,])
  cluster1_samples = row.names(pgm.data[cluster1,])
  cluster2_samples = row.names(pgm.data[cluster2,])
  
  folds_0 <- cvfolds(length(cluster0), 5, seed = 4000)
  folds_1 <- cvfolds(length(cluster1), 5, seed = 4000)
  folds_2 <- cvfolds(length(cluster2), 5, seed = 4000)
  
  fold = 1
  training.samples = c(cluster0_samples[which(folds_0!=fold)],
                       cluster1_samples[which(folds_1!=fold)],
                       cluster2_samples[which(folds_2!=fold)])
  training.data = pgm.data[training.samples,]
  
  testing.samples = c(cluster0_samples[which(folds_0==fold)],
                      cluster1_samples[which(folds_1==fold)],
                      cluster2_samples[which(folds_2==fold)])
  testing.data = pgm.data[testing.samples,]
  
  training.data$DMM_cluster = as.factor(training.data$DMM_cluster )
  
  # Pause here and open Cytoscape to find markov balnket of DMM_cluster variable
  train.control <- trainControl(method = "cv",number = 10) # set the 10-fold cross-validation parameters
  training.data$DMM_cluster<-relevel(training.data$DMM_cluster, ref="0")
  
  if(comp == "oral"){
    model<-multinom(DMM_cluster~Pasteurellaceae_uncl+
                    Alloprevotella+
                    Anaerovoracaceae_ge+
                    Centipeda+
                    Fusobacterium+
                    Streptococcus+
                    Veillonella, data=training.data, 
                    trControl = train.control)
    
  }else if(comp == "lung"){
    model<-multinom(DMM_cluster~Porphyromonas+
                    Peptostreptococcus+
                    Gemella+
                    Actinomyces+
                    Rothia+
                    Treponema+
                    Bergeyella+
                    Alloprevotella+
                    Solobacterium+
                    Atopobium+
                    Neisseriaceae_uncl+
                    ShannonIndex, data=training.data, 
                    trControl = train.control)
    
    ###  ALIR COVID testing - lung taxa only

    covid_testing.data = as.data.frame(t(assays(se)$clr[c("Porphyromonas",
                                                          "Peptostreptococcus",
                                                          "Gemella",
                                                          "Actinomyces",
                                                          "Rothia",
                                                          "Treponema",
                                                          "Bergeyella",
                                                          "Alloprevotella",
                                                          "Solobacterium",
                                                          "Atopobium",
                                                          "Neisseriaceae_uncl"),which(se$COVID=="Yes")]))
    
    covid_testing.data$ShannonIndex = se$ShannonIndex[which(se$COVID=="Yes")]
    
    predicted_labels = predict(model,covid_testing.data)
    
    df = as.data.frame(colData(se))
    df = df[which(df$COVID=="Yes"),]
    df$dysbiosis_predicted_clusters_COVID = predicted_labels
    
    dysbiosis_predicted_clusters_COVID = df[,c("sample_id", "dysbiosis_predicted_clusters_COVID")]
    write.csv(dysbiosis_predicted_clusters_COVID,"MultiCompMbio/results/lung_dysbiosis_predicted_clusters_ALIR_COVID.csv")
    
    #MGH for testing
    mgh_clr = as.data.frame(read.csv("MultiCompMbio/results/MGH_lung_clr_sputum1.csv", row.names = 1))
    

    model_data = mgh_clr[,c("Porphyromonas",
                            "Peptostreptococcus",
                            "Gemella",
                            "Actinomyces",
                            "Rothia",
                            "Treponema",
                            "Alloprevotella",
                            "Solobacterium",
                            "Atopobium",
                            "ShannonIndex"),]
    
    model_data$Bergeyella = 0
    model_data$Neisseriaceae_uncl = 0
    
    
    predicted_labels = predict(model,model_data)
    
    
    
    df = data.frame(row.names(model_data),predicted_labels)
    names(df) = c("MGH_sample_id","dysbiosis_predicted_cluster")
    
    write.csv(df,"MultiCompMbio/results/lung_dysbiosis_predicted_clusters_MGH.csv")
    
  }else if(comp=="gut"){
    model<-multinom(DMM_cluster~Campylobacter+
                    Fenollaria+
                    Ezakiella+
                    Enterococcus+
                    ShannonIndex+
                    Blautia, data=training.data, 
                    trControl = train.control)
    
    #MGH for testing
    mgh_clr = as.data.frame(read.csv("MultiCompMbio/results/MGH_gut_clr_stool1.csv", row.names = 1))

    model_data = mgh_clr[,c("Enterococcus",
                            "ShannonIndex",
                            "Blautia"),]
    
    model_data$Campylobacter = 0
    model_data$Fenollaria= 0
    model_data$Ezakiella = 0

    
    predicted_labels = predict(model,model_data)
    
    
    
    df = data.frame(row.names(model_data),predicted_labels)
    names(df) = c("MGH_sample_id","dysbiosis_predicted_cluster")
    
    write.csv(df,"MultiCompMbio/results/gut_dysbiosis_predicted_clusters_MGH.csv")
  }
  
  
  
  summary(model)
  
  # 2-tailed z test
  z <- summary(model)$coefficients/summary(model)$standard.errors
  p <- (1 - pnorm(abs(z), 0, 1)) * 2
  p
  
  predicted_labels = predict(model,testing.data)
  predicted_prop = predict(model,testing.data,type="prob")
  cf1 = confusionMatrix(predicted_labels,as.factor(testing.data$DMM_cluster))
  sink(paste0('MultiCompMbio/results/confusionMatrix_',comp,'_testing.txt'))
  cf1
  sink()
  
  predicted_labels = as.character(predict(model,pgm.data,type="class"))
  row.names(pgm.data)
  alir_noCOVID = data.frame(row.names(pgm.data),predicted_labels)
  names(alir_noCOVID) = c("sample_id","dysbiosis_clusters")
  write.csv(alir_noCOVID,paste0("MultiCompMbio/results/",comp,"_dysbiosis_predicted_clusters.csv"))
  
}



# all cluster assignments

all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_withCOVID.csv")

oral_dysbiosis_predicted_clusters = read.csv("MultiCompMbio/results/oral_dysbiosis_predicted_clusters.csv")
oral_dysbiosis_predicted_clusters = oral_dysbiosis_predicted_clusters[,-1]
oral_dysbiosis_predicted_clusters$dysbiosis_clusters[which(oral_dysbiosis_predicted_clusters$dysbiosis_clusters=="0")] = "HighDiversity"
oral_dysbiosis_predicted_clusters$dysbiosis_clusters[which(oral_dysbiosis_predicted_clusters$dysbiosis_clusters=="1")] = "Intermediate"
oral_dysbiosis_predicted_clusters$dysbiosis_clusters[which(oral_dysbiosis_predicted_clusters$dysbiosis_clusters=="2")] = "LowDiversity"
names(oral_dysbiosis_predicted_clusters) = c("sample_id","predicted_cluster_dysbiosis")

lung_dysbiosis_predicted_clusters = read.csv("MultiCompMbio/results/lung_dysbiosis_predicted_clusters.csv")
lung_dysbiosis_predicted_clusters = lung_dysbiosis_predicted_clusters[,-1]
lung_dysbiosis_predicted_clusters$dysbiosis_clusters[which(lung_dysbiosis_predicted_clusters$dysbiosis_clusters=="0")] = "HighDiversity"
lung_dysbiosis_predicted_clusters$dysbiosis_clusters[which(lung_dysbiosis_predicted_clusters$dysbiosis_clusters=="1")] = "Intermediate"
lung_dysbiosis_predicted_clusters$dysbiosis_clusters[which(lung_dysbiosis_predicted_clusters$dysbiosis_clusters=="2")] = "LowDiversity"
names(lung_dysbiosis_predicted_clusters) = c("sample_id","predicted_cluster_dysbiosis")

gut_dysbiosis_predicted_clusters = read.csv("MultiCompMbio/results/gut_dysbiosis_predicted_clusters.csv")
gut_dysbiosis_predicted_clusters = gut_dysbiosis_predicted_clusters[,-1]
gut_dysbiosis_predicted_clusters$dysbiosis_clusters[which(gut_dysbiosis_predicted_clusters$dysbiosis_clusters=="0")] = "HighDiversity"
gut_dysbiosis_predicted_clusters$dysbiosis_clusters[which(gut_dysbiosis_predicted_clusters$dysbiosis_clusters=="1")] = "Intermediate"
gut_dysbiosis_predicted_clusters$dysbiosis_clusters[which(gut_dysbiosis_predicted_clusters$dysbiosis_clusters=="2")] = "LowDiversity"
names(gut_dysbiosis_predicted_clusters) = c("sample_id","predicted_cluster_dysbiosis")

dysbiosis_predicted_clusters = rbind(oral_dysbiosis_predicted_clusters,
                                     lung_dysbiosis_predicted_clusters,
                                     gut_dysbiosis_predicted_clusters)
all_clusters = left_join(all_clusters,dysbiosis_predicted_clusters, by = "sample_id")

lung_dysbiosis_predicted_clusters_ALIR_COVID = read.csv("MultiCompMbio/results/lung_dysbiosis_predicted_clusters_ALIR_COVID.csv")
lung_dysbiosis_predicted_clusters_ALIR_COVID = lung_dysbiosis_predicted_clusters_ALIR_COVID[,-1]
lung_dysbiosis_predicted_clusters_ALIR_COVID$dysbiosis_predicted_clusters_COVID[which(lung_dysbiosis_predicted_clusters_ALIR_COVID$dysbiosis_predicted_clusters_COVID=="0")] = "HighDiversity"
lung_dysbiosis_predicted_clusters_ALIR_COVID$dysbiosis_predicted_clusters_COVID[which(lung_dysbiosis_predicted_clusters_ALIR_COVID$dysbiosis_predicted_clusters_COVID=="1")] = "Intermediate"
lung_dysbiosis_predicted_clusters_ALIR_COVID$dysbiosis_predicted_clusters_COVID[which(lung_dysbiosis_predicted_clusters_ALIR_COVID$dysbiosis_predicted_clusters_COVID=="2")] = "LowDiversity"
names(lung_dysbiosis_predicted_clusters_ALIR_COVID) = c("sample_id","predicted_cluster_dysbiosis_ALIR_COVID")
all_clusters = left_join(all_clusters,lung_dysbiosis_predicted_clusters_ALIR_COVID, by = "sample_id")

MGH_lung_clusters = read.csv("MultiCompMbio/results/MGH_lung_clusters.csv")
MGH_lung_clusters = MGH_lung_clusters[,-1]
names(MGH_lung_clusters) = c("sample_id",         "ShannonIndex",      "AnaerobeSum",       "AerobeSum",
                             "FacultAnaerobeSum", "OralSum", "PathogenSum",      "OtherSum", "MGH_DMM_clusters")

MGH_gut_clusters = read.csv("MultiCompMbio/results/MGH_gut_clusters.csv")
MGH_gut_clusters = MGH_gut_clusters[,-1]
names(MGH_gut_clusters) = c("sample_id",         "ShannonIndex",      "AnaerobeSum",       "AerobeSum",
                            "FacultAnaerobeSum", "OralSum", "PathogenSum",      "OtherSum", "MGH_DMM_clusters")

MGH_clusters = rbind(MGH_lung_clusters,MGH_gut_clusters)

lung_dysbiosis_predicted_clusters_MGH = read.csv("MultiCompMbio/results/lung_dysbiosis_predicted_clusters_MGH.csv")
lung_dysbiosis_predicted_clusters_MGH = lung_dysbiosis_predicted_clusters_MGH[,-1]
names(lung_dysbiosis_predicted_clusters_MGH) = c("sample_id","predicted_cluster_dysbiosis")

gut_dysbiosis_predicted_clusters_MGH = read.csv("MultiCompMbio/results/gut_dysbiosis_predicted_clusters_MGH.csv")
gut_dysbiosis_predicted_clusters_MGH = gut_dysbiosis_predicted_clusters_MGH[,-1]
names(gut_dysbiosis_predicted_clusters_MGH) = c("sample_id","predicted_cluster_dysbiosis")

dysbiosis_predicted_clusters_MGH = rbind(lung_dysbiosis_predicted_clusters_MGH,gut_dysbiosis_predicted_clusters_MGH)

MGH_clusters = merge(MGH_clusters,dysbiosis_predicted_clusters_MGH, by = "sample_id")
MGH_clusters$predicted_cluster_dysbiosis[which(MGH_clusters$predicted_cluster_dysbiosis=="0")] = "HighDiversity"
MGH_clusters$predicted_cluster_dysbiosis[which(MGH_clusters$predicted_cluster_dysbiosis=="1")] = "Intermediate"
MGH_clusters$predicted_cluster_dysbiosis[which(MGH_clusters$predicted_cluster_dysbiosis=="2")] = "LowDiversity"

all_clusters = all_clusters[,-1]
all_clusters$MGH_DMM_clusters = NA

MGH_clusters_new = as.data.frame(matrix(NA, nrow = dim(MGH_clusters)[1], ncol = dim(all_clusters)[2]))
names(MGH_clusters_new) = names(all_clusters)

MGH_clusters_new$sample_id = MGH_clusters$sample_id
MGH_clusters_new$ShannonIndex = MGH_clusters$ShannonIndex
MGH_clusters_new$AnaerobeSum = MGH_clusters$AnaerobeSum
MGH_clusters_new$AerobeSum = MGH_clusters$AerobeSum
MGH_clusters_new$FacultAnaerobeSum = MGH_clusters$FacultAnaerobeSum
MGH_clusters_new$OralSum = MGH_clusters$OralSum
MGH_clusters_new$PathogenSum = MGH_clusters$PathogenSum
MGH_clusters_new$OtherSum = MGH_clusters$OtherSum
MGH_clusters_new$MGH_DMM_clusters = MGH_clusters$MGH_DMM_clusters
MGH_clusters_new$predicted_cluster_dysbiosis = MGH_clusters$predicted_cluster_dysbiosis


all_clusters = rbind(all_clusters,MGH_clusters_new)

write.csv(all_clusters,"MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")


```



```{r FigS - HeatMaps for top taxa in each cluster}

################# Read clusters
clustring.method = "DMM_baseline"
compartment = "oral"
cluster.name = "LowDiversity"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]
  

relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))


################# Read clusters
clustring.method = "DMM_baseline"
compartment = "oral"
cluster.name = "Intermediate"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))

################# Read clusters
clustring.method = "DMM_baseline"
compartment = "oral"
cluster.name = "HighDiversity"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))
################# Read clusters
clustring.method = "DMM_baseline"
compartment = "lung"
cluster.name = "LowDiversity"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))


################# Read clusters
clustring.method = "DMM_baseline"
compartment = "lung"
cluster.name = "Intermediate"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))

################# Read clusters
clustring.method = "DMM_baseline"
compartment = "lung"
cluster.name = "HighDiversity"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))
################# Read clusters
################# Read clusters
clustring.method = "DMM_baseline"
compartment = "gut"
cluster.name = "LowDiversity"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))


################# Read clusters
clustring.method = "DMM_baseline"
compartment = "gut"
cluster.name = "Intermediate"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))

################# Read clusters
clustring.method = "DMM_baseline"
compartment = "gut"
cluster.name = "HighDiversity"

if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))
################# Read clusters
clustring.method = "Nanopore_DMM_cluster"
compartment = "lung"
cluster.name = "LowDiversity"


if(clustring.method == "DMM_baseline"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$baselineonlyanalysis_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","baselineonlyanalysis_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="lung_bacterial_fungal_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$lung_bacterial_fungal_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","lung_bacterial_fungal_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}else if(clustring.method=="Nanopore_DMM_cluster"){
  all_clusters = read.csv("MultiCompMbio/results/MultiCompMbio_all_clusters_allCohorts.csv")
  row.names(all_clusters) = all_clusters$sample_id
  relabundance = as.data.frame(assays(se)$relabundance)
  
  samples_of_interest = all_clusters$sample_id[which(!is.na(all_clusters$Nanopore_DMM_cluster))]
  current_clusters = all_clusters[samples_of_interest, which(names(all_clusters) %in% c("sample_id","Nanopore_DMM_cluster"))]
  relabundance = relabundance[,samples_of_interest]
}

names(current_clusters) = c("sample_id","cluster")
samples_available = names(relabundance)
sample.meta.data = as.data.frame(colData(se)[samples_available,])


identical(names(relabundance[,1:dim(relabundance)[2]]),row.names(sample.meta.data))
identical(row.names(current_clusters),row.names(sample.meta.data))

df = data.frame(sample.meta.data$sample_id,sample.meta.data$compartment,
                sample.meta.data$studydayfollowup,current_clusters$cluster)

names(df) = c("sample_id","compartment","studydayfollowup","cluster")


relabundance_cluster_comp = relabundance[,df$sample_id[which(df$compartment == compartment & df$cluster== cluster.name)]]


relabundance_cluster_comp$mean = rowMeans(relabundance_cluster_comp)
ordered.relabundance_data = relabundance_cluster_comp[order(relabundance_cluster_comp$mean,decreasing = TRUE),]

top.taxa = as.data.frame(ordered.relabundance_data$mean[1:10])
top.taxa$names = row.names(ordered.relabundance_data)[1:10]
names(top.taxa) = c(paste0(clustring.method,'.',compartment,'.',cluster.name,'.mean'),"taxon")
write.csv(top.taxa,file = paste0('MultiCompMbio/results/ordered_taxa_',clustring.method,'.',compartment,'.',cluster.name,'.csv'))

### reading all saved files to create a heatmap for the average abundance of taxa in each cluster

x1 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.lung.LowDiversity.csv")
x1 = x1[,-1]


x2 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.lung.Intermediate.csv")
x2 = x2[,-1]

x = merge(x1,x2, by = "taxon", all = TRUE )

x3 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.lung.HighDiversity.csv")
x3 = x3[,-1]

x = merge(x,x3, by = "taxon", all = TRUE )


names(x) = c("taxa", "DMM.lung.LowDiversity","DMM.lung.Intermediate","DMM.lung.HighDiversity")

x_melt <- reshape2::melt(x)
x_melt$value[which(is.na(x_melt$value))] = 0


ggplot(x_melt, aes(x = variable, y = taxa, fill = value)) +
  geom_tile() +
  labs(title = "Average relabundance value per cluster",
       x = "Cluster",
       y = "Top 10 Taxa") + theme(text = element_text(size=16),
                                  axis.text.x = element_text(angle=45, hjust=1, size = 10),
                                  legend.key.size = unit(1, "cm"))+
  scale_fill_gradient(low="white", high="blue")

ggsave(paste0('MultiCompMbio/results/averagerelabundancepercluster_lung.jpg'), width = 10, height = 10)

x1 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.oral.LowDiversity.csv")
x1 = x1[,-1]

x2 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.oral.Intermediate.csv")
x2 = x2[,-1]

x = merge(x1,x2, by = "taxon", all = TRUE )

x3 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.oral.HighDiversity.csv")
x3 = x3[,-1]

x = merge(x,x3, by = "taxon", all = TRUE )


names(x) = c("taxa", "DMM.oral.LowDiversity","DMM.oral.Intermediate","DMM.oral.HighDiversity")

x_melt <- reshape2::melt(x)
x_melt$value[which(is.na(x_melt$value))] = 0


ggplot(x_melt, aes(x = variable, y = taxa, fill = value)) +
  geom_tile() +
  labs(title = "Average relabundance value per cluster",
       x = "Cluster",
       y = "Top 10 Taxa") + theme(text = element_text(size=16),
                                  axis.text.x = element_text(angle=45, hjust=1, size = 10),
                                  legend.key.size = unit(1, "cm"))+
  scale_fill_gradient(low="white", high="blue")

ggsave(paste0('MultiCompMbio/results/averagerelabundancepercluster_oral.jpg'), width = 10, height = 10)


x1 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.gut.LowDiversity.csv")
x1 = x1[,-1]

x2 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.gut.Intermediate.csv")
x2 = x2[,-1]

x = merge(x1,x2, by = "taxon", all = TRUE )

x3 = read.csv("MultiCompMbio/results/ordered_taxa_DMM_baseline.gut.HighDiversity.csv")
x3 = x3[,-1]

x = merge(x,x3, by = "taxon", all = TRUE )

names(x) = c("taxa", "DMM.gut.LowDiversity","DMM.gut.Intermediate","DMM.gut.HighDiversity")

x_melt <- reshape2::melt(x)
x_melt$value[which(is.na(x_melt$value))] = 0


ggplot(x_melt, aes(x = variable, y = taxa, fill = value)) +
  geom_tile() +
  labs(title = "Average relabundance value per cluster",
       x = "Cluster",
       y = "Top 10 Taxa") + theme(text = element_text(size=16),
                                  axis.text.x = element_text(angle=45, hjust=1, size = 10),
                                  legend.key.size = unit(1, "cm"))+
  scale_fill_gradient(low="white", high="blue")

ggsave(paste0('MultiCompMbio/results/averagerelabundancepercluster_gut.jpg'), width = 10, height = 10)

```





```{r - descriptive demographics UPMC}
#### Descriptive analyses for noncovid and covid subjects from the UPMC/Pitt cohorts. 
alirsubjectid_mbio <- filter(alirmbio, !duplicated(SubjectID))

alirsubjectid_mbio <- alirsubjectid_mbio %>% mutate(studyday1_OnSteroids  = case_when(studyday1_mgpred>0 ~ "yes",   studyday1_mgpred==0  ~ "no" ))

vars_continuous <- c("Age","BMI", "WBC",  "HGB", 
                     "Platelets", "Creatinine", "BUN", "CO2", "Temperature", "il6", "il8", "il10", "tnfr1_final", "st2", "fractalkine",  "rage",  "ang2", "procalcitonin", "pentraxin3", 
    "il6_eta_pradj", "il8_eta_pradj",  "il10_eta_pradj", "tnfr1_eta_pradj", "st2_eta_pradj",  "fractalkine_eta_pradj", "rage_eta_pradj",  "ang2_eta_pradj", "procalcitonin_eta_pradj","pentraxin3_eta_pradj" , "totProt_eta_pradj", "IgM_eta_pradj", "NeutElas_eta_pradj", "HR" , "RR", "SBP", "SaturationSpO2", "PHa", "PaCO2", "TidalVolume", "TVmlperkg", "TotalRR", "MinuteVolumeTotal", "PEEP" , "PeakInspiratory", "PlateauPressure", "PlateauPr_imput", "VentRatio",  "WorstPaO2FiO2Ratio",   "SOFAScRSD" , "VFD",  "lips_score", "total_ABx_score_day1_Convex","studyday1_mgpred") 

vars_categorical <- c( "Gender", "Race",  "HistDiabetes", 
          "HistCOPD", "HistCCF",   "HistCRF",   "IsActiveNeoplasm",   "HistImmunosuppression" , "HistCLD", "HistPulmonaryFibrosis", "group",  "nonpulmsepsis", "Pneumonia", "Aspiration",  "HistAlcohol", "steroids", "antiIL6", "EndOfStayType",    "DischargeTo",    "Mortality30Day",  "Mortality60Day",  "Mortality90Day" , "day1OnABX" , "day3OnABX"  , "day7OnABX",  "VasopressorFirstWeek",   "phenotype", "CalfeePhenotype", "SinhaPhenotype", "VentMode", "studyday1_OnSteroids") 

############## 

cat_table_alirmbiome<- print(CreateCatTable(vars = vars_categorical, data=alirsubjectid_mbio,  includeNA = FALSE))
con_table_alirmbiome <- print(CreateContTable(vars = vars_continuous, data=alirsubjectid_mbio), nonnormal = vars_continuous, digits  = 1)

write.csv(cat_table_alirmbiome, "cat_table_alirmbiome.csv")
write.csv(con_table_alirmbiome, "con_table_alirmbiome.csv")

min(alirsubjectid_mbio$StudyDate, na.rm = TRUE)
max(alirsubjectid_mbio$StudyDate, na.rm = TRUE)

cat_table_covidmbiome<- print(CreateCatTable(vars = vars_categorical, data=covidsubjectid_mbio,  includeNA = FALSE))
con_table_covidmbiome <- print(CreateContTable(vars = vars_continuous, data=covidsubjectid_mbio), nonnormal = vars_continuous, digits  = 1)

write.csv(cat_table_covidmbiome, "cat_table_covidmbiome.csv")
write.csv(con_table_covidmbiome, "con_table_covidmbiome.csv")


### break down by 60 day survivors

cat_table_alir_Mortality60Day<- print(CreateCatTable(vars = vars_categorical, strata = "Mortality60Day" , data=alirsubjectid_mbio, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_alir_Mortality60Day <- print(CreateContTable(vars = vars_continuous, strata = "Mortality60Day" , data=alirsubjectid_mbio), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_alir_Mortality60Day, "cat_table_alir_Mortality60Day.csv")
write.csv(con_table_alir_Mortality60Day, "con_table_alir_Mortality60Day.csv")



### break down by 60 day survivors

cat_table_covid_Mortality60Day<- print(CreateCatTable(vars = vars_categorical, strata = "Mortality60Day" , data=covidsubjectid_mbio, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_covid_Mortality60Day <- print(CreateContTable(vars = vars_continuous, strata = "Mortality60Day" , data=covidsubjectid_mbio), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_covid_Mortality60Day, "cat_table_covid_Mortality60Day.csv")
write.csv(con_table_covid_Mortality60Day, "con_table_covid_Mortality60Day.csv")


```

```{r Comparisons between clusters}

#### import the cluster file with all available data - MultiCompMbio_all_clusters_allCohorts_3.13.24

MultiCompMbio_all_clusters_allCohorts_2 <- read_csv("data/MultiCompMbio_all_clusters_allCohorts.csv")

names(MultiCompMbio_all_clusters_allCohorts_2)
table(is.na(MultiCompMbio_all_clusters_allCohorts_2$ShannonIndex)) # no missing

alirmbio <- merge(MetaData,MultiCompMbio_all_clusters_allCohorts_2, by = "sample_id", all.y =  TRUE) 

alirmbio <- filter(alirmbio, StudyName=="alir" | StudyName=="alirseeds")

table(is.na(alir_microbiome_metadata_all$ShannonIndex), alir_microbiome_metadata_all$StudyName) 
table(is.na(alir_microbiome_metadata_all$ShannonIndex), alir_microbiome_metadata_all$rectalswab_soiled) 


table(alirmbio$rectalswab_soiled)

alirmbio <- alirmbio %>% filter(rectalswab_soiled =="yes" | is.na(rectalswab_soiled)) # 1771
alirmbio$SampleType <- alirmbio$SampleType.x
alirmbio$shannon <- alirmbio$ShannonIndex

alirmbio %>% ggplot(aes(x=studydayfollowup, y = shannon, color = SampleType)) + geom_violin() + geom_beeswarm() + facet_wrap(~SampleType)
alirmbio$studydayfollowup <- ordered(alirmbio$studydayfollowup, levels = c("baseline", "middle", "late"))

# alirmbio <- alirmbio %>% mutate(gut = case_when(SampleType=="stool" ~ "yes", rectalswab_soiled=="yes"   ~ "yes"))
table(alirmbio$SampleType_3comp, alirmbio$SampleType)
table(is.na(alirmbio$SampleType_3comp))
alirmbio$SampleTypeFactor <- alirmbio$SampleType_3comp

n_distinct(alirmbio$SubjectID.x)

table(alirmbio$SampleTypeFactor, alirmbio$studydayfollowup)
table(alirmbio$SampleType, alirmbio$studydayfollowup)

##### create publishable figures for diversity and abundances by clusters. 
# https://nanx.me/ggsci/reference/pal_npg.html
#### low diversity #7E6148FF
#### INTERM #F39B7FFF
#### HIGH #4DBBD5FF


color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
# fix the clusters name
alirmbio <- alirmbio %>% mutate(BaslClusters = case_when(baselineonlyanalysis_cluster=="LowDiversity" ~ "Low_Diversity", 
                                                         baselineonlyanalysis_cluster=="Intermediate" ~ "Interm_Diversity",
                                                         baselineonlyanalysis_cluster=="HighDiversity" ~ "High_Diversity"))


my_comparisons <- list(c("Low_Diversity", "Interm_Diversity"), c("Interm_Diversity", "High_Diversity"))

table(alirmbio$BaslClusters, alirmbio$SampleTypeFactor)

### clusters diversity
oralclustershannon <- alirmbio %>% filter(SampleTypeFactor=="oral") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = shannon, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral") + theme(axis.text.x = element_text(size = 8)) 

lungclustershannon <-  alirmbio %>% filter(SampleTypeFactor=="lung") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = shannon, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))

gutclustershannon <-  alirmbio %>% filter(SampleTypeFactor=="gut") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = shannon, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut") + theme(axis.text.x = element_text(size = 8))

shannonbyclusters <- ggarrange(oralclustershannon, lungclustershannon, gutclustershannon, ncol = 3)
ggsave("shannonbyclusters.jpeg", width = 10, height = 6)

table(alirmbio$SampleType_3comp, alirmbio$BaslClusters)

##### sankey plot for different clusters by compartment. 

library(ggsankey)

#### there will be 3 "timepoints"
#### create a dataframe with subject id and the the three clusters. 
alirmbio$SubjectID <- alirmbio$SubjectID.x
alimbioclusters <- filter(alirmbio, studydayfollowup=="baseline")
alimbioclusters <- filter(alimbioclusters, !is.na(BaslClusters))
alimbioclusters$SubjectIDSampletype <- paste0(alimbioclusters$SubjectID, "_", alimbioclusters$SampleTypeFactor)

##### create new variables for clusters. 
alimbioclusters <- alimbioclusters %>% mutate(LungCluster  = case_when(SampleTypeFactor=="lung" & BaslClusters=="Low_Diversity" ~ "Low_Diversity",
                                                                       SampleTypeFactor=="lung" & BaslClusters=="Interm_Diversity" ~"Interm_Diversity",
                                                                       SampleTypeFactor=="lung" & BaslClusters=="High_Diversity" ~"High_Diversity"))

alimbioclusters <- alimbioclusters %>% mutate(OralCluster  = case_when(SampleTypeFactor=="oral" & BaslClusters=="Low_Diversity" ~"Low_Diversity",
                                                                       SampleTypeFactor=="oral" & BaslClusters=="Interm_Diversity" ~"Interm_Diversity",
                                                                       SampleTypeFactor=="oral" & BaslClusters=="High_Diversity" ~"High_Diversity"))

alimbioclusters <- alimbioclusters %>% mutate(GutCluster  = case_when(SampleTypeFactor=="gut" & BaslClusters=="Low_Diversity" ~ "Low_Diversity",
                                                                       SampleTypeFactor=="gut" & BaslClusters=="Interm_Diversity" ~ "Interm_Diversity",
                                                                       SampleTypeFactor=="gut" & BaslClusters=="High_Diversity" ~ "High_Diversity"))


table(alimbioclusters$LungCluster)

table(alimbioclusters$LungCluster, alimbioclusters$SampleTypeFactor)

alimbioclusters <- subset(alimbioclusters, select =c(SubjectIDSampletype, SubjectID, OralCluster, LungCluster, GutCluster))
alimbioclusters <- filter(alimbioclusters, !duplicated(SubjectIDSampletype))
alimbioclusters <-subset(alimbioclusters, select = c(SubjectID, OralCluster, LungCluster, GutCluster))
### to remove the missingness, I will break this into 3 dataframes
alimbioclustersOral <- subset(alimbioclusters, select = c(SubjectID, OralCluster))
alimbioclustersOral <- drop_na(alimbioclustersOral)
alimbioclustersLung <- subset(alimbioclusters, select = c(SubjectID, LungCluster))
alimbioclustersLung <- drop_na(alimbioclustersLung)
alimbioclustersGut <- subset(alimbioclusters, select = c(SubjectID, GutCluster))
alimbioclustersGut <- drop_na(alimbioclustersGut)

### remerge the dataframe to remove the issue with missingness. ### lung has the most observations, so merge by lung. 
alimbioclusters <- merge(alimbioclustersOral,alimbioclustersLung, by= "SubjectID", all.y = TRUE )
alimbioclusters <- merge(alimbioclusters,alimbioclustersGut, by= "SubjectID", all.x = TRUE )


#### chord diagram: 
# install.packages("circlize")
library(circlize)
subset_data <- subset(alimbioclusters, select = c(SubjectID, OralCluster, LungCluster, GutCluster))
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)


###sankey for 3 clusters
library(ggsankey)
alimbioclusters_basl_sankey <- alimbioclusters %>% arrange(factor(LungCluster))

alimbioclusters_basl_sankey_long <- alimbioclusters_basl_sankey %>% make_long(OralCluster, LungCluster, GutCluster)

#### order the nodes
# https://datacornering.com/dplyr-error-n-must-only-be-used-inside-dplyr-verbs/
# install.packages("dplyr")
library(plyr)
library(dplyr)

  detach("package:plyr", unload = TRUE)
#library(tidyr)
pl <- ggplot(alimbioclusters_basl_sankey_long, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node), n = dplyr::n
             )
pl <- pl +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
# pl <- pl +geom_sankey_label(size = 3, color = "black", fill= "white", hjust = -0.5)
pl <- pl +  theme_bw()
pl <- pl + theme(legend.position = "none")
pl <- pl +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank())
pl <- pl + scale_fill_manual(values = color_scale)
pl <- pl + labs(fill = 'Nodes')
alimbioclusters_basl_sankey <- pl
ggsave("alimbioclusters_basl_sankey.jpeg", width = 12, height = 6)


### comparisons between clusters. 
table(alimbioclusters$OralCluster,  alimbioclusters$LungCluster)
chisq.test(alimbioclusters$OralCluster,alimbioclusters$LungCluster)

#### mcnemar's test for paired test. 
cont_table <- table(alimbioclusters$OralCluster, alimbioclusters$LungCluster)
# Perform McNemar's test
result <- mcnemar.test(cont_table)
# Print the test statistics and p-value
print(result)

cont_table <- table(alimbioclusters$OralCluster, alimbioclusters$LungCluster)

# Calculate the Jaccard similarity coefficient
jaccard_sim <- sum(diag(cont_table)) / sum(cont_table)

# Print the Jaccard similarity coefficient
print(jaccard_sim) ###### 0.608

#### probability of lowDiversityCluster.

alimbioclusters <- alimbioclusters %>% mutate(OralLowDiv = case_when(OralCluster=="Low_Diversity" ~ "yes",
                                                                     OralCluster!="Low_Diversity" ~ "no"))

alimbioclusters <- alimbioclusters %>% mutate(LungLowDiv = case_when(LungCluster=="Low_Diversity" ~ "yes",
                                                                     LungCluster!="Low_Diversity" ~ "no"))

alimbioclusters <- alimbioclusters %>% mutate(GutLowDiv = case_when(GutCluster=="Low_Diversity" ~ "yes",
                                                                     GutCluster!="Low_Diversity" ~ "no"))

oddsratio.fisher(alimbioclusters$OralLowDiv, alimbioclusters$LungLowDiv) # 7.674399 4.229267 14.24917 1.091557e-18
oddsratio.fisher(alimbioclusters$LungLowDiv, alimbioclusters$GutLowDiv) #  2.110021 0.978023 4.582402 0.0446211
oddsratio.fisher(alimbioclusters$OralLowDiv, alimbioclusters$GutLowDiv) #  ns

### comparisons between clusters. 
table(alimbioclusters$LungCluster,  alimbioclusters$GutCluster)
chisq.test(alimbioclusters$LungCluster,alimbioclusters$GutCluster) # 0.p-value = 0.05

#### mcnemar's test for paired test. 
cont_table <- table(alimbioclusters$LungCluster, alimbioclusters$GutCluster)
# Perform McNemar's test
result <- mcnemar.test(cont_table)
# Print the test statistics and p-value
print(result)


cont_table <- table(alimbioclusters$LungCluster, alimbioclusters$GutCluster)
# Calculate the Jaccard similarity coefficient
jaccard_sim <- sum(diag(cont_table)) / sum(cont_table)
# Print the Jaccard similarity coefficient
print(jaccard_sim) # 0.425


#### longitudinal clusters showed relative stability in classifications. 
##### sankey plots for oral,lung and gut separately. 

##### create new variables for clusters. 
alirmbio <- alirmbio %>% mutate(LongitudCluster  = case_when(alltimepoints_cluster=="LowDiversity" ~ "Low_Diversity",
                                                                       alltimepoints_cluster=="Intermediate" ~"Interm_Diversity",
                                                                       alltimepoints_cluster=="HighDiversity" ~"High_Diversity"))

##### define direct vs. indirect lung injury

alirmbio <- alirmbio %>% mutate(lunginjury = case_when(Pneumonia==1 ~ "direct", 
                                                       Aspiration==1 ~ "direct", 
                                                       Pneumonia==0 & nonpulmsepsis==1 ~"indirect", 
                                                      Pneumonia==0 & Pancreatitis==1 ~"indirect" ))

table(alirmbio$lunginjury)

###### sankey plot for oral overtime
#### oral 
oralmbio <- filter(alirmbio, SampleTypeFactor=="oral")

table(oralmbio$LongitudCluster)

OralSankey <- subset(oralmbio, select = c(SubjectID, LongitudCluster, studydayfollowup))

OralSankey_wide <- spread(OralSankey, studydayfollowup, LongitudCluster) 
# filter absent observations
OralSankey_wide <- filter( OralSankey_wide,!is.na(baseline) )
OralSankey_wide <- filter( OralSankey_wide,!is.na(middle) )

table(OralSankey_wide$baseline)
table(OralSankey_wide$middle)
table(OralSankey_wide$late)

OralSankey_long <- OralSankey_wide %>% make_long( baseline, middle, late)
chisq.test(OralSankey_wide$baseline, OralSankey_wide$middle)
table(OralSankey_wide$baseline, OralSankey_wide$middle)
#### order the nodes
# Clinical_Sankey_long$node <- factor(Clinical_Sankey_long$node,levels = c("LFO","NIV_HFO","IMV", "ECMO", "NoEscalation", "0", "1"))
# Clinical_Sankey_long$next_node <- factor(Clinical_Sankey_long$next_node,levels = c("LFO","NIV_HFO","IMV", "ECMO", "NoEscalation", "0", "1"))

# https://datacornering.com/dplyr-error-n-must-only-be-used-inside-dplyr-verbs/
library(dplyr)
pl <- ggplot(OralSankey_long, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             )
pl <- pl +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl <- pl +  theme_bw()
pl <- pl + theme(legend.position = "none")
pl <- pl +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank())
pl <- pl + scale_fill_manual(values = color_scale)
pl <- pl + labs(fill = 'Nodes')

ggsave("oralclusters_longit_sankey.jpeg", width = 12, height = 6)

####### lung longitudinal

lungmbio <- filter(alirmbio, SampleTypeFactor=="lung")

table(lungmbio$LongitudCluster)

lungSankey <- subset(lungmbio, select = c(SubjectID, SubjectIDDay, LongitudCluster, studydayfollowup))

lungSankey <- filter(lungSankey, !duplicated(SubjectIDDay))
lungSankey$SubjectIDDay <- NULL
lungSankey_wide <- spread(lungSankey, studydayfollowup, LongitudCluster) 
# filter absent observations
lungSankey_wide <- filter( lungSankey_wide,!is.na(baseline) )
lungSankey_wide <- filter( lungSankey_wide,!is.na(middle) )
lungSankey_long <- lungSankey_wide %>% make_long( baseline, middle, late)

table(lungSankey_wide$baseline)
table(lungSankey_wide$middle)
table(lungSankey_wide$late)

table(lungSankey_wide$baseline, lungSankey_wide$middle)
#### order the nodes
# Clinical_Sankey_long$node <- factor(Clinical_Sankey_long$node,levels = c("LFO","NIV_HFO","IMV", "ECMO", "NoEscalation", "0", "1"))
# Clinical_Sankey_long$next_node <- factor(Clinical_Sankey_long$next_node,levels = c("LFO","NIV_HFO","IMV", "ECMO", "NoEscalation", "0", "1"))

# https://datacornering.com/dplyr-error-n-must-only-be-used-inside-dplyr-verbs/
library(dplyr)
pl <- ggplot(lungSankey_long, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             )
pl <- pl +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl <- pl +  theme_bw()
pl <- pl + theme(legend.position = "none")
pl <- pl +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank())
pl <- pl + scale_fill_manual(values = color_scale)
pl <- pl + labs(fill = 'Nodes')

ggsave("lungclusters_longit_sankey.jpeg", width = 12, height = 6)


######### gut sankey
gutmbio <- filter(alirmbio, SampleTypeFactor=="gut")

table(gutmbio$LongitudCluster)

gutSankey <- subset(gutmbio, select = c(SubjectID, SubjectIDDay, LongitudCluster, studydayfollowup))

gutSankey <- filter(gutSankey, !duplicated(SubjectIDDay))
gutSankey$SubjectIDDay <- NULL
gutSankey_wide <- spread(gutSankey, studydayfollowup, LongitudCluster) 
# filter absent observations
gutSankey_wide <- filter( gutSankey_wide,!is.na(baseline) )
gutSankey_wide <- filter( gutSankey_wide,!is.na(middle) )
gutSankey_long <- gutSankey_wide %>% make_long( baseline, middle, late)


table(gutSankey_wide$baseline)
table(gutSankey_wide$middle)
table(gutSankey_wide$late)


table(gutSankey_wide$baseline, gutSankey_wide$middle)
#### order the nodes
# Clinical_Sankey_long$node <- factor(Clinical_Sankey_long$node,levels = c("LFO","NIV_HFO","IMV", "ECMO", "NoEscalation", "0", "1"))
# Clinical_Sankey_long$next_node <- factor(Clinical_Sankey_long$next_node,levels = c("LFO","NIV_HFO","IMV", "ECMO", "NoEscalation", "0", "1"))

# https://datacornering.com/dplyr-error-n-must-only-be-used-inside-dplyr-verbs/
# https://datacornering.com/dplyr-error-n-must-only-be-used-inside-dplyr-verbs/
# library(dplyr)
pl <- ggplot(gutSankey_long, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             )
pl <- pl +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl <- pl +  theme_bw()
pl <- pl + theme(legend.position = "none")
pl <- pl +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank())
pl <- pl + scale_fill_manual(values = color_scale)
pl <- pl + labs(fill = 'Nodes')

ggsave("gutclusters_longit_sankey.jpeg", width = 12, height = 6)


#### write the gutmbio dataset
write.csv(gutmbio, "gutmbio.csv")

####### look at abundance of specific categories of bacteria. 

oralclusterAnaerobeSum <- alirmbio %>% filter(SampleTypeFactor=="oral") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = AnaerobeSum, color = BaslClusters))  + geom_boxplot( outlier.shape=NA) + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral") + theme(axis.text.x = element_text(size = 8)) + ylab("Obligate Anaerobe abundance") 

lungclusterAnaerobeSum <-  alirmbio %>% filter(SampleTypeFactor=="lung") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = AnaerobeSum, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Obligate Anaerobe abundance")

gutclusterAnaerobeSum <-  alirmbio %>% filter(SampleTypeFactor=="gut") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = AnaerobeSum, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut") + theme(axis.text.x = element_text(size = 8))  + ylab("Obligate Anaerobe abundance")

AnaerobeSumbyclusters <- ggarrange(oralclusterAnaerobeSum, lungclusterAnaerobeSum, gutclusterAnaerobeSum, ncol = 3)
ggsave("AnaerobeSumbyclusters.jpeg", width = 10, height = 6)


###### pathogen 
oralclusterPathogenSum <- alirmbio %>% filter(SampleTypeFactor=="oral") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = PathogenSum, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_jitter(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")

lungclusterPathogenSum <-  alirmbio %>% filter(SampleTypeFactor=="lung") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = PathogenSum, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_jitter(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8)) + ylab("Pathogen abundance")

gutclusterPathogenSum <-  alirmbio %>% filter(SampleTypeFactor=="gut") %>%  filter(!is.na(BaslClusters))  %>% ggplot(aes(x=BaslClusters, y = PathogenSum, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_jitter(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut") + theme(axis.text.x = element_text(size = 8)) + ylab("Pathogen abundance")

PathogenSumbyclusters <- ggarrange(oralclusterPathogenSum, lungclusterPathogenSum, gutclusterPathogenSum, ncol = 3)
ggsave("PathogenSumbyclusters.jpeg", width = 10, height = 6)


FigureClusters <- ggarrange(shannonbyclusters, AnaerobeSumbyclusters, PathogenSumbyclusters, alimbioclusters_basl_sankey, nrow = 4)
ggsave("FigureClusters.jpeg", width = 14, height = 16)

FigureDMMClusters <- ggarrange(shannonbyclusters,  PathogenSumbyclusters, alimbioclusters_basl_sankey, nrow = 3)
ggsave("FigureDMMClusters.jpeg", width = 14, height = 12)


##### incporate qPCR. 
lungclusterqPCR <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = Nof16scopies, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("N of 16S copies by qPCR") + ggtitle("Lung")

oralclusterqPCR <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = Nof16scopies, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("N of 16S copies by qPCR")  + ggtitle("Oral")

gutclusterqPCR <-  gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = Nof16scopies, color = BaslClusters))  + geom_boxplot(outlier.shape=NA) + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("N of 16S copies by qPCR") + ggtitle("Gut")

clustersqPCR <- ggarrange(oralclusterqPCR,  lungclusterqPCR, gutclusterqPCR, ncol = 3)
ggsave("clustersqPCR.jpeg", width = 14, height = 8)




```

```{r gut origin taxa enrichment}
filtered_taxa_table <- read_csv("data/filtered_taxa_table.csv")

 names(filtered_taxa_table)

 alirtaxatable <- as.data.frame(t(filtered_taxa_table))
colnames(alirtaxatable) <- alirtaxatable[1, ]
# Remove the first row (which now contains the variable names)
alirtaxatable <- alirtaxatable[-1, ]
alirtaxatable$sample_id <- rownames(alirtaxatable)
alirtaxatable  <- filter(alirtaxatable, sample_id %in% alirmbio$sample_id)
# this alirtaxa table has already been filtered
filtered_taxa <- alirtaxatable

names(filtered_taxa)
# Now filtered_taxa contains your filtered taxatable
# Calculate the row sums of columns 3 to 154
# row_sums_abundance <- rowSums(filtered_taxa[, 3:154])

# Subtract the row sums from the "total" column
# reads_filtered <- filtered_taxa$total - row_sums_abundance
# hist(reads_filtered)

# Now 'reads_filtered' contains the number of reads filtered per sample
#### obtain the taxa that we need for the assessment of specific bug abundance. 

lungmbio_taxa <- merge(lungmbio, filtered_taxa, by = "sample_id", all.x = TRUE)

### define key gut origin taxa

lungmbio_taxa$Enterococcus <-as.numeric(lungmbio_taxa$Enterococcus )
lungmbio_taxa$Enterobacteriaceae  <-as.numeric(lungmbio_taxa$Enterobacteriaceae )
lungmbio_taxa$Escherichia_Shigella <- as.numeric(lungmbio_taxa$Escherichia_Shigella)
lungmbio_taxa$Klebsiella <- as.numeric(lungmbio_taxa$Klebsiella)

lungmbio_taxa$Bacteroides <-  as.numeric(lungmbio_taxa$Bacteroides)
lungmbio_taxa$Lachnoclostridium <- as.numeric(lungmbio_taxa$Lachnoclostridium)
lungmbio_taxa$Lachnospiraceae_ge <- as.numeric(lungmbio_taxa$Lachnospiraceae_ge)
lungmbio_taxa$Lachnospiraceae_uncl <- as.numeric(lungmbio_taxa$Lachnospiraceae_uncl)
lungmbio_taxa$Anaerococcus <- as.numeric(lungmbio_taxa$Anaerococcus)

#### obtain relative abundances for these taxa
lungmbio_taxa$total16sreads
lungmbio_taxa$Enterococcus_relab <- lungmbio_taxa$Enterococcus / lungmbio_taxa$total16sreads
lungmbio_taxa$Enterobacteriaceae_relab <- lungmbio_taxa$Enterobacteriaceae / lungmbio_taxa$total16sreads
lungmbio_taxa$Escherichia_Shigella_relab <- lungmbio_taxa$Escherichia_Shigella / lungmbio_taxa$total16sreads
lungmbio_taxa$Klebsiella_relab <- lungmbio_taxa$Klebsiella / lungmbio_taxa$total16sreads
lungmbio_taxa$Bacteroides_relab <- lungmbio_taxa$Bacteroides / lungmbio_taxa$total16sreads
lungmbio_taxa$Lachnoclostridium_relab <- lungmbio_taxa$Lachnoclostridium / lungmbio_taxa$total16sreads
lungmbio_taxa$Lachnospiraceae_ge_relab <- lungmbio_taxa$Lachnospiraceae_ge / lungmbio_taxa$total16sreads
lungmbio_taxa$Lachnospiraceae_uncl_relab <- lungmbio_taxa$Lachnospiraceae_uncl / lungmbio_taxa$total16sreads
lungmbio_taxa$Anaerococcus_relab <- lungmbio_taxa$Anaerococcus / lungmbio_taxa$total16sreads

lungmbio_taxa %>%  ggplot(aes(Enterococcus_relab)) + geom_histogram()
lungmbio_taxa %>%  ggplot(aes(Enterobacteriaceae_relab)) + geom_histogram()
lungmbio_taxa %>%  ggplot(aes(Escherichia_Shigella_relab)) + geom_histogram()
lungmbio_taxa %>%  ggplot(aes(Klebsiella_relab)) + geom_histogram()
lungmbio_taxa %>%  ggplot(aes(Bacteroides_relab)) + geom_histogram()


### define gut origin abundance

lungmbio_taxa$gutorigintaxa <- lungmbio_taxa$Enterococcus_relab + lungmbio_taxa$Enterobacteriaceae_relab + lungmbio_taxa$Escherichia_Shigella_relab + lungmbio_taxa$Klebsiella_relab + lungmbio_taxa$Bacteroides_relab + lungmbio_taxa$Lachnoclostridium_relab + lungmbio_taxa$Lachnospiraceae_ge_relab + lungmbio_taxa$Lachnospiraceae_uncl_relab + lungmbio_taxa$Anaerococcus_relab

lungmbio_taxa$OtherTaxa <- 1 - lungmbio_taxa$gutorigintaxa

lungmbio_taxa$GutOriginTaxa <- lungmbio_taxa$gutorigintaxa

### example for graph. 
exampleforgraph <- data.frame(
  sample_id = c("0102.1031.1.T", "0102.1031.3.T", "0102.1357.1.T", "0102.1686.7.T", "0102.1767.1.T", "0102.1767.3.T"),
  gutorigintaxa = c(0.0012939959, 0.0002000200, 0.0000000000, 0.1917274939, 0.0000000000, 0.0007116425),
  OtherTaxa = c(0.9987060, 0.9998000, 1.0000000, 0.8082725, 1.0000000, 0.9992884)
)
melted_data <- exampleforgraph %>%
  gather(key = "variable", value = "proportion", -sample_id)
# Print the melted data
print(melted_data)
ggplot(melted_data, aes(x = sample_id, y = proportion, fill = variable)) +
  geom_bar(position = "stack", stat = "identity") +
  scale_fill_manual(values = c("gutorigintaxa" = "blue", "OtherTaxa" = "green")) +
  labs(x = "Observation", y = "Proportion", fill = "Variable") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  guides(fill = guide_legend(title = NULL))


##### now take a dataframe with these 3 variables to create the barplot we need. 

lungmbio_taxa_gutGraph <- subset(lungmbio_taxa, select = c(sample_id, GutOriginTaxa,OtherTaxa ))
head(lungmbio_taxa_gutGraph)

melted_data <- lungmbio_taxa_gutGraph %>%
  gather(key = "variable", value = "proportion", -sample_id)

ordered_sample_id <- melted_data %>%
  filter(variable == "GutOriginTaxa") %>%
  arrange(desc(proportion)) %>%
  pull(sample_id)

### remove the x-axis sampleids. 
ggplot(melted_data, aes(x = factor(sample_id, levels = ordered_sample_id), y = proportion, fill = variable)) +
  geom_bar(position = "stack", stat = "identity") +
  scale_fill_manual(values = c("GutOriginTaxa" = "blue", "OtherTaxa" = "gray")) +
  labs(x = "Lung Samples", y = "Relative Abundance", fill = "Variable") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  guides(fill = guide_legend(title = NULL))
ggsave("GutOriginTaxainETAsamples.jpeg", width = 8, height = 3)

#### define variable of GutTaxaEnrichment as relabundance>0.3

lungmbio_taxa  <- lungmbio_taxa %>% mutate(GutTaxaEnrichment = case_when(GutOriginTaxa >0.3 ~ "Yes",GutOriginTaxa <0.3005  ~ "No" ))
table(lungmbio_taxa$GutTaxaEnrichment) # 42 enriched.
table(lungmbio_taxa$GutTaxaEnrichment, lungmbio_taxa$studydayfollowup)

42/(459+42) # 8.3% enriched. 
21/(21+345) # 5.7 baseline
16/99 # 16.2 middle
5/36 # 13.9 late

##### create barplots for the specific bugs for the gut enriched cases. 

lungmbio_taxa_GutEnriched <- filter(lungmbio_taxa,GutTaxaEnrichment=="Yes" )

enterococcusabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Enterococcus_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#0033FF") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Enterococcus Abundance")

Enterobacteriaceaeabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Enterobacteriaceae_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#3366FF") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Enterobacteriaceae Abundance")

Escherichia_Shigellaabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Escherichia_Shigella_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#6666FF") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Escherichia_Shigella Abundance")

 Klebsiellaabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Klebsiella_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#9999FF") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle(" Klebsiella Abundance")

Bacteroidesabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Bacteroides_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#003366") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Bacteroides Abundance")

Lachnoclostridium_relababund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Lachnoclostridium_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Lachnoclostridium_relab Abundance")


Lachnospiraceae_geabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Lachnospiraceae_ge_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Lachnospiraceae_ge Abundance")

Lachnospiraceae_unclabund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Lachnospiraceae_uncl_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Lachnospiraceae_uncl Abundance")

Anaerococcus_abund <- ggplot(lungmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Anaerococcus_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "Lung Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Anaerococcus Abundance")

gutenrichedabund <- ggarrange(enterococcusabund, Enterobacteriaceaeabund, Escherichia_Shigellaabund, Klebsiellaabund, Bacteroidesabund,Lachnoclostridium_relababund, Lachnospiraceae_geabund, Lachnospiraceae_unclabund, Anaerococcus_abund ,nrow = 5, ncol=2)
ggsave("gutenrichedabund.jpeg", width = 10, height = 16)

##### look for differences in the subjects with and without gut enrichment
lungmbio_taxa_basl <- filter(lungmbio_taxa, studydayfollowup=="baseline")

cat_table_alir_GutEnrichment<- print(CreateCatTable(vars = vars_categorical, strata = "GutTaxaEnrichment" , data=lungmbio_taxa_basl, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_alir_GutEnrichment <- print(CreateContTable(vars = vars_continuous, strata = "GutTaxaEnrichment" , data=lungmbio_taxa_basl), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_alir_GutEnrichment, "cat_table_alir_GutEnrichment.csv")
write.csv(con_table_alir_GutEnrichment, "con_table_alir_GutEnrichment.csv")

#### significant association with calfee phenotype

#### lung survival
#### create publishable figure
lungmbio_taxa_basl <- lungmbio_taxa_basl %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))

     lungmbio_taxa_basl$survival_time60_fromICUAdmission <- lungmbio_taxa_basl$survival_time90_fromICUAdmission 
      lungmbio_taxa_basl$survival_time60_fromICUAdmission[ lungmbio_taxa_basl$survival_time60_fromICUAdmission >60] <- 61       

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~GutTaxaEnrichment , data=lungmbio_taxa_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 
  
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

surv_lungGutEnrichment <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("NoGutOriginTaxaEnrichment",  "GutOriginTaxaEnrichment"), palette = c("gray", "blue"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Strata:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_lungGutEnrichment.jpeg", width = 580, height = 418)
print(surv_lungGutEnrichment, newpage = FALSE)

##### repeat this analysis in ARDS patients only. consistent trend
lungmbio_taxa_basl_ards <- filter(lungmbio_taxa_basl, group=="ards" )
survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~GutTaxaEnrichment , data=lungmbio_taxa_basl_ards)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 


lungmbio_taxa_basl_pneumonia <- filter(lungmbio_taxa_basl, Pneumonia=="1" )
survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~GutTaxaEnrichment , data=lungmbio_taxa_basl_pneumonia)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days")


#### comparison of gutorigintaxa between survivors and non-survivors. 
color_scale <- c("Non-Survivors" = "black", "Survivors" = "green4")

lungmbio_taxa_basl$survivors60day[lungmbio_taxa_basl$Mortality60Day==0] <- "Survivors"
lungmbio_taxa_basl$survivors60day[lungmbio_taxa_basl$Mortality60Day==1] <- "Non-Survivors"

lungmbio_taxa_basl$survivors60day <- ordered(lungmbio_taxa_basl$survivors60day, levels = c("Non-Survivors", "Survivors"))

my_comparisons <- list(c("Non-Survivors", "Survivors"))


GutOriginTaxabysurvivors <-  lungmbio_taxa_basl %>%  filter(!is.na(survivors60day))  %>% ggplot(aes(x=as.factor(survivors60day), y = GutOriginTaxa, color = as.factor(survivors60day)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") 
ggsave("GutOriginTaxabysurvivors.jpeg", width = 4, height = 5)


# no difference in bdg
 lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = bdg, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means()

 # no differenc in FABP2
  lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = FABP2, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means()
 
 # no differenc in spd
  lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = spd_bwh, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means()

    lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = spd_luminex, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means()
    
     # no differenc in LBP-CD14
  lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = LBP, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means()

    lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = cd14, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means()
  
    
        lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = total_mcfdna, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means() ### higher total mcfdna
        
        
         lungmbio_taxa_basl %>%  filter(!is.na(GutTaxaEnrichment))  %>% ggplot(aes(x=as.factor(GutTaxaEnrichment), y = tnfr1_final, color = as.factor(GutTaxaEnrichment)))  + geom_boxplot(outlier.shape = NA) + geom_jitter(size = 0.3, width = 0.1)  + scale_color_manual(values = color_scale) + scale_y_log10() + stat_compare_means() 
         
         
         oddsratio.fisher(lungmbio_taxa_basl$phenotype, lungmbio_taxa_basl$GutTaxaEnrichment) # ns

         
         
##### examine whether such enrichment by gut origin taxa is explained by oral colonization by these organisms. 

oralmbio_taxa <- merge(oralmbio, filtered_taxa, by = "sample_id", all.x = TRUE)

### define key gut origin taxa

oralmbio_taxa$Enterococcus <-as.numeric(oralmbio_taxa$Enterococcus )
oralmbio_taxa$Enterobacteriaceae  <-as.numeric(oralmbio_taxa$Enterobacteriaceae )
oralmbio_taxa$Escherichia_Shigella <- as.numeric(oralmbio_taxa$Escherichia_Shigella)
oralmbio_taxa$Klebsiella <- as.numeric(oralmbio_taxa$Klebsiella)

oralmbio_taxa$Bacteroides <-  as.numeric(oralmbio_taxa$Bacteroides)
oralmbio_taxa$Lachnoclostridium <- as.numeric(oralmbio_taxa$Lachnoclostridium)
oralmbio_taxa$Lachnospiraceae_ge <- as.numeric(oralmbio_taxa$Lachnospiraceae_ge)
oralmbio_taxa$Lachnospiraceae_uncl <- as.numeric(oralmbio_taxa$Lachnospiraceae_uncl)
oralmbio_taxa$Anaerococcus <- as.numeric(oralmbio_taxa$Anaerococcus)


#### obtain relative abundances for these taxa
oralmbio_taxa$total16sreads
oralmbio_taxa$Enterococcus_relab <- oralmbio_taxa$Enterococcus / oralmbio_taxa$total16sreads
oralmbio_taxa$Enterobacteriaceae_relab <- oralmbio_taxa$Enterobacteriaceae / oralmbio_taxa$total16sreads
oralmbio_taxa$Escherichia_Shigella_relab <- oralmbio_taxa$Escherichia_Shigella / oralmbio_taxa$total16sreads
oralmbio_taxa$Klebsiella_relab <- oralmbio_taxa$Klebsiella / oralmbio_taxa$total16sreads
oralmbio_taxa$Bacteroides_relab <- oralmbio_taxa$Bacteroides / oralmbio_taxa$total16sreads
oralmbio_taxa$Lachnoclostridium_relab <- oralmbio_taxa$Lachnoclostridium / oralmbio_taxa$total16sreads
oralmbio_taxa$Lachnospiraceae_ge_relab <- oralmbio_taxa$Lachnospiraceae_ge / oralmbio_taxa$total16sreads
oralmbio_taxa$Lachnospiraceae_uncl_relab <- oralmbio_taxa$Lachnospiraceae_uncl / oralmbio_taxa$total16sreads
oralmbio_taxa$Anaerococcus_relab <- oralmbio_taxa$Anaerococcus / oralmbio_taxa$total16sreads

oralmbio_taxa %>%  ggplot(aes(Enterococcus_relab)) + geom_histogram()
oralmbio_taxa %>%  ggplot(aes(Enterobacteriaceae_relab)) + geom_histogram()
oralmbio_taxa %>%  ggplot(aes(Escherichia_Shigella_relab)) + geom_histogram()
oralmbio_taxa %>%  ggplot(aes(Klebsiella_relab)) + geom_histogram()
oralmbio_taxa %>%  ggplot(aes(Bacteroides_relab)) + geom_histogram()


### define gut origin abundance

oralmbio_taxa$gutorigintaxa <- oralmbio_taxa$Enterococcus_relab + oralmbio_taxa$Enterobacteriaceae_relab + oralmbio_taxa$Escherichia_Shigella_relab + oralmbio_taxa$Klebsiella_relab + oralmbio_taxa$Bacteroides_relab + oralmbio_taxa$Lachnoclostridium_relab + oralmbio_taxa$Lachnospiraceae_ge_relab + oralmbio_taxa$Lachnospiraceae_uncl_relab + oralmbio_taxa$Anaerococcus_relab

oralmbio_taxa$OtherTaxa <- 1 - oralmbio_taxa$gutorigintaxa

oralmbio_taxa$GutOriginTaxa <- oralmbio_taxa$gutorigintaxa

##### now take a dataframe with these 3 variables to create the barplot we need. 

oralmbio_taxa_gutGraph <- subset(oralmbio_taxa, select = c(sample_id, GutOriginTaxa,OtherTaxa ))
head(oralmbio_taxa_gutGraph)

melted_data <- oralmbio_taxa_gutGraph %>%
  gather(key = "variable", value = "proportion", -sample_id)

ordered_sample_id <- melted_data %>%
  filter(variable == "GutOriginTaxa") %>%
  arrange(desc(proportion)) %>%
  pull(sample_id)

### remove the x-axis sampleids. 
ggplot(melted_data, aes(x = factor(sample_id, levels = ordered_sample_id), y = proportion, fill = variable)) +
  geom_bar(position = "stack", stat = "identity") +
  scale_fill_manual(values = c("GutOriginTaxa" = "blue", "OtherTaxa" = "gray")) +
  labs(x = "Oral Samples", y = "Relative Abundance", fill = "Variable") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  guides(fill = guide_legend(title = NULL))
ggsave("GutOriginTaxainOralsamples.jpeg", width = 8, height = 3)


n_distinct(melted_data$sample_id) # 593 oral samples

#### define variable of GutTaxaEnrichment as relabundance>0.3

oralmbio_taxa  <- oralmbio_taxa %>% mutate(GutTaxaEnrichment = case_when(GutOriginTaxa >0.3 ~ "Yes",GutOriginTaxa <0.3005  ~ "No" ))
table(oralmbio_taxa$GutTaxaEnrichment) # 47 enriched.
table(oralmbio_taxa$GutTaxaEnrichment, oralmbio_taxa$studydayfollowup)

26/(543+26) # 0.045% enriched. 
15/(15+358) # 0.041 baseline
9/142 # 0.06.3 middle
2/54 # 0.03.7 late


##### create barplots for the specific bugs for the gut enriched cases. 

oralmbio_taxa_GutEnriched <- filter(oralmbio_taxa,GutTaxaEnrichment=="Yes" )

enterococcusabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Enterococcus_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#0033FF") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Enterococcus Abundance")

Enterobacteriaceaeabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Enterobacteriaceae_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#3366FF") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Enterobacteriaceae Abundance")

Escherichia_Shigellaabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Escherichia_Shigella_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#6666FF") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Escherichia_Shigella Abundance")

 Klebsiellaabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Klebsiella_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#9999FF") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle(" Klebsiella Abundance")

Bacteroidesabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Bacteroides_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "#003366") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Bacteroides Abundance")

Lachnoclostridium_relababund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Lachnoclostridium_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Lachnoclostridium_relab Abundance")


Lachnospiraceae_geabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Lachnospiraceae_ge_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Lachnospiraceae_ge Abundance")

Lachnospiraceae_unclabund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Lachnospiraceae_uncl_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Lachnospiraceae_uncl Abundance")

Anaerococcus_abund <- ggplot(oralmbio_taxa_GutEnriched, aes(x = factor(sample_id), y = Anaerococcus_relab)) +
  geom_bar(position = "stack", stat = "identity", fill = "gray") +
  labs(x = "oral Samples", y = "Relative Abundance") +
  theme_minimal() +
  theme(axis.text.x = element_blank()) +
  ggtitle("Anaerococcus Abundance")

gutenrichedabundOral <- ggarrange(enterococcusabund, Enterobacteriaceaeabund, Escherichia_Shigellaabund, Klebsiellaabund, Bacteroidesabund,Lachnoclostridium_relababund, Lachnospiraceae_geabund, Lachnospiraceae_unclabund, Anaerococcus_abund ,nrow = 5, ncol=2)
ggsave("gutenrichedabundOral.jpeg", width = 10, height = 16)

#### survival for gut enrichment in oral samples
oralmbio_taxa_basl <- filter(oralmbio_taxa, studydayfollowup=="baseline")
oralmbio_taxa_basl <- oralmbio_taxa_basl %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))

     oralmbio_taxa_basl$survival_time60_fromICUAdmission <- oralmbio_taxa_basl$survival_time90_fromICUAdmission 
      oralmbio_taxa_basl$survival_time60_fromICUAdmission[ oralmbio_taxa_basl$survival_time60_fromICUAdmission >60] <- 61       

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~GutTaxaEnrichment , data=oralmbio_taxa_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 
  
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

surv_oralGutEnrichmentOral <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("NoGutOriginTaxaEnrichment",  "GutOriginTaxaEnrichment"), palette = c("gray", "blue"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Strata:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_oralGutEnrichmentOral.jpeg", width = 580, height = 418)
print(surv_oralGutEnrichmentOral, newpage = FALSE)


```



```{r abundance graphs for individual categories}

### oral

my_comparisons <- list(c("baseline", "middle"), c("middle", "late"))

####### pathogens abundances
alirmbio$OtherSum
oralcommensalLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = OralSum))  + geom_boxplot(color = "#00CCFF", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#00CCFF")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral Commensals") + theme(axis.text.x = element_text(size = 8))  + ylab("Oral Commensal abundance")
ggsave("oralcommensalLongit.jpeg", width = 8, height = 4)

oralPathogenLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot(color = "#FF3399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#FF3399")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Respiratory Pathogens") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")
ggsave("oralPathogenLongit.jpeg", width = 8, height = 4)

oralOtherLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = OtherSum))  + geom_boxplot(color = "gray", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "gray")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Other Taxa") + theme(axis.text.x = element_text(size = 8))  + ylab("Other Taxa abundance")
ggsave("oralOtherLongit.jpeg", width = 8, height = 4)

oralPathogens <- ggarrange(oralcommensalLongit, oralPathogenLongit, oralOtherLongit, ncol = 3)
ggsave("oralPathogens.jpeg", width = 10, height = 5)

lungcommensalLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = OralSum))  + geom_boxplot(color = "#00CCFF", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#00CCFF")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral Commensals") + theme(axis.text.x = element_text(size = 8))  + ylab("Oral Commensal abundance")
ggsave("lungcommensalLongit.jpeg", width = 8, height = 4)

lungPathogenLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot(color = "#FF3399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#FF3399")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Respiratory Pathogens") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")
ggsave("lungPathogenLongit.jpeg", width = 8, height = 4)

lungOtherLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = OtherSum))  + geom_boxplot(color = "gray", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "gray")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Other Taxa") + theme(axis.text.x = element_text(size = 8))  + ylab("Other Taxa abundance")
ggsave("lungOtherLongit.jpeg", width = 8, height = 4)

lungPathogens <- ggarrange(lungcommensalLongit, lungPathogenLongit, lungOtherLongit, ncol = 3)
ggsave("lungPathogens.jpeg", width = 10, height = 5)


gutcommensalLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = OralSum))  + geom_boxplot(color = "#00CCFF", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#00CCFF")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral Commensals") + theme(axis.text.x = element_text(size = 8))  + ylab("Oral Commensal abundance")
ggsave("gutcommensalLongit.jpeg", width = 8, height = 4)

gutPathogenLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot(color = "#FF3399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#FF3399")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Respiratory Pathogens") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")
ggsave("gutPathogenLongit.jpeg", width = 8, height = 4)

gutOtherLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = OtherSum))  + geom_boxplot(color = "gray", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "gray")   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Other Taxa") + theme(axis.text.x = element_text(size = 8))  + ylab(" Other Taxa abundance")
ggsave("gutOtherLongit.jpeg", width = 8, height = 4)

gutPathogens <- ggarrange(gutcommensalLongit, gutPathogenLongit, gutOtherLongit, ncol = 3)
ggsave("gutPathogens.jpeg", width = 10, height = 5)


####### repeat these graphs but perform CLR transformations 
###### take the oxygen variables and then the pathogenicity variables
oralmbio_oxygentaxa <- subset(oralmbio, select = c(AnaerobeSum, AerobeSum , FacultAnaerobeSum))
oralmbio_oxygentaxa$OtherSum <- 1 - (oralmbio_oxygentaxa$AnaerobeSum + oralmbio_oxygentaxa$AerobeSum + oralmbio_oxygentaxa$FacultAnaerobeSum)

oralmbio_oxygentaxa_clr <- as.data.frame(clr(oralmbio_oxygentaxa))
oldnames = c( "AnaerobeSum", "AerobeSum", "FacultAnaerobeSum","OtherSum" )
newnames = c("AnaerobeCLR", "AerobeCLR", "FacultAnaerobeCLR","OtherCLR" )
oralmbio_oxygentaxa_clr <- oralmbio_oxygentaxa_clr %>% rename_at(vars(oldnames), ~ newnames)

oralmbio_clr <- cbind(oralmbio, oralmbio_oxygentaxa_clr)


### manual p-values
# anaerobes
pvalues_adjustedforpairwise <- compare_means(AnaerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = oralmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

oralanaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = AnaerobeSum))  + geom_boxplot(color = "#660066", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#660066")  + 
   stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
   theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Anaerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe abundance")
ggsave("oralanaerobeLongit.jpeg", width = 8, height = 4)
# aerobes
pvalues_adjustedforpairwise <- compare_means(AerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = oralmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

oralaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = AerobeSum))  + geom_boxplot(color = "#003399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#003399")   + 
 stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Aerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Aerobe abundance")
ggsave("oralaerobeLongit.jpeg", width = 8, height = 4)
# facultative
pvalues_adjustedforpairwise <- compare_means(FacultAnaerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = oralmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

oralfacultaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = FacultAnaerobeSum))  + geom_boxplot(color = "#339999", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#339999")   + 
  stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Facultative Anaerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Facultative Anaerobe abundance")
ggsave("oralfacultaerobeLongit.jpeg", width = 8, height = 4)

oraloxygen <- ggarrange(oralanaerobeLongit, oralaerobeLongit, oralfacultaerobeLongit, ncol = 3)
ggsave("oraloxygen.jpeg", width = 10, height = 5)


####### CLR comparisons and adjustments for lung oxygen

lungmbio_oxygentaxa <- subset(lungmbio, select = c(AnaerobeSum, AerobeSum , FacultAnaerobeSum))
lungmbio_oxygentaxa$OtherSum <- 1 - (lungmbio_oxygentaxa$AnaerobeSum + lungmbio_oxygentaxa$AerobeSum + lungmbio_oxygentaxa$FacultAnaerobeSum)

lungmbio_oxygentaxa_clr <- as.data.frame(clr(lungmbio_oxygentaxa))
oldnames = c( "AnaerobeSum", "AerobeSum", "FacultAnaerobeSum","OtherSum" )
newnames = c("AnaerobeCLR", "AerobeCLR", "FacultAnaerobeCLR","OtherCLR" )
lungmbio_oxygentaxa_clr <- lungmbio_oxygentaxa_clr %>% rename_at(vars(oldnames), ~ newnames)

lungmbio_clr <- cbind(lungmbio, lungmbio_oxygentaxa_clr)

### manual p-values
# anaerobes
pvalues_adjustedforpairwise <- compare_means(AnaerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = lungmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

lunganaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = AnaerobeSum))  + geom_boxplot(color = "#660066", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#660066")  + 
   stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
   theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Anaerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe abundance")
ggsave("lunganaerobeLongit.jpeg", width = 8, height = 4)
# aerobes
pvalues_adjustedforpairwise <- compare_means(AerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = lungmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

lungaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = AerobeSum))  + geom_boxplot(color = "#003399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#003399")   + 
 stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Aerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Aerobe abundance")
ggsave("lungaerobeLongit.jpeg", width = 8, height = 4)
# facultative
pvalues_adjustedforpairwise <- compare_means(FacultAnaerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = lungmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

lungfacultaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = FacultAnaerobeSum))  + geom_boxplot(color = "#339999", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#339999")   + 
  stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Facultative Anaerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Facultative Anaerobe abundance")
ggsave("lungfacultaerobeLongit.jpeg", width = 8, height = 4)

lungoxygen <- ggarrange(lunganaerobeLongit, lungaerobeLongit, lungfacultaerobeLongit, ncol = 3)
ggsave("lungoxygen.jpeg", width = 10, height = 5)


###### gut CLR
gutmbio_oxygentaxa <- subset(gutmbio, select = c(AnaerobeSum, AerobeSum , FacultAnaerobeSum))
gutmbio_oxygentaxa$OtherSum <- 1 - (gutmbio_oxygentaxa$AnaerobeSum + gutmbio_oxygentaxa$AerobeSum + gutmbio_oxygentaxa$FacultAnaerobeSum)

gutmbio_oxygentaxa_clr <- as.data.frame(clr(gutmbio_oxygentaxa))
oldnames = c( "AnaerobeSum", "AerobeSum", "FacultAnaerobeSum","OtherSum" )
newnames = c("AnaerobeCLR", "AerobeCLR", "FacultAnaerobeCLR","OtherCLR" )
gutmbio_oxygentaxa_clr <- gutmbio_oxygentaxa_clr %>% rename_at(vars(oldnames), ~ newnames)

gutmbio_clr <- cbind(gutmbio, gutmbio_oxygentaxa_clr)

### manual p-values
# anaerobes
pvalues_adjustedforpairwise <- compare_means(AnaerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = gutmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

gutanaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = AnaerobeSum))  + geom_boxplot(color = "#660066", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#660066")  + 
   stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
   theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Anaerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe abundance")
ggsave("gutanaerobeLongit.jpeg", width = 8, height = 4)
# aerobes
pvalues_adjustedforpairwise <- compare_means(AerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = gutmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

gutaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = AerobeSum))  + geom_boxplot(color = "#003399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#003399")   + 
 stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Aerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Aerobe abundance")
ggsave("gutaerobeLongit.jpeg", width = 8, height = 4)
# facultative
pvalues_adjustedforpairwise <- compare_means(FacultAnaerobeCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = gutmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

gutfacultaerobeLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = FacultAnaerobeSum))  + geom_boxplot(color = "#339999", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#339999")   + 
  stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Facultative Anaerobes") + theme(axis.text.x = element_text(size = 8))  + ylab("Facultative Anaerobe abundance")
ggsave("gutfacultaerobeLongit.jpeg", width = 8, height = 4)

gutoxygen <- ggarrange(gutanaerobeLongit, gutaerobeLongit, gutfacultaerobeLongit, ncol = 3)
ggsave("gutoxygen.jpeg", width = 10, height = 5)


########### look at pathogens vs. oral commensals in 3 compartments. 

####### repeat these graphs but perform CLR transformations 
###### take the oxygen variables and then the pathogenicity variables
oralmbio_pathogens <- subset(oralmbio, select = c(OralSum, PathogenSum , OtherSum))
oralmbio_pathogens_clr <- as.data.frame(clr(oralmbio_pathogens))
oldnames = c( "OralSum", "PathogenSum", "OtherSum" )
newnames = c("OralCLR", "PathogenCLR", "OtherNonPathogenCLR" )
oralmbio_pathogens_clr <- oralmbio_pathogens_clr %>% rename_at(vars(oldnames), ~ newnames)

oralmbio_clr_pathogens <- cbind(oralmbio, oralmbio_pathogens_clr)

### oral commensals and pathogens

pvalues_adjustedforpairwise <- compare_means(OralCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = oralmbio_clr_pathogens)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

oralcommensalLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = OralSum))  + geom_boxplot(color = "#00CCFF", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#00CCFF")   + 
 stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
 theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral Commensals") + theme(axis.text.x = element_text(size = 8))  + ylab("Oral Commensal abundance")
ggsave("oralcommensalLongit.jpeg", width = 8, height = 4)


pvalues_adjustedforpairwise <- compare_means(PathogenCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = oralmbio_clr_pathogens)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

oralPathogenLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot(color = "#FF3399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#FF3399")   + 
  stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Respiratory Pathogens") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")
ggsave("oralPathogenLongit.jpeg", width = 8, height = 4)


pvalues_adjustedforpairwise <- compare_means(OtherNonPathogenCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = oralmbio_clr_pathogens)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

oralOtherLongit <- alirmbio %>% filter(SampleTypeFactor=="oral") %>% ggplot(aes(x=studydayfollowup, y = OtherSum))  + geom_boxplot(color = "gray", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "gray")   + 
    stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj")  + 
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Other Taxa") + theme(axis.text.x = element_text(size = 8))  + ylab("Other Taxa abundance")
ggsave("oralOtherLongit.jpeg", width = 8, height = 4)

oralPathogens <- ggarrange(oralcommensalLongit, oralPathogenLongit, oralOtherLongit, ncol = 3)
ggsave("oralPathogens.jpeg", width = 10, height = 5)



##### lung pathogens
###### take the oxygen variables and then the pathogenicity variables
lungmbio_pathogens <- subset(lungmbio, select = c(OralSum, PathogenSum , OtherSum))
lungmbio_pathogens_clr <- as.data.frame(clr(lungmbio_pathogens))
oldnames = c( "OralSum", "PathogenSum", "OtherSum" )
newnames = c("OralCLR", "PathogenCLR", "OtherNonPathogenCLR" )
lungmbio_pathogens_clr <- lungmbio_pathogens_clr %>% rename_at(vars(oldnames), ~ newnames)

lungmbio_clr_path <- cbind(lungmbio, lungmbio_pathogens_clr)

### lung commensals and pathogens

pvalues_adjustedforpairwise <- compare_means(OralCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = lungmbio_clr_path)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

lungcommensalLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = OralSum))  + geom_boxplot(color = "#00CCFF", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#00CCFF")   + 
 stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
 theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral Commensals") + theme(axis.text.x = element_text(size = 8))  + ylab("Oral Commensal abundance")
ggsave("lungcommensalLongit.jpeg", width = 8, height = 4)


pvalues_adjustedforpairwise <- compare_means(PathogenCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = lungmbio_clr_path)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

lungPathogenLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot(color = "#FF3399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#FF3399")   + 
  stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Respiratory Pathogens") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")
ggsave("lungPathogenLongit.jpeg", width = 8, height = 4)


pvalues_adjustedforpairwise <- compare_means(OtherNonPathogenCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = lungmbio_clr_path)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

lungOtherLongit <- alirmbio %>% filter(SampleTypeFactor=="lung") %>% ggplot(aes(x=studydayfollowup, y = OtherSum))  + geom_boxplot(color = "gray", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "gray")   + 
    stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj")  + 
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Other Taxa") + theme(axis.text.x = element_text(size = 8))  + ylab("Other Taxa abundance")
ggsave("lungOtherLongit.jpeg", width = 8, height = 4)

lungPathogens <- ggarrange(lungcommensalLongit, lungPathogenLongit, lungOtherLongit, ncol = 3)
ggsave("lungPathogens.jpeg", width = 10, height = 5)



##### gut commensals vs pathogens
gutmbio_pathogens <- subset(gutmbio, select = c(OralSum, PathogenSum , OtherSum))
gutmbio_pathogens_clr <- as.data.frame(clr(gutmbio_pathogens))
oldnames = c( "OralSum", "PathogenSum", "OtherSum" )
newnames = c("OralCLR", "PathogenCLR", "OtherNonPathogenCLR" )
gutmbio_pathogens_clr <- gutmbio_pathogens_clr %>% rename_at(vars(oldnames), ~ newnames)

gutmbio_clr <- cbind(gutmbio, gutmbio_pathogens_clr)

### gut commensals and pathogens

pvalues_adjustedforpairwise <- compare_means(OralCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = gutmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

gutcommensalLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = OralSum))  + geom_boxplot(color = "#00CCFF", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "#00CCFF")   + 
 stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
 theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral Commensals") + theme(axis.text.x = element_text(size = 8))  + ylab("Oral Commensal abundance")
ggsave("gutcommensalLongit.jpeg", width = 8, height = 4)


pvalues_adjustedforpairwise <- compare_means(PathogenCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = gutmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

gutPathogenLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot(color = "#FF3399", outlier.shape = NA) + geom_jitter(size = 0.3, color = "#FF3399")   + 
  stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj") +
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Respiratory Pathogens") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen abundance")
ggsave("gutPathogenLongit.jpeg", width = 8, height = 4)


pvalues_adjustedforpairwise <- compare_means(OtherNonPathogenCLR ~ studydayfollowup, comparisons = my_comparisons, p.adjust.method = "bonferroni", method='wilcox.test', data = gutmbio_clr)
pvalues_adjustedforpairwise <- pvalues_adjustedforpairwise %>% mutate(y.position = c(0.8, 1.0, 0.9))

gutOtherLongit <- alirmbio %>% filter(SampleTypeFactor=="gut") %>% ggplot(aes(x=studydayfollowup, y = OtherSum))  + geom_boxplot(color = "gray", outlier.shape = NA) + geom_beeswarm(size = 0.3, color = "gray")   + 
    stat_pvalue_manual(pvalues_adjustedforpairwise, label = "p.adj")  + 
  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Other Taxa") + theme(axis.text.x = element_text(size = 8))  + ylab("Other Taxa abundance")
ggsave("gutOtherLongit.jpeg", width = 8, height = 4)

gutPathogens <- ggarrange(gutcommensalLongit, gutPathogenLongit, gutOtherLongit, ncol = 3)
ggsave("gutPathogens.jpeg", width = 10, height = 5)





```


```{r antibiotic effects in mixed linear models}

tblAntibioticHistory = read_xlsx("tblAntibioticHistory.xlsx") 

# tblAntibioticHistory <- filter(tblAntibioticHistory, day1Date !="")

tblAntibioticHistory$ZosynStartDate <- as_date(tblAntibioticHistory$ZosynStartDate)
tblAntibioticHistory$ZosynEndDate <- as_date(tblAntibioticHistory$ZosynEndDate)
tblAntibioticHistory$day1Date<- as_date(tblAntibioticHistory$day1Date)
tblAntibioticHistory$MetronidazoleStartDate<- as_date(tblAntibioticHistory$MetronidazoleStartDate)
tblAntibioticHistory$MetronidazoleEndDate<- as_date(tblAntibioticHistory$MetronidazoleEndDate)
tblAntibioticHistory$MeropenemStartDate<- as_date(tblAntibioticHistory$MeropenemStartDate)
tblAntibioticHistory$MeropenemEndDate<- as_date(tblAntibioticHistory$MeropenemEndDate)
tblAntibioticHistory$SulbactamStartDate<- as_date(tblAntibioticHistory$SulbactamStartDate)
tblAntibioticHistory$SulbactamEndDate<- as_date(tblAntibioticHistory$SulbactamEndDate)


tblAntibioticHistory <- tblAntibioticHistory %>%
  rowwise() %>%
  mutate(
    onZosyn = ifelse(!is.na(day1Date) & 
                     day1Date >= ZosynStartDate & 
                     day1Date <= ZosynEndDate, 1, 0),
    onMetronidazole = ifelse(!is.na(day1Date) & 
                             day1Date >= MetronidazoleStartDate & 
                             day1Date <= MetronidazoleEndDate, 1, 0),
    onMeropenem = ifelse(!is.na(day1Date) & 
                         day1Date >= MeropenemStartDate & 
                         day1Date <= MeropenemEndDate, 1, 0),
    onSulbactam = ifelse(!is.na(day1Date) & 
                         day1Date >= SulbactamStartDate & 
                         day1Date <= SulbactamEndDate, 1, 0)
  )

table(tblAntibioticHistory$onZosyn)
table(tblAntibioticHistory$onMetronidazole)
table(tblAntibioticHistory$onMeropenem)

tblAntibioticHistory <- tblAntibioticHistory %>%  mutate(day1AnaerobicAbx = case_when(onZosyn==1 ~ "yes",
                                                                                     onMetronidazole==1 ~ "yes",
                                                                                     onMeropenem==1 ~ "yes",
                                                                                     onSulbactam==1 ~ "yes"))


tblAntibioticHistory$day1AnaerobicAbx[is.na(tblAntibioticHistory$day1AnaerobicAbx)] <- "no"

NAT.score = read_csv("Abx_NAT_score_45days.csv")
NAT.score = NAT.score[,c(1,16)] # day1 = V15

NAT.score$SubjectID <-  NAT.score$...1 

NAT.score$NAT <-  NAT.score$V15
NAT.score <- subset(NAT.score, select = c(SubjectID, NAT))
AnaerobicAbx <- subset(tblAntibioticHistory, select = c(SubjectID, day1AnaerobicAbx))

abxexposure <- merge(AnaerobicAbx, NAT.score, by = "SubjectID", all = TRUE)


#### oral over time mixed linear regression models. 

oralmbio_clr <- merge(oralmbio_clr, abxexposure, by = "SubjectID", all.x = TRUE)

oralmbio_clr$day1AnaerobicAbx <- as.factor(oralmbio_clr$day1AnaerobicAbx)
names(oralmbio_clr)
oralmbio_clr$AnaerobeCLR
# shannon oral
model=lme(ShannonIndex~timesamplefromicuadmission + NAT, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
# Anaerobe oral
model=lme(AnaerobeSum~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(AnaerobeSum~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(AnaerobeSum~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(AnaerobeSum~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 

# Pathogen oral
model=lme(PathogenSum~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(PathogenSum~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(PathogenSum~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(PathogenSum~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 

### qpcr oral

model=lme(Nof16scopies~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(Nof16scopies~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(Nof16scopies~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(Nof16scopies~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = oralmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 



##### lung


lungmbio_clr <- merge(lungmbio_clr, abxexposure, by = "SubjectID", all.x = TRUE)

lungmbio_clr$day1AnaerobicAbx <- as.factor(lungmbio_clr$day1AnaerobicAbx)
names(lungmbio_clr)
lungmbio_clr$AnaerobeCLR
# shannon lung

model=lme(ShannonIndex~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + NAT, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(ShannonIndex~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
# Anaerobe lung
model=lme(AnaerobeSum~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(AnaerobeSum~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(AnaerobeSum~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(AnaerobeSum~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 

# Pathogen lung

model=lme(PathogenSum~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(PathogenSum~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(PathogenSum~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(PathogenSum~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 

####qpcr
model=lme(Nof16scopies~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(Nof16scopies~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(Nof16scopies~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(Nof16scopies~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = lungmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 


####### gut
gutmbio_clr <- merge(gutmbio_clr, abxexposure, by = "SubjectID", all.x = TRUE)

gutmbio_clr$day1AnaerobicAbx <- as.factor(gutmbio_clr$day1AnaerobicAbx)
names(gutmbio_clr)
gutmbio_clr$AnaerobeCLR
# shannon gut

model=lme(ShannonIndex~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(ShannonIndex~timesamplefromicuadmission + NAT, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(ShannonIndex~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
# Anaerobe gut
model=lme(AnaerobeSum~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(AnaerobeSum~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(AnaerobeSum~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(AnaerobeSum~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 

# Pathogen gut

model=lme(PathogenSum~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(PathogenSum~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(PathogenSum~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(PathogenSum~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 

# qpcr
model=lme(Nof16scopies~timesamplefromicuadmission + day1AnaerobicAbx, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(Nof16scopies~timesamplefromicuadmission + total_ABx_score_day1_Convex, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 
model=lme(Nof16scopies~ timesamplefromicuadmission + NAT, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 0.
model=lme(Nof16scopies~timesamplefromicuadmission + studyday0_mgpred, random=~1|SubjectID, data = gutmbio_clr, method = "REML", na.action = na.omit)
anova(model) # 


```



```{r representative taxa barplots}

#### obtain bug classifications and use the filtered taxa table to create relative abundance plots for each compartment and each type of bugs. 

# names(filtered_taxa)

## convert all to numeric
filtered_taxa <- mutate_all(filtered_taxa, as.numeric)
##### remove the sample_id
filtered_taxa$sample_id <- NULL
# Divide each value in columns 3 to 311 by the corresponding "total" value

relative_abundance <- filtered_taxa /  rowSums(filtered_taxa)

# relative_abundance <- filtered_taxa /  filtered_taxa$total
# rownames(relative_abundance) <- filtered_taxa$sample_id

# Now 'relative_abundance' contains the relative abundance values for each taxon in each sample

### take oralmbio sample_ids to filtered relative abundance
relative_abundance_oralsamples <- relative_abundance[rownames(relative_abundance) %in% oralmbio$sample_id, ]

#### find the most abundant samples. 

# Calculate mean relative abundance for each taxon across all samples
mean_relative_abundance_oralsamples <- colMeans(relative_abundance_oralsamples, na.rm = TRUE)

# Sort taxa by mean relative abundance in descending order
top_taxa_oral <- sort(mean_relative_abundance_oralsamples, decreasing = TRUE)

# Extract top 15 taxa
top_taxa_oral <- head(top_taxa_oral, 15)

# Now 'top_15_taxa' contains the top 15 taxa by relative abundance

#### now filter the relative_abundance_oralsamples to include only these 15 taxa. 
top_taxa_oral_names <- names(top_taxa_oral)
col_names_relative_abundance_oralsamples <- colnames(relative_abundance_oralsamples)

relative_abundance_oralsamples_toptaxa <- relative_abundance_oralsamples[, col_names_relative_abundance_oralsamples %in% top_taxa_oral_names]

# Get the last part of the variable names after the last ";"
# new_names <- sapply(strsplit(colnames(relative_abundance_oralsamples_toptaxa), ";"), function(x) tail(x, 1))

# Rename the variables
# colnames(relative_abundance_oralsamples_toptaxa) <- new_names
# Now the variables are renamed with the last part of the original variable names after the last ";"

# Calculate mean relative abundance for each taxon
mean_relative_abundance <- colMeans(relative_abundance_oralsamples_toptaxa, na.rm = TRUE)

# Calculate standard error for each taxon
std_error_relative_abundance <- apply(relative_abundance_oralsamples_toptaxa, 2, function(x) sd(x, na.rm = TRUE)/sqrt(length(x)))

# Create a dataframe for plotting
plot_data <- data.frame(
  Taxon = names(mean_relative_abundance),
  Mean_Relative_Abundance = mean_relative_abundance,
  Std_Error = std_error_relative_abundance
)

# Order dataframe by mean relative abundance in descending order
plot_data <- plot_data[order(plot_data$Mean_Relative_Abundance, decreasing = TRUE),]

# Create the bar plot
oralsamples_top15taxabarplot <- 
ggplot(plot_data, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = Taxon)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
  scale_fill_viridis_d(option = "magma") + # Use Viridis color scale for discrete data
  labs(x = "", y = "Mean Relative Abundance", title = "Top Taxa by Relative Abundance in Oral Samples") +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("oralsamples_top15taxabarplot.jpeg", width = 8, height = 6)


### create subset for the anaerobes and pathogens
# new_names
### from these organisms, it is "Alloprevotella"         "Prevotella"             "Prevotellaceae_Uncltrd"VeillonellaFusobacterium who are anaerobes. 
# define new dataset for these anaerobes. 
plotdata_anaerobes <- filter(plot_data, Taxon %in% c("Alloprevotella", "Prevotella", "Prevotellaceae_Uncltrd", "Veillonella", "Fusobacterium"))

oralsamples_anaerobestaxabarplot <- 
ggplot(plotdata_anaerobes, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = "#660066")) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
    labs(x = "", y = "Mean Relative Abundance", title = "Obligate Anaerobe Taxa") +
  scale_fill_manual(values = "#660066" ) +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("oralsamples_anaerobestaxabarplot.jpeg", width = 4, height = 6)

# pathogens
plotdata_pathogens <- filter(plot_data, Taxon %in% c("Enterococcus", "Staphylococcus", "Pseudomonas"))
oralsamples_pathogensstaxabarplot <- 
ggplot(plotdata_pathogens, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = "#FF3399")) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
    labs(x = "", y = "Mean Relative Abundance", title = "Respiratory Pathogens") +
  scale_fill_manual(values = "#FF3399" ) +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("ooralsamples_pathogensstaxabarplot.jpeg", width = 3, height = 6)




######## create barplot for top lung taxa. 


### take lungmbio sample_ids to filtered relative abundance
relative_abundance_lungsamples <- relative_abundance[rownames(relative_abundance) %in% lungmbio$sample_id, ]

#### find the most abundant samples. 

# Calculate mean relative abundance for each taxon across all samples
mean_relative_abundance_lungsamples <- colMeans(relative_abundance_lungsamples, na.rm = TRUE)

# Sort taxa by mean relative abundance in descending order
top_taxa_lung <- sort(mean_relative_abundance_lungsamples, decreasing = TRUE)

# Extract top 15 taxa
top_taxa_lung <- head(top_taxa_lung, 15)

# Now 'top_15_taxa' contains the top 15 taxa by relative abundance

#### now filter the relative_abundance_lungsamples to include only these 15 taxa. 
top_taxa_lung_names <- names(top_taxa_lung)
col_names_relative_abundance_lungsamples <- colnames(relative_abundance_lungsamples)

relative_abundance_lungsamples_toptaxa <- relative_abundance_lungsamples[, col_names_relative_abundance_lungsamples %in% top_taxa_lung_names]

# Get the last part of the variable names after the last ";"
# new_names <- sapply(strsplit(colnames(relative_abundance_lungsamples_toptaxa), ";"), function(x) tail(x, 1))

# Rename the variables
# colnames(relative_abundance_lungsamples_toptaxa) <- new_names

# Now the variables are renamed with the last part of the original variable names after the last ";"

# Calculate mean relative abundance for each taxon
mean_relative_abundance <- colMeans(relative_abundance_lungsamples_toptaxa, na.rm = TRUE)

# Calculate standard error for each taxon
std_error_relative_abundance <- apply(relative_abundance_lungsamples_toptaxa, 2, function(x) sd(x, na.rm = TRUE)/sqrt(length(x)))

# Create a dataframe for plotting
plot_data <- data.frame(
  Taxon = names(mean_relative_abundance),
  Mean_Relative_Abundance = mean_relative_abundance,
  Std_Error = std_error_relative_abundance
)

# Order dataframe by mean relative abundance in descending order
plot_data <- plot_data[order(plot_data$Mean_Relative_Abundance, decreasing = TRUE),]

# Create the bar plot
lungsamples_top15taxabarplot <- 
ggplot(plot_data, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = Taxon)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
  scale_fill_viridis_d(option = "magma") + # Use Viridis color scale for discrete data
  labs(x = "", y = "Mean Relative Abundance", title = "Top Taxa by Relative Abundance in Lung Samples") +
  theme_pubr() +theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("lungsamples_top15taxabarplot.jpeg", width = 8, height = 6)


### create subset for the anaerobes and pathogens
# new_names
### from these organisms, it is "Alloprevotella"         "Prevotella"             "Prevotellaceae_Uncltrd"VeillonellaFusobacterium who are anaerobes. 
# define new dataset for these anaerobes. 
plotdata_anaerobes <- filter(plot_data, Taxon %in% c( "Prevotella", "Veillonella", "Fusobacterium"))

lungsamples_anaerobestaxabarplot <- 
ggplot(plotdata_anaerobes, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = "#660066")) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
    labs(x = "", y = "Mean Relative Abundance", title = "Obligate Anaerobe Taxa") +
  scale_fill_manual(values = "#660066" ) +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("lungsamples_anaerobestaxabarplot.jpeg", width = 3, height = 6)

# pathogens
plotdata_pathogens <- filter(plot_data, Taxon %in% c("Enterococcus", "Staphylococcus", "Pseudomonas", "Escherichia_Shigella", "Klebsiella","Stenotrophomonas"  ))
lungsamples_pathogensstaxabarplot <- 
ggplot(plotdata_pathogens, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = "#FF3399")) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
    labs(x = "", y = "Mean Relative Abundance", title = "Respiratory Pathogens") +
  scale_fill_manual(values = "#FF3399" ) +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("lungsamples_pathogensstaxabarplot.jpeg", width = 5, height = 6)




######## create barplot for top gut taxa. 


### take gutmbio sample_ids to filtered relative abundance
relative_abundance_gutsamples <- relative_abundance[rownames(relative_abundance) %in% gutmbio$sample_id, ]

#### find the most abundant samples. 

# Calculate mean relative abundance for each taxon across all samples
mean_relative_abundance_gutsamples <- colMeans(relative_abundance_gutsamples, na.rm = TRUE)

# Sort taxa by mean relative abundance in descending order
top_taxa_gut <- sort(mean_relative_abundance_gutsamples, decreasing = TRUE)

# Extract top 15 taxa
top_taxa_gut <- head(top_taxa_gut, 15)

# Now 'top_15_taxa' contains the top 15 taxa by relative abundance

#### now filter the relative_abundance_gutsamples to include only these 15 taxa. 
top_taxa_gut_names <- names(top_taxa_gut)
col_names_relative_abundance_gutsamples <- colnames(relative_abundance_gutsamples)

relative_abundance_gutsamples_toptaxa <- relative_abundance_gutsamples[, col_names_relative_abundance_gutsamples %in% top_taxa_gut_names]

# Get the last part of the variable names after the last ";"
# new_names <- sapply(strsplit(colnames(relative_abundance_gutsamples_toptaxa), ";"), function(x) tail(x, 1))

# Rename the variables
# colnames(relative_abundance_gutsamples_toptaxa) <- new_names

# Now the variables are renamed with the last part of the original variable names after the last ";"

# Calculate mean relative abundance for each taxon
mean_relative_abundance <- colMeans(relative_abundance_gutsamples_toptaxa, na.rm = TRUE)

# Calculate standard error for each taxon
std_error_relative_abundance <- apply(relative_abundance_gutsamples_toptaxa, 2, function(x) sd(x, na.rm = TRUE)/sqrt(length(x)))

# Create a dataframe for plotting
plot_data <- data.frame(
  Taxon = names(mean_relative_abundance),
  Mean_Relative_Abundance = mean_relative_abundance,
  Std_Error = std_error_relative_abundance
)

# Order dataframe by mean relative abundance in descending order
plot_data <- plot_data[order(plot_data$Mean_Relative_Abundance, decreasing = TRUE),]

# Create the bar plot
gutsamples_top15taxabarplot <- 
ggplot(plot_data, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = Taxon)) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
  scale_fill_viridis_d(option = "cividis") + # Use Viridis color scale for discrete data
  labs(x = "", y = "Mean Relative Abundance", title = "Top Taxa by Relative Abundance in Gut Samples") +
theme_pubr() +  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("gutsamples_top15taxabarplot.jpeg", width = 8, height = 6)


### create subset for the anaerobes and pathogens
# new_names
### from these organisms, it is "Alloprevotella"         "Prevotella"             "Prevotellaceae_Uncltrd"VeillonellaFusobacterium who are anaerobes. 
# define new dataset for these anaerobes. 
plotdata_anaerobes <- filter(plot_data, Taxon %in% c( "Prevotella", "Bacteroides", "Veillonella", "Porphyromonas", "Alistipes", "Clostridia_vadinBB60_grp_ge", "Anaerococcus", "Finegoldia", "Akkermansia"))

gutsamples_anaerobestaxabarplot <- 
ggplot(plotdata_anaerobes, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = "#660066")) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
    labs(x = "", y = "Mean Relative Abundance", title = "Obligate Anaerobe Taxa") +
  scale_fill_manual(values = "#660066" ) +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("gutsamples_anaerobestaxabarplot.jpeg", width = 6, height = 6)

# pathogens
plotdata_pathogens <- filter(plot_data, Taxon %in% c("Enterococcus", "Staphylococcus", "Escherichia_Shigella", "Klebsiella"  ))
gutsamples_pathogensstaxabarplot <- 
ggplot(plotdata_pathogens, aes(x = reorder(Taxon, -Mean_Relative_Abundance), y = Mean_Relative_Abundance, fill = "#FF3399")) +
  geom_bar(stat = "identity") +
  geom_errorbar(aes(ymin = Mean_Relative_Abundance - Std_Error, ymax = Mean_Relative_Abundance + Std_Error), width = 0.2) +
    labs(x = "", y = "Mean Relative Abundance", title = "Respiratory Pathogens") +
  scale_fill_manual(values = "#FF3399" ) +
  theme_pubr() + 
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none") + ylim(c(0, 0.20))
ggsave("gutsamples_pathogensstaxabarplot.jpeg", width = 4, height = 6)


```




```{r host response}
#### examine for differences in biomarkers by clusters. 

#### oral 
color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
my_comparisons <- list(c("Low_Diversity", "Interm_Diversity"), c("Interm_Diversity", "High_Diversity"))

### oral clusters diversity

oralil6 <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = il6, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma IL6") + ylab("") + theme(axis.text.x = element_blank())
oralil6_eta_pradj <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = il6_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA IL6") + ylab("") + theme(axis.text.x = element_blank())

oraltnfr1 <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = tnfr1_final, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sTNFR1") + ylab("") + theme(axis.text.x = element_blank())
oraltnfr1_eta_pradj <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = tnfr1_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sTNFR1") + ylab("") + theme(axis.text.x = element_blank())

oralrage <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = rage, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sRAGE") + ylab("") + theme(axis.text.x = element_blank())
oralrage_eta_pradj <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = rage_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sRAGE") + ylab("") + theme(axis.text.x = element_blank())

oralang2 <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = ang2, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Ang-2") + ylab("") + theme(axis.text.x = element_blank())
oralang2_eta_pradj <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = ang2_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Ang-2") + ylab("") + theme(axis.text.x = element_blank()) 


oralprocalcitonin <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = procalcitonin, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Procalcitonin") + ylab("") + theme(axis.text.x = element_blank())
oralprocalcitonin_eta_pradj <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = procalcitonin_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Procalcitonin") + ylab("") + theme(axis.text.x = element_blank())


oralpentraxin3 <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = pentraxin3, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Pentraxin3") + ylab("") + theme(axis.text.x = element_blank())
oralpentraxin3_eta_pradj <- oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = pentraxin3_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Pentraxin3") + ylab("") + theme(axis.text.x = element_blank())

oralbiomarkers <- ggarrange(oralil6_eta_pradj, oraltnfr1_eta_pradj, oralrage_eta_pradj, oralang2_eta_pradj, oralprocalcitonin_eta_pradj, oralpentraxin3_eta_pradj, oralil6, oraltnfr1, oralrage, oralang2, oralprocalcitonin, oralpentraxin3, ncol = 6, nrow = 2)
ggsave("oralbiomarkers.jpeg", width = 16, height = 8)


##### look at additional biomarkers
lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = bdg, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("") 

oralmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = bdg, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("") 

gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = bdg, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("") 

####  lung
# lungmbio <- filter(alirmbio, SampleTypeFactor=="lung")
table(lungmbio$BaslClusters)

lungil6 <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = il6, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma IL6") + ylab("") + theme(axis.text.x = element_blank())
lungil6_eta_pradj <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = il6_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA IL6") + ylab("") + theme(axis.text.x = element_blank())

lungtnfr1 <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = tnfr1_final, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sTNFR1") + ylab("") + theme(axis.text.x = element_blank())
lungtnfr1_eta_pradj <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = tnfr1_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sTNFR1") + ylab("") + theme(axis.text.x = element_blank())

lungrage <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = rage, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sRAGE") + ylab("") + theme(axis.text.x = element_blank())
lungrage_eta_pradj <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = rage_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sRAGE") + ylab("") + theme(axis.text.x = element_blank())

lungang2 <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = ang2, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Ang-2") + ylab("") + theme(axis.text.x = element_blank())
lungang2_eta_pradj <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = ang2_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Ang-2") + ylab("") + theme(axis.text.x = element_blank()) 


lungprocalcitonin <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = procalcitonin, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Procalcitonin") + ylab("") + theme(axis.text.x = element_blank())
lungprocalcitonin_eta_pradj <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = procalcitonin_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Procalcitonin") + ylab("") + theme(axis.text.x = element_blank())


lungpentraxin3 <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = pentraxin3, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Pentraxin3") + ylab("") + theme(axis.text.x = element_blank())
lungpentraxin3_eta_pradj <- lungmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = pentraxin3_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Pentraxin3") + ylab("") + theme(axis.text.x = element_blank())

lungbiomarkers <- ggarrange(lungil6_eta_pradj, lungtnfr1_eta_pradj, lungrage_eta_pradj, lungang2_eta_pradj, lungprocalcitonin_eta_pradj, lungpentraxin3_eta_pradj, lungil6, lungtnfr1, lungrage, lungang2, lungprocalcitonin, lungpentraxin3, ncol = 6, nrow = 2)
ggsave("lungbiomarkers.jpeg", width = 16, height = 8)



#### gut and biomarkers

gutil6 <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = il6, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma IL6") + ylab("") + theme(axis.text.x = element_blank())
gutil6_eta_pradj <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = il6_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA IL6") + ylab("") + theme(axis.text.x = element_blank())

guttnfr1 <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = tnfr1_final, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sTNFR1") + ylab("") + theme(axis.text.x = element_blank())
guttnfr1_eta_pradj <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = tnfr1_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sTNFR1") + ylab("") + theme(axis.text.x = element_blank())

gutrage <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = rage, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sRAGE") + ylab("") + theme(axis.text.x = element_blank())
gutrage_eta_pradj <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = rage_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sRAGE") + ylab("") + theme(axis.text.x = element_blank())

gutang2 <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = ang2, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Ang-2") + ylab("") + theme(axis.text.x = element_blank())
gutang2_eta_pradj <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = ang2_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Ang-2") + ylab("") + theme(axis.text.x = element_blank()) 


gutprocalcitonin <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = procalcitonin, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Procalcitonin") + ylab("") + theme(axis.text.x = element_blank())
gutprocalcitonin_eta_pradj <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = procalcitonin_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Procalcitonin") + ylab("") + theme(axis.text.x = element_blank())


gutpentraxin3 <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = pentraxin3, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Pentraxin3") + ylab("") + theme(axis.text.x = element_blank())
gutpentraxin3_eta_pradj <- gutmbio %>% filter(!is.na(BaslClusters)) %>% ggplot(aes(x=BaslClusters, y = pentraxin3_eta_pradj, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) +scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Pentraxin3") + ylab("") + theme(axis.text.x = element_blank())

gutbiomarkers <- ggarrange(gutil6_eta_pradj, guttnfr1_eta_pradj, gutrage_eta_pradj, gutang2_eta_pradj, gutprocalcitonin_eta_pradj, gutpentraxin3_eta_pradj, gutil6, guttnfr1, gutrage, gutang2, gutprocalcitonin, gutpentraxin3, ncol = 6, nrow = 2)
ggsave("gutbiomarkers.jpeg", width = 16, height = 8)



```



```{r phenotypes}

### look at relationship with phenotypes

#### chord diagram: 
# install.packages("circlize")
library(circlize)
# oral 
oralmbio$DrohanPhenotype <- oralmbio$phenotype
oralmbio$DrohanPhenotype[oralmbio$DrohanPhenotype=="hyperinf"] <-"HyperInflammatory"
oralmbio$DrohanPhenotype[oralmbio$DrohanPhenotype=="hypoinfl"] <-"HypoInflammatory"
subset_data <- subset(oralmbio, select = c(SubjectID, BaslClusters, DrohanPhenotype))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
category_colors <- c("#CC0033", "#FF3399", "#FFCCCC", "#000033", "#0000FF", "#0099CC") # Adjust the colors as desired
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
grid.col = c(High_Diversity = "#4DBBD5FF", Interm_Diversity = "#F39B7FFF", Low_Diversity = "#7E6148FF",
    HyperInflammatory = "red", HypoInflammatory = "blue")
chordDiagram(category_matrix, col = category_colors, grid.col = grid.col)


##### look at relationship
chisq.test(category_matrix) # p=0.68

oralmbio <- oralmbio %>% mutate(OralLowDiv = case_when(BaslClusters=="Low_Diversity" ~ "yes",
                                                                     BaslClusters!="Low_Diversity" ~ "no"))


oddsratio.fisher(oralmbio$OralLowDiv, oralmbio$DrohanPhenotype) # ns -  yes    1.110652 0.6401219 1.962872 p =0.702 

#####
# lung 
lungmbio$DrohanPhenotype <- lungmbio$phenotype
lungmbio$DrohanPhenotype[lungmbio$DrohanPhenotype=="hyperinf"] <-"HyperInflammatory"
lungmbio$DrohanPhenotype[lungmbio$DrohanPhenotype=="hypoinfl"] <-"HypoInflammatory"
subset_data <- subset(lungmbio, select = c(SubjectID, BaslClusters, DrohanPhenotype))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
# category_colors <- c("#CC0033", "#FF3399", "#FFCCCC", "#000033", "#0000FF", "#0099CC") # Adjust the colors as desired
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
# grid.col = c(High_Diversity = "#4DBBD5FF", Interm_Diversity = "#F39B7FFF", Low_Diversity = "#7E6148FF",HyperInflammatory = "red", HypoInflammatory = "blue")
chordDiagram(category_matrix, col = category_colors, grid.col = grid.col)


chisq.test(category_matrix) # p=0.4018

lungmbio <- lungmbio %>% mutate(lungLowDiv = case_when(BaslClusters=="Low_Diversity" ~ "yes",
                                                                     BaslClusters!="Low_Diversity" ~ "no"))
oddsratio.fisher(lungmbio$lungLowDiv, lungmbio$DrohanPhenotype) # ns -  yes    0.7510685 0.4587907 0.231698


####### look at integrative clusters for lung data.

subset_data <- subset(lungmbio, select = c(SubjectID, subject_level_cluster, DrohanPhenotype))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
chisq.test(category_matrix) # p=0.83


# gut 
gutmbio$DrohanPhenotype <- gutmbio$phenotype
gutmbio$DrohanPhenotype[gutmbio$DrohanPhenotype=="hyperinf"] <-"HyperInflammatory"
gutmbio$DrohanPhenotype[gutmbio$DrohanPhenotype=="hypoinfl"] <-"HypoInflammatory"
subset_data <- subset(gutmbio, select = c(SubjectID, BaslClusters, DrohanPhenotype))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
# category_colors <- c("#CC0033", "#FF3399", "#FFCCCC", "#000033", "#0000FF", "#0099CC") # Adjust the colors as desired
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
# grid.col = c(High_Diversity = "#4DBBD5FF", Interm_Diversity = "#F39B7FFF", Low_Diversity = "#7E6148FF",HyperInflammatory = "red", HypoInflammatory = "blue")
chordDiagram(category_matrix, col = category_colors, grid.col = grid.col)

chisq.test(category_matrix) # p= 0.4715
gutmbio <- gutmbio %>% mutate(gutLowDiv = case_when(BaslClusters=="Low_Diversity" ~ "yes",
                                                                     BaslClusters!="Low_Diversity" ~ "no"))
oddsratio.fisher(gutmbio$gutLowDiv, gutmbio$DrohanPhenotype) # ns -  yes    1.135725 0.5475231 2.443042 , p =  0.7433918 




##### diversity and abundance by phenotypes #### focus on baseline
color_scale_phen <- c("HyperInflammatory" = "red", "HypoInflammatory" = "blue")
my_comparisons <- list(c("HyperInflammatory", "HypoInflammatory"))

# shannon by phenotypes
oralPhenosshannon <- oralmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = shannon, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral")

lungPhenosshannon <- lungmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = shannon, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung")

gutPhenosshannon <- gutmbio %>%   filter(!is.na(DrohanPhenotype))  %>%  filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = shannon, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut")

#### anaerobe abundance
oralPhenosAnaerobe <- oralmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = AnaerobeSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral")

lungPhenosAnaerobe <- lungmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = AnaerobeSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung")

gutPhenosAnaerobe <- gutmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = AnaerobeSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut")


#### Pathogen abundance
oralPhenosPathogen <- oralmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Oral")

lungPhenosPathogen <- lungmbio %>%   filter(!is.na(DrohanPhenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung")

gutPhenosPathogen <- gutmbio %>%   filter(!is.na(DrohanPhenotype))  %>%  filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut")

phenosAbundances <- ggarrange(oralPhenosshannon, lungPhenosshannon, gutPhenosshannon, oralPhenosAnaerobe, lungPhenosAnaerobe, gutPhenosAnaerobe, oralPhenosPathogen, lungPhenosPathogen, gutPhenosPathogen, ncol = 3, nrow = 3)
ggsave("phenosAbundances.jpeg", width = 16, height = 12)

lungmbio %>%   filter(!is.na(phenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=phenotype, y = Nof16scopies, color = phenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + 
  #scale_color_manual(values = color_scale_phen) +
  stat_compare_means() + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + scale_y_log10()

# significant association with baseline only for pathogens in lung
 lungmbio %>%   filter(!is.na(DrohanPhenotype))  %>%  ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jco() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") 

 
 
####### stratify by direct or indirect lung injury
###### pneumonia
##### important finding
lungmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + facet_wrap(~Pneumonia)

###### export this figure
lungmbio$PneumoniaFactor <- lungmbio$Pneumonia
lungmbio$PneumoniaFactor[lungmbio$PneumoniaFactor==1] <- "Pneumonia"
lungmbio$PneumoniaFactor[lungmbio$PneumoniaFactor==0] <- "Not_Pneumonia"


PathogenAbundByPhenosPneumonia <- lungmbio %>%   filter(!is.na(DrohanPhenotype)) %>%  filter(studydayfollowup=="baseline") %>%  filter(!is.na(PneumoniaFactor))  %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_boxplot() + geom_jitter(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + facet_wrap(~PneumoniaFactor)
ggsave("PathogenAbundByPhenosPneumonia.jpeg", width = 12, height = 6)


lungmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = shannon, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + facet_wrap(~Pneumonia)

lungmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = AnaerobeSum, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + facet_wrap(~Pneumonia)

#### no significant effects for gut
gutmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("gut") + facet_wrap(~Pneumonia)

gutmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = shannon, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("gut") + facet_wrap(~Pneumonia)

gutmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = AnaerobeSum, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("gut") + facet_wrap(~Pneumonia)

# no significant effects for oral
oralmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = PathogenSum, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("oral") + facet_wrap(~Pneumonia)

oralmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = shannon, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("oral") + facet_wrap(~Pneumonia)

oralmbio %>%   filter(!is.na(DrohanPhenotype)) %>% filter(!is.na(Pneumonia))  %>% ggplot(aes(x=DrohanPhenotype, y = AnaerobeSum, color = DrohanPhenotype))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale_phen) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("oral") + facet_wrap(~Pneumonia)


```





```{r survival}

#### oral cluster survival

#### create publishable figure
oralmbio_basl <- oralmbio_basl %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))

     oralmbio_basl$survival_time60_fromICUAdmission <- oralmbio_basl$survival_time90_fromICUAdmission 
      oralmbio_basl$survival_time60_fromICUAdmission[ oralmbio_basl$survival_time60_fromICUAdmission >60] <- 61       

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~BaslClusters , data=oralmbio_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 
  
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

surv_oralclusters <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("High_Diversity",  "Interm_Diversity", "Low_Diversity"), palette = c("#4DBBD5FF", "#F39B7FFF", "#7E6148FF"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_oralclusters.jpeg", width = 580, height = 418)
print(surv_oralclusters, newpage = FALSE)
dev.off()

##### cox model for oral
oralmbio_basl <- oralmbio_basl %>% mutate(OralLowDiv = case_when(BaslClusters=="Low_Diversity" ~ "yes",
                                                                     BaslClusters!="Low_Diversity" ~ "no"))
survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~OralLowDiv, data=oralmbio_basl)
summary(survival_coxph_bygroup60) # 

# OralLowDivyes     1.487    0.6919    0.99     2.22 p=0.05 

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~OralLowDiv + Age + Gender + HistCOPD + HistImmunosuppression + phenotype + SOFAScRSD, data=oralmbio_basl)
summary(survival_coxph_bygroup60) # ns




#### lung survival
#### create publishable figure
lungmbio_basl <- lungmbio_basl %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))

     lungmbio_basl$survival_time60_fromICUAdmission <- lungmbio_basl$survival_time90_fromICUAdmission 
      lungmbio_basl$survival_time60_fromICUAdmission[ lungmbio_basl$survival_time60_fromICUAdmission >60] <- 61       

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~BaslClusters , data=lungmbio_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 
  
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

surv_lungclusters <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("High_Diversity",  "Interm_Diversity", "Low_Diversity"), palette = c("#4DBBD5FF", "#F39B7FFF", "#7E6148FF"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_lungclusters.jpeg", width = 580, height = 418)
print(surv_lungclusters, newpage = FALSE)
dev.off()
# cox by lung
lungmbio_basl <- lungmbio_basl %>% mutate(lungLowDiv = case_when(BaslClusters=="Low_Diversity" ~ "yes",
                                                                     BaslClusters!="Low_Diversity" ~ "no"))

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~lungLowDiv, data=lungmbio_basl)
summary(survival_coxph_bygroup60) # 
# lungLowDivyes     2.294     0.4359     1.494     3.524 0.000149 ***
### adjusted cox


survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~lungLowDiv + Age + Gender + HistCOPD + HistImmunosuppression + phenotype + SOFAScRSD , data=lungmbio_basl)
summary(survival_coxph_bygroup60) # ##### important - significant result even after adjustment for phenotype and SOFA score. 

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~BaslClusters + Age + Gender + HistCOPD + HistImmunosuppression + phenotype + SOFAScRSD , data=lungmbio_basl)

summary(survival_coxph_bygroup60) # ##### important - significant result even after adjustment for phenotype and SOFA score. 
 #   coef exp(coef)  se(coef)      z Pr(>|z|)    
# BaslClustersInterm_Diversity  0.757102  2.132088  0.398652  1.899   0.0575 .  
# BaslClustersLow_Diversity     1.028230  2.796113  0.384565  2.674   0.0075 ** 

# exp(coef) exp(-coef) lower .95 upper .95
# BaslClustersInterm_Diversity    2.1321     0.4690    0.9760     4.657
# BaslClustersLow_Diversity       2.7961     0.3576    1.3159     5.942
lungmbio_basl$total_ABx_score_day1_Convex
#### create publishable figure
### gut
gutmbio_basl <- filter(gutmbio, studydayfollowup=="baseline")
gutmbio_basl <- gutmbio_basl %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))

     gutmbio_basl$survival_time60_fromICUAdmission <- gutmbio_basl$survival_time90_fromICUAdmission 
      gutmbio_basl$survival_time60_fromICUAdmission[ gutmbio_basl$survival_time60_fromICUAdmission >60] <- 61       

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~BaslClusters , data=gutmbio_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 
  
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

surv_gutclusters <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("High_Diversity",  "Interm_Diversity", "Low_Diversity"), palette = c("#4DBBD5FF", "#F39B7FFF", "#7E6148FF"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_gutclusters.jpeg", width = 580, height = 418)
print(surv_gutclusters, newpage = FALSE)
dev.off()

# cox gut

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~BaslClusters, data=gutmbio_basl)
summary(survival_coxph_bygroup60) # 
# BaslClustersInterm_Diversity -0.668394  0.512531  0.427735 -1.563    0.118
# BaslClustersLow_Diversity    -0.002242  0.997761  0.309175 -0.007    0.994


##### 

#### address R2 comment on longutidinal clusters. 

##### take the lung clusters. 
table(lungSankey_wide$baseline, lungSankey_wide$middle)

### classify these cases. 
LowDivPersisters <- lungSankey_wide$SubjectID[lungSankey_wide$baseline=="Low_Diversity" & lungSankey_wide$middle=="Low_Diversity"]
LowDivResolvers <- lungSankey_wide$SubjectID[lungSankey_wide$baseline=="Low_Diversity" & lungSankey_wide$middle!="Low_Diversity"]
LowDivEmergers <- lungSankey_wide$SubjectID[lungSankey_wide$baseline!="Low_Diversity" & lungSankey_wide$middle=="Low_Diversity"]
NonLowDiv <- lungSankey_wide$SubjectID[lungSankey_wide$baseline!="Low_Diversity" & lungSankey_wide$middle!="Low_Diversity"]

lungmbio_basl <- lungmbio_basl %>% mutate(LungClustersEvolution =  case_when(SubjectID %in% LowDivPersisters ~ "LowDivPersisters", 
                                                                             SubjectID %in% LowDivResolvers ~ "LowDivResolvers",
                                                                             SubjectID %in% LowDivEmergers ~ "LowDivEmergers",
                                                                             SubjectID %in% NonLowDiv ~ "NonLowDiv"
                                                                               ))
                                          
table(lungmbio_basl$LungClustersEvolution)                                          

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~LungClustersEvolution , data=lungmbio_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,   title = "Kaplan-Meier Curve within 60 days") 


survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~LungClustersEvolution, data=lungmbio_basl)
summary(survival_coxph_bygroup60) # 


surv_lungclusters_longitud <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("LowDivEmergers",  "LowDivPersisters", "LowDivResolvers", "NonLowDiv"), palette = c("brown", "black", "blue", "green"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_lungclusters_longitud.jpeg", width = 580, height = 418)
print(surv_lungclusters_longitud, newpage = FALSE)
dev.off()


#### take the two extreme groups
lungmbio_basl <- lungmbio_basl %>% mutate(lungLowDivPersisters = case_when(LungClustersEvolution=="LowDivPersisters" ~ "B_LowDivPersisters", 
                                                                          LungClustersEvolution!="LowDivPersisters" ~ "A_NonPersisters" ))

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~lungLowDivPersisters + Age , data=lungmbio_basl)
summary(survival_coxph_bygroup60) # 2.762     0.3620     1.188     6.421


##### take the oral clusters. 
table(OralSankey_wide$baseline, OralSankey_wide$middle)

### classify these cases. 
LowDivPersisters <- OralSankey_wide$SubjectID[OralSankey_wide$baseline=="Low_Diversity" & OralSankey_wide$middle=="Low_Diversity"]
LowDivResolvers <- OralSankey_wide$SubjectID[OralSankey_wide$baseline=="Low_Diversity" & OralSankey_wide$middle!="Low_Diversity"]
LowDivEmergers <- OralSankey_wide$SubjectID[OralSankey_wide$baseline!="Low_Diversity" & OralSankey_wide$middle=="Low_Diversity"]
NonLowDiv <- OralSankey_wide$SubjectID[OralSankey_wide$baseline!="Low_Diversity" & OralSankey_wide$middle!="Low_Diversity"]

oralmbio_basl <- oralmbio_basl %>% mutate(oralClustersEvolution =  case_when(SubjectID %in% LowDivPersisters ~ "LowDivPersisters", 
                                                                             SubjectID %in% LowDivResolvers ~ "LowDivResolvers",
                                                                             SubjectID %in% LowDivEmergers ~ "LowDivEmergers",
                                                                             SubjectID %in% NonLowDiv ~ "NonLowDiv"
                                                                               ))
                                          
table(oralmbio_basl$oralClustersEvolution)                           



survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~oralClustersEvolution , data=oralmbio_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,   title = "Kaplan-Meier Curve within 60 days") 


survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~oralClustersEvolution, data=oralmbio_basl)
summary(survival_coxph_bygroup60) # 



#### take the two extreme groups
oralmbio_basl <- oralmbio_basl %>% mutate(oralLowDivPersisters = case_when(oralClustersEvolution=="LowDivPersisters" ~ "B_LowDivPersisters", 
                                                                          oralClustersEvolution!="LowDivPersisters" ~ "A_NonPersisters" ))

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~oralLowDivPersisters, data=oralmbio_basl)
summary(survival_coxph_bygroup60) # ns

```


```{r joint modeling}

# Load necessary libraries
library(lme4)
library(survival)

lungmbio <- lungmbio %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))
     lungmbio$survival_time60_fromICUAdmission <- lungmbio$survival_time90_fromICUAdmission 
      lungmbio$survival_time60_fromICUAdmission[ lungmbio$survival_time60_fromICUAdmission >60] <- 61       


library(joineR)
#> Loading required package: survival
data(heart.valve)
heart.surv <- UniqueVariables(heart.valve, 
                              var.col = c("fuyrs", "status"),
                              id.col = "num")
heart.long <- heart.valve[, c("num", "time", "log.lvmi")]
heart.cov <- UniqueVariables(heart.valve, 
                             c("age", "hs", "sex"), 
                             id.col = "num")
heart.valve.jd <- jointdata(longitudinal = heart.long, 
                            baseline = heart.cov, 
                            survival = heart.surv, 
                            id.col = "num", 
                            time.col = "time")

summary(heart.valve.jd)

plot(heart.valve.jd)
# With the creation of the heart.valve.jd object, we can fit a joint model using the joint function. For this, we need 4 arguments:

# jointdata: the data object we created above
# long.formula: the linear mixed effects model formula for the longitudinal sub-model
# surv.formula: the survival formula the survival sub-model
# model: the latent association structure.
fit <- joint(data = heart.valve.jd, 
             long.formula = log.lvmi ~ 1 + time + hs, 
             surv.formula = Surv(fuyrs, status) ~ hs, 
             model = "intslope")

summary(fit)

##### https://cran.r-project.org/web/packages/joineR/vignettes/joineR.html

### do the same for the lungmbio_joint object. 
lungmbio_joint <- filter(lungmbio, !duplicated(SubjectIDDay))
lungmbio_joint <- filter(lungmbio_joint, !is.na(survival_time60_fromICUAdmission))
lungmbio_joint$timesamplefromicuadmission[is.na(lungmbio_joint$timesamplefromicuadmission)] <- 5

lungmbio.surv <- UniqueVariables(lungmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

lungmbio.long <- lungmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"shannon")]

lungmbio.cov <- UniqueVariables(lungmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
lungmbio.jd <- jointdata(longitudinal = lungmbio.long, 
                            baseline = lungmbio.cov, 
                            survival = lungmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")
class(lungmbio.jd)
summary(lungmbio.jd)
plot(lungmbio.jd)

fit <- joint(data = lungmbio.jd, 
             long.formula = shannon ~  timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit)
# model.jointrandom.se_shannonlung <- jointSE(fit, n.boot = 100)

# Extract the relevant rows for variables 2 and 5
relevant_rows <- model.jointrandom.se_shannonlung[c(2, 5), ]
# Calculate exponentiated estimates, 95% lower and upper intervals
exp_estimates <- exp(relevant_rows$Estimate)
lower_intervals <- exp(relevant_rows$"95%Lower")
upper_intervals <- exp(relevant_rows$"95%Upper")
# Combine the results into a dataframe
model.jointrandom.se_shannonlung_results <- data.frame(
  Parameter = relevant_rows$Parameter,
  Exponentiated_Estimate = exp_estimates,
  Lower_Interval = lower_intervals,
  Upper_Interval = upper_intervals
)
# Parameter Exponentiated_Estimate Lower_Interval Upper_Interval
# 1 timesamplefromicuadmission              0.9720967   9.443107e-01      0.9938192
# 2                    gamma_0              0.5339783   2.539523e-15      1.0974617


# sanity check with regular coxph model
survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ shannon + Age, data = lungmbio_joint, x = TRUE)
summary(survival_model)
# for lung shannon, it is the baseline and not the dynamic change. 
# shannon    0.6249     1.6004    0.5074    0.7695


#### repeat for qpcr
lungmbio_jointqpcr <- filter(lungmbio_joint, !is.na(Nof16scopies))
lungmbio_jointqpcr$Nof16scopies <- log10(lungmbio_jointqpcr$Nof16scopies)
### do the same for the lungmbio_joint object. 
lungmbio.surv <- UniqueVariables(lungmbio_jointqpcr, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

lungmbio.long <- lungmbio_jointqpcr[, c("SubjectID", "timesamplefromicuadmission"  ,"Nof16scopies")]

lungmbio.cov <- UniqueVariables(lungmbio_jointqpcr, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
lungmbio.jd <- jointdata(longitudinal = lungmbio.long, 
                            baseline = lungmbio.cov, 
                            survival = lungmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")
class(lungmbio.jd)
summary(lungmbio.jd)
plot(lungmbio.jd)

fit <- joint(data = lungmbio.jd, 
             long.formula = Nof16scopies ~  timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit) # non-significant for qpcr

# model.jointrandom.se_qpcrlung <- jointSE(fit, n.boot = 150)
# model.jointrandom.se_qpcrlung


# Extract the relevant rows for variables 2 and 5
relevant_rows <- model.jointrandom.se_qpcrlung[c(2, 5), ]
# Calculate exponentiated estimates, 95% lower and upper intervals
exp_estimates <- exp(relevant_rows$Estimate)
lower_intervals <- exp(relevant_rows$"95%Lower")
upper_intervals <- exp(relevant_rows$"95%Upper")
# Combine the results into a dataframe
model.jointrandom.se_qpcrlung_results <- data.frame(
  Parameter = relevant_rows$Parameter,
  Exponentiated_Estimate = exp_estimates,
  Lower_Interval = lower_intervals,
  Upper_Interval = upper_intervals
)
# Parameter Exponentiated_Estimate Lower_Interval Upper_Interval
# 1 timesamplefromicuadmission              0.9153946   8.156255e-01      0.9757001
# 2                    gamma_0              0.1693649   7.149881e-07    509.3829035


# sanity check with regular coxph model
survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ log10(Nof16scopies) + Age, data = lungmbio_joint, x = TRUE)
summary(survival_model)
# log10(Nof16scopies)    0.9681     1.0329    0.9039     1.037



#### anaerobe abundance CLR
lungmbio_clr <- lungmbio_clr %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))
     lungmbio_clr$survival_time60_fromICUAdmission <- lungmbio_clr$survival_time90_fromICUAdmission 
      lungmbio_clr$survival_time60_fromICUAdmission[ lungmbio_clr$survival_time60_fromICUAdmission >60] <- 61       


lungmbio_joint <- filter(lungmbio_clr, !duplicated(SubjectIDDay))
lungmbio_joint <- filter(lungmbio_joint, !is.na(survival_time60_fromICUAdmission))
# define timesamplefromicuadmission
lungmbio_joint$SubjectIDDay[is.na(lungmbio_joint$timesamplefromicuadmission)]
lungmbio_joint$timesamplefromicuadmission[is.na(lungmbio_joint$timesamplefromicuadmission)] <- 5


lungmbio.surv <- UniqueVariables(lungmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

lungmbio.long <- lungmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"AnaerobeCLR")]

lungmbio.cov <- UniqueVariables(lungmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
lungmbio.jd <- jointdata(longitudinal = lungmbio.long, 
                            baseline = lungmbio.cov, 
                            survival = lungmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")

fit <- joint(data = lungmbio.jd, 
             long.formula = AnaerobeCLR ~ 1+ timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit, variance = FALSE)

survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ AnaerobeCLR + Age, data = lungmbio_joint, x = TRUE)
summary(survival_model) # aerobeCLR    0.8528     1.1726    0.7821    0.9299

# model.jointrandom.anaerobelung <- jointSE(fit, n.boot = 100)
model.jointrandom.anaerobelung
# Association                    gamma_0  -0.1386 1.2709  -8.0350   0.0212

relevant_rows <- model.jointrandom.anaerobelung[c(2, 5), ]
# Calculate exponentiated estimates, 95% lower and upper intervals
exp_estimates <- exp(relevant_rows$Estimate)
lower_intervals <- exp(relevant_rows$"95%Lower")
upper_intervals <- exp(relevant_rows$"95%Upper")
# Combine the results into a dataframe
model.jointrandom.se_qpcrlung_results <- data.frame(
  Parameter = relevant_rows$Parameter,
  Exponentiated_Estimate = exp_estimates,
  Lower_Interval = lower_intervals,
  Upper_Interval = upper_intervals
)
# Parameter Exponentiated_Estimate Lower_Interval Upper_Interval
#    Parameter Exponentiated_Estimate Lower_Interval Upper_Interval
# 1 timesamplefromicuadmission              0.9194313      0.8672743      0.9584863
 # 2                    gamma_0              0.8544477      0.5321127      1.0010005






### Pathogen Abundance
lungmbio_clr_path <- lungmbio_clr_path %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))
     lungmbio_clr_path$survival_time60_fromICUAdmission <- lungmbio_clr_path$survival_time90_fromICUAdmission 
      lungmbio_clr_path$survival_time60_fromICUAdmission[ lungmbio_clr_path$survival_time60_fromICUAdmission >60] <- 61       



lungmbio_joint <- filter(lungmbio_clr_path, !duplicated(SubjectIDDay))
lungmbio_joint <- filter(lungmbio_joint, !is.na(survival_time60_fromICUAdmission))
# define timesamplefromicuadmission
lungmbio_joint$SubjectIDDay[is.na(lungmbio_joint$timesamplefromicuadmission)]
lungmbio_joint$timesamplefromicuadmission[is.na(lungmbio_joint$timesamplefromicuadmission)] <- 5

lungmbio.surv <- UniqueVariables(lungmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

lungmbio.long <- lungmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"PathogenCLR")]

lungmbio.cov <- UniqueVariables(lungmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
lungmbio.jd <- jointdata(longitudinal = lungmbio.long, 
                            baseline = lungmbio.cov, 
                            survival = lungmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")


fit <- joint(data = lungmbio.jd, 
             long.formula = PathogenCLR ~ 1+ timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit, variance = FALSE)

survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ PathogenCLR + Age, data = lungmbio_joint, x = TRUE)
summary(survival_model) # PathogenCLR     1.148     0.8713     1.080     1.220


# model.jointrandom.pathogenlung <- jointSE(fit, n.boot = 100)
model.jointrandom.pathogenlung
# Association	gamma_0	0.1895	1.7861	-0.0744	0.4669

relevant_rows <- model.jointrandom.pathogenlung[c(2, 5), ]
# Calculate exponentiated estimates, 95% lower and upper intervals
exp_estimates <- exp(relevant_rows$Estimate)
lower_intervals <- exp(relevant_rows$"95%Lower")
upper_intervals <- exp(relevant_rows$"95%Upper")
# Combine the results into a dataframe
model.jointrandom.se_qpcrlung_results <- data.frame(
  Parameter = relevant_rows$Parameter,
  Exponentiated_Estimate = exp_estimates,
  Lower_Interval = lower_intervals,
  Upper_Interval = upper_intervals)

      # Parameter Exponentiated_Estimate Lower_Interval Upper_Interval
# 1 timesamplefromicuadmission               1.243960      1.1717519       1.344739
# 2                    gamma_0               1.208645      0.9074653      49.151138
  
  
  
  

################################## GUT ###########################33



gutmbio <- gutmbio %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))
     gutmbio$survival_time60_fromICUAdmission <- gutmbio$survival_time90_fromICUAdmission 
      gutmbio$survival_time60_fromICUAdmission[ gutmbio$survival_time60_fromICUAdmission >60] <- 61       


gutmbio_joint <- filter(gutmbio, !duplicated(SubjectIDDay))

gutmbio_joint <- filter(gutmbio_joint, !is.na(survival_time60_fromICUAdmission))
gutmbio_joint$timesamplefromicuadmission[is.na(gutmbio_joint$timesamplefromicuadmission)] <- 5

gutmbio.surv <- UniqueVariables(gutmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

gutmbio.long <- gutmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"shannon")]

gutmbio.cov <- UniqueVariables(gutmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
gutmbio.jd <- jointdata(longitudinal = gutmbio.long, 
                            baseline = gutmbio.cov, 
                            survival = gutmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")

fit <- joint(data = gutmbio.jd, 
             long.formula = shannon ~  timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit)
# model.jointrandom.se_shannongut <- jointSE(fit, n.boot = 100)
model.jointrandom.se_shannongut # 	gamma_0	0.0900	0.9135	-0.5938	0.6615

exp(model.jointrandom.se_shannongut$Estimate)
# 
 exp(model.jointrandom.se_shannongut$`95%Lower`)
# 
exp(model.jointrandom.se_shannongut$`95%Upper`)
# 

# Association                    gamma_0  
# sanity check with regular coxph model
survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ shannon + Age, data = gutmbio_joint, x = TRUE)
summary(survival_model)
# shannon    0.9641     1.0373    0.7771     1.196





######### create graph

# Example data
data <- data.frame(
  Variable = c("Variable1", "Variable2", "Variable3"),
  HR = c(1.2, 0.8, 1.5),  # Example hazard ratios
  LowerCI = c(1.1, 0.7, 1.3),  # Example lower bounds of CI
  UpperCI = c(1.3, 0.9, 1.7)   # Example upper bounds of CI
)

# Create the plot
p <- ggplot(data, aes(x = Variable, y = HR)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "Variable", y = "Hazard Ratio", title = "Hazard Ratios with 95% CI") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  # Rotate x-axis labels

# Print the plot
print(p)


#### create a dataframe like this to display all the effect sizes
#### start with the lme models

## lung model results

# shannon
# Parameter Exponentiated_Estimate Lower_Interval Upper_Interval
# 1 timesamplefromicuadmission              0.9720967   9.443107e-01      0.9938192
# 2                    gamma_0              0.5339783   2.539523e-15      1.0974617

# shannon    0.6249     1.6004    0.5074    0.7695

# qpcr
# 1 timesamplefromicuadmission              0.9153946   8.156255e-01      0.9757001
# 2                    gamma_0              0.1693649   7.149881e-07    509.3829035
# log10(Nof16scopies)    0.9681     1.0329    0.9039     1.037

## anaerobe
# 1 timesamplefromicuadmission              0.9194313      0.8672743      0.9584863
 # 2                    gamma_0              0.8544477      0.5321127      1.0010005
# aerobeCLR    0.8528     1.1726    0.7821    0.9299

# pathogenclr
# 1 timesamplefromicuadmission               1.243960      1.1717519       1.344739
# 2                    gamma_0               1.208645      0.9074653      49.151138
# PathogenCLR     1.148     0.8713     1.080     1.220


lmechangelung <- data.frame(
  Variable = c("Shannon", "qPCRload", "Anaerobe_Abundance" ,"Pathogen_Abundance"),
  EffectSize = c(0.9720967, 0.9153946, 0.9194313, 1.243960),  
  LowerCI = c(9.443107e-01 ,8.156255e-01, 0.8672743, 1.1717519 ),  
  UpperCI = c( 0.9938192,  0.9757001, 0.9584863,  1.344739)  
)
lmechangelung_plot <- ggplot(lmechangelung, aes(x = Variable, y = EffectSize)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "", y = "", title = "Longitudinal changes") +
  theme_minimal() +
 theme(axis.text.y = element_text(size =12, face = "bold")) + coord_flip() # Rotate x-axis labels


CoxHRlung <- data.frame(
  Variable = c("Shannon", "qPCRload", "Anaerobe_Abundance" ,"Pathogen_Abundance"),
  EffectSize = c(0.6249, 0.9681, 0.8528, 1.148),  
  LowerCI = c(0.5074 , 0.9039, 0.7821, 1.080 ),  
  UpperCI = c( 0.7695,  1.037, 0.9299,  1.220)  
)
CoxHRlung_plot <- ggplot(CoxHRlung, aes(x = Variable, y = EffectSize)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "", y = "", title = "Hazard Ratios") +
  theme_minimal() +
  theme(axis.text.y = element_blank()) + coord_flip() # Rotate x-axis labels


JointModel_lung <- data.frame(
  Variable = c("Shannon", "qPCRload", "Anaerobe_Abundance" ,"Pathogen_Abundance"),
  EffectSize = c(0.5339783, 0.1693649, 0.8528, 1.208645),  
  LowerCI = c(0.4,  0.1 ,  0.5321127 , 0.9074653 ),  
  UpperCI = c( 1.0974617,  3.1, 0.99,  3)  
)
JointModel_lung_plot <- ggplot(JointModel_lung, aes(x = Variable, y = EffectSize)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "", y = "", title = "Joint Model Effect Sizes") +
  theme_minimal() + ylim(0.4, 3) +
  theme(axis.text.y = element_blank()) + coord_flip() # Rotate x-axis labels


#### create a single graphic with these parameters
lungEffectSizePlot <- ggarrange(lmechangelung_plot, CoxHRlung_plot, JointModel_lung_plot, ncol = 3)
ggsave("lungEffectSizePlot.jpeg", width = 12, height = 6)


################# effect sizes for oral



  
##### perform oral joint models ########################################
#
oralmbio <- oralmbio %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))
     oralmbio$survival_time60_fromICUAdmission <- oralmbio$survival_time90_fromICUAdmission 
      oralmbio$survival_time60_fromICUAdmission[ oralmbio$survival_time60_fromICUAdmission >60] <- 61       


oralmbio_joint <- filter(oralmbio, !duplicated(SubjectIDDay))

oralmbio_joint <- filter(oralmbio_joint, !is.na(survival_time60_fromICUAdmission))
oralmbio_joint$timesamplefromicuadmission[is.na(oralmbio_joint$timesamplefromicuadmission)] <- 5

oralmbio.surv <- UniqueVariables(oralmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

oralmbio.long <- oralmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"shannon")]

oralmbio.cov <- UniqueVariables(oralmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
oralmbio.jd <- jointdata(longitudinal = oralmbio.long, 
                            baseline = oralmbio.cov, 
                            survival = oralmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")

fit <- joint(data = oralmbio.jd, 
             long.formula = shannon ~  timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit)
 model.jointrandom.se_shannonoral <- jointSE(fit, n.boot = 100)
model.jointrandom.se_shannonoral # 	gamma_0	0.0900	0.9135	-0.5938	0.6615

relevant_rows <- model.jointrandom.se_shannonoral[c(2, 5), ]
# Calculate exponentiated estimates, 95% lower and upper intervals
exp_estimates <- exp(relevant_rows$Estimate)
lower_intervals <- exp(relevant_rows$"95%Lower")
upper_intervals <- exp(relevant_rows$"95%Upper")
# Combine the results into a dataframe
model.jointrandom.se_qpcrlung_results <- data.frame(
  Parameter = relevant_rows$Parameter,
  Exponentiated_Estimate = exp_estimates,
  Lower_Interval = lower_intervals,
  Upper_Interval = upper_intervals)



exp(model.jointrandom.se_shannonoral$Estimate)
# 7.1785686 0.9598291 0.9963068 1.0326208 1.0941743 1.2165269 1.0007002 1.3564893
 exp(model.jointrandom.se_shannonoral$`95%Lower`)
# 5.4091935 0.9417645 0.9914369 1.0194874 0.5522248 1.0000000 1.0000000 1.2642765
exp(model.jointrandom.se_shannonoral$`95%Upper`)
# 9.1449334 0.9809831 1.0001000 1.0442511 1.9376967 1.3115374 1.0034058 1.4405140
# Association                    gamma_0  
# sanity check with regular coxph model
survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ shannon + Age, data = oralmbio_joint, x = TRUE)
summary(survival_model)
# shannon    0.9641     1.0373    0.7771     1.196




#### repeat for qpcr
oralmbio_jointqpcr <- filter(oralmbio_joint, !is.na(Nof16scopies))

### do the same for the oralmbio_joint object. 
oralmbio.surv <- UniqueVariables(oralmbio_jointqpcr, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

oralmbio.long <- oralmbio_jointqpcr[, c("SubjectID", "timesamplefromicuadmission"  ,"Nof16scopies")]
oralmbio.long$Nof16scopies <- log10(oralmbio.long$Nof16scopies)
oralmbio.cov <- UniqueVariables(oralmbio_jointqpcr, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
oralmbio.jd <- jointdata(longitudinal = oralmbio.long, 
                            baseline = oralmbio.cov, 
                            survival = oralmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")
class(oralmbio.jd)
summary(oralmbio.jd)
plot(oralmbio.jd)

fit <- joint(data = oralmbio.jd, 
             long.formula = Nof16scopies ~  timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit) # non-significant for qpcr

# model.jointrandom.se_qpcroral <- jointSE(fit, n.boot = 100)
 model.jointrandom.se_qpcroral
 
 exp(model.jointrandom.se_qpcroral$Estimate)
# 745.6080053   0.8817912   1.0005001   1.0338607   1.1283992   1.3265744   1.0125785   2.1884642
 exp(model.jointrandom.se_qpcroral$`95%Lower`)
# 365.4757758   0.8344354   0.9915360   1.0203034   0.2609830   1.0000000   1.0000000   1.3423207
exp(model.jointrandom.se_qpcroral$`95%Upper`)
# 1197.1511840    0.9420471    1.0093434    1.0521124  168.9157382    2.7014806    1.0550625    3.5952014


 survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ log10(Nof16scopies) + Age, data = oralmbio_joint, x = TRUE)
summary(survival_model)
# log10(Nof16scopies)     0.9891     1.0110     0.841     1.163


##### oral anaerobes

oralmbio_joint <- filter(oralmbio_clr, !duplicated(SubjectIDDay))
oralmbio_joint <- filter(oralmbio_joint, !is.na(survival_time60_fromICUAdmission))
# define timesamplefromicuadmission
oralmbio_joint$SubjectIDDay[is.na(oralmbio_joint$timesamplefromicuadmission)]
oralmbio_joint$timesamplefromicuadmission[is.na(oralmbio_joint$timesamplefromicuadmission)] <- 5


oralmbio.surv <- UniqueVariables(oralmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

oralmbio.long <- oralmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"AnaerobeCLR")]

oralmbio.cov <- UniqueVariables(oralmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
oralmbio.jd <- jointdata(longitudinal = oralmbio.long, 
                            baseline = oralmbio.cov, 
                            survival = oralmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")
class(oralmbio.jd)
summary(oralmbio.jd)
 plot(oralmbio.jd)


fit <- joint(data = oralmbio.jd, 
             long.formula = AnaerobeCLR ~ 1+ timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit, variance = FALSE)

survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ AnaerobeCLR + Age, data = oralmbio_joint, x = TRUE)
summary(survival_model) # 0.9354     1.0691    0.8552     1.023


# model.jointrandom.se_oralanaerobe <- jointSE(fit, n.boot = 100)
model.jointrandom.se_oralanaerobe # 	gamma_0	

exp(model.jointrandom.se_oralanaerobe$Estimate) # 4.5285418 0.9141140 0.9893570 1.0326208 0.9078283 3.1641992 1.0005001 4.3859226
exp(model.jointrandom.se_oralanaerobe$`95%Lower`) # 2.05648867 0.87932563 0.98157191 1.02111993 0.06367903 1.00000000 1.00000000 2.89504310
exp(model.jointrandom.se_oralanaerobe$`95%Upper`) # 7.1427653  0.9581030  0.9989006  1.0463417  1.1049499  6.3047291  1.0097472 10.3563515


### oral samples pathogens: 

oralmbio_joint <- filter(oralmbio_clr_pathogens, !duplicated(SubjectIDDay))

oralmbio_joint <- oralmbio_joint %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))
     oralmbio_joint$survival_time60_fromICUAdmission <- oralmbio_joint$survival_time90_fromICUAdmission 
      oralmbio_joint$survival_time60_fromICUAdmission[ oralmbio_joint$survival_time60_fromICUAdmission >60] <- 61       

oralmbio_joint <- filter(oralmbio_joint, !is.na(survival_time60_fromICUAdmission))
# define timesamplefromicuadmission
oralmbio_joint$SubjectIDDay[is.na(oralmbio_joint$timesamplefromicuadmission)]
oralmbio_joint$timesamplefromicuadmission[is.na(oralmbio_joint$timesamplefromicuadmission)] <- 5

oralmbio.surv <- UniqueVariables(oralmbio_joint, 
                              var.col = c("survival_time60_fromICUAdmission", "survival_status60"),
                              id.col = "SubjectID")

oralmbio.long <- oralmbio_joint[, c("SubjectID", "timesamplefromicuadmission"  ,"PathogenCLR")]

oralmbio.cov <- UniqueVariables(oralmbio_joint, 
                             c("Age", "Gender"), 
                             id.col = "SubjectID")
oralmbio.jd <- jointdata(longitudinal = oralmbio.long, 
                            baseline = oralmbio.cov, 
                            survival = oralmbio.surv, 
                            id.col = "SubjectID", 
                            time.col = "timesamplefromicuadmission")

fit <- joint(data = oralmbio.jd, 
             long.formula = PathogenCLR ~ 1+ timesamplefromicuadmission + Age, 
             surv.formula = Surv(survival_time60_fromICUAdmission, survival_status60) ~ Age, 
             model = "intslope")

summary(fit, variance = FALSE)

survival_model <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60) ~ PathogenCLR + Age, data = oralmbio_joint, x = TRUE)
summary(survival_model) # PathogenCLR    1.078     0.9275     1.012     1.149


# model.jointrandom.pathogenoral <- jointSE(fit, n.boot = 100)

model.jointrandom.pathogenoral

exp(model.jointrandom.pathogenoral$Estimate) # 
exp(model.jointrandom.pathogenoral$`95%Lower`) # 
exp(model.jointrandom.pathogenoral$`95%Upper`) # 



 # PathogenCLR ~ 1+ timesamplefromicuadmission + Age

longitudinal_model_pathogenoral <- lme(PathogenCLR ~ timesamplefromicuadmission , random = ~ 1 | SubjectID, data = oralmbio_joint)
summary(longitudinal_model_pathogenoral)
# Extract the coefficient estimate, standard error, and degrees of freedom
estimate <- 0.1728128
std_error <- 0.02982946
df <- 196
# Calculate the critical value from the t-distribution
critical_value <- qt(0.975, df)
# Calculate the margin of error
margin_of_error <- critical_value * std_error
# Calculate the confidence interval
lower_ci <- estimate - margin_of_error
upper_ci <- estimate + margin_of_error
# Print the confidence interval
cat("95% CI for timesamplefromicuadmission:", lower_ci, "-", upper_ci, "\n")
exp(0.1728128) # 1.188644
exp (0.1139849) #  1.120735
exp(0.2316407) # 1.260667

#  -0.040544 -0.05828722 - -0.02280078

# Oral Shannon
# 0.9598291 - 0.9417645 - 0.9809831
# gamma 1.0941743 - 0.5522248 - 1.9376967
# shannon    0.9641        0.7771     1.196 # cox

# oral qpcr
# 0.8817912 -0.8344354 - 0.9420471
# gamma - 1.1283992 - 0.2609830 - 168.9157382
# log10(Nof16scopies)     0.9891          0.841     1.163

### oral anaerobes
# 0.9141140. -  0.87932563 - 0.9581030
# 0.9078283 - 0.06367903 - 1.1049499
# # 0.9354        0.8552     1.023

# pathogens;
# 1.188644  1.120735 - 1.260667
# gamma 1.176724 
# # PathogenCLR    1.078         1.012     1.149



lmechangeoral <- data.frame(
  Variable = c("Shannon", "qPCRload", "Anaerobe_Abundance" ,"Pathogen_Abundance"),
  EffectSize = c(0.9598291, 0.8817912, 0.9141140, 1.188644),  
  LowerCI = c(0.9417645 ,0.8344354, 0.87932563, 1.120735  ),  
  UpperCI = c( 0.9809831,  0.9420471, 0.9581030,  1.260667)  
)
lmechangeoral_plot <- ggplot(lmechangeoral, aes(x = Variable, y = EffectSize)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "", y = "", title = "Longitudinal changes") +
  theme_minimal() + 
  theme(axis.text.y = element_text(size =12, face = "bold"))+ coord_flip() # Rotate x-axis labels



CoxHRoral <- data.frame(
  Variable = c("Shannon", "qPCRload", "Anaerobe_Abundance" ,"Pathogen_Abundance"),
  EffectSize = c(0.9641, 0.9891, 0.9354, 1.078),  
  LowerCI = c(0.7771 , 0.841, 0.8552, 1.012 ),  
  UpperCI = c( 1.196,  1.163, 1.023,  1.149)  
)
CoxHRoral_plot <- ggplot(CoxHRoral, aes(x = Variable, y = EffectSize)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "", y = "", title = "Hazard Ratios") +
  theme_minimal() +  
  theme(axis.text.y = element_blank()) + coord_flip() # Rotate x-axis labels


JointModel_oral <- data.frame(
  Variable = c("Shannon", "qPCRload", "Anaerobe_Abundance" ,"Pathogen_Abundance"),
  EffectSize = c(1.0941743, 1.1283992, 0.9078283, 1.176724),  
  LowerCI = c(0.7771,  0.2609830 ,  0.4 , 1.01 ),  
  UpperCI = c( 1.9376967,  168.9157382, 3,  1.35)  
)
JointModel_oral_plot <- ggplot(JointModel_oral, aes(x = Variable, y = EffectSize)) +
  geom_point(size = 3) +  # Plot point estimates
  geom_errorbar(aes(ymin = LowerCI, ymax = UpperCI), width = 0.2) +  # Plot CI bars
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +  # Add line at HR = 1
  labs(x = "", y = "", title = "Joint Model Effect Sizes") +
  theme_minimal() + ylim(0.4, 3)  +
  theme(axis.text.y = element_blank()) + coord_flip() # Rotate x-axis labels


#### create a single graphic with these parameters
oralEffectSizePlot <- ggarrange(lmechangeoral_plot, CoxHRoral_plot, JointModel_oral_plot, ncol = 3)
ggsave("oralEffectSizePlot.jpeg", width = 12, height = 6)


```




```{r abundance by survival}

### shannon by survival in all 3 compartments.
color_scale <- c("Non-Survivors" = "black", "Survivors" = "green4")

alirmbio$survivors60day[alirmbio$Mortality60Day==0] <- "Survivors"
alirmbio$survivors60day[alirmbio$Mortality60Day==1] <- "Non-Survivors"

alirmbio$survivors60day <- ordered(alirmbio$survivors60day, levels = c("Non-Survivors", "Survivors"))

##### alirmbio_clr

###### take the oxygen variables and then the pathogenicity variables
alirmbio_oxygentaxa <- subset(alirmbio, select = c(AnaerobeSum, AerobeSum , FacultAnaerobeSum))
alirmbio_oxygentaxa$OtherSum <- 1 - (alirmbio_oxygentaxa$AnaerobeSum + alirmbio_oxygentaxa$AerobeSum + alirmbio_oxygentaxa$FacultAnaerobeSum)

alirmbio_oxygentaxa_clr <- as.data.frame(clr(alirmbio_oxygentaxa))
oldnames = c( "AnaerobeSum", "AerobeSum", "FacultAnaerobeSum","OtherSum" )
newnames = c("AnaerobeCLR", "AerobeCLR", "FacultAnaerobeCLR","OtherCLR" )
alirmbio_oxygentaxa_clr <- alirmbio_oxygentaxa_clr %>% rename_at(vars(oldnames), ~ newnames)

alirmbio_clr <- cbind(alirmbio, alirmbio_oxygentaxa_clr)


alirmbio_pathogens <- subset(alirmbio, select = c(OralSum, PathogenSum , OtherSum))
alirmbio_pathogens_clr <- as.data.frame(clr(alirmbio_pathogens))
oldnames = c( "OralSum", "PathogenSum", "OtherSum" )
newnames = c("OralCLR", "PathogenCLR", "OtherNonPathogenCLR" )
alirmbio_pathogens_clr <- alirmbio_pathogens_clr %>% rename_at(vars(oldnames), ~ newnames)

alirmbio_clr_pathogens <- cbind(alirmbio, alirmbio_pathogens_clr)




my_comparisons <- list(c("Non-Survivors", "Survivors"))
shannonbysurvivors_allcomp <-  alirmbio %>%  filter(!is.na(survivors60day))  %>% ggplot(aes(x=as.factor(survivors60day), y = shannon, color = as.factor(survivors60day)))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + facet_grid(SampleTypeFactor ~studydayfollowup)

qpcrbysurvivors_allcomp <-  alirmbio %>%  filter(!is.na(survivors60day))  %>% ggplot(aes(x=as.factor(survivors60day), y = Nof16scopies, color = as.factor(survivors60day)))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) +  scale_y_log10() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + facet_grid(SampleTypeFactor ~studydayfollowup)


### for anaerobes and pathogens obtain the CLR comparisons

anaerobesbysurvivors_allcomp <-  alirmbio_clr %>%  filter(!is.na(survivors60day))  %>% ggplot(aes(x=as.factor(survivors60day), y = AnaerobeCLR, color = as.factor(survivors60day)))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + facet_grid(SampleTypeFactor ~studydayfollowup) + ylab("Obligate Anaerobe CLR Abundance")


pathogensbysurvivors_allcomp <-  alirmbio_clr_pathogens %>%  filter(!is.na(survivors60day))  %>% ggplot(aes(x=as.factor(survivors60day), y = PathogenCLR, color = as.factor(survivors60day)))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + facet_grid(SampleTypeFactor ~studydayfollowup) + ylab("Pathogens CLR Abundance")


SurvivorsIndividualMetricsAllComp <- ggarrange(shannonbysurvivors_allcomp,qpcrbysurvivors_allcomp, anaerobesbysurvivors_allcomp, pathogensbysurvivors_allcomp, ncol = 2, nrow = 2 )
ggsave("SurvivorsIndividualMetricsAllComp.jpeg", width = 26, height = 30)


#### compare Candida.albicans between survivors and nonsurvivors. 
color_scale <- c("1" = "black", "0" = "green4")

my_comparisons <- list(c("0", "1"))
lungCalbmortality <- lungmbio %>%  filter(!is.na(Mortality60Day)) %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=as.factor(Mortality60Day), y = Candida.albicans, color = as.factor(Mortality60Day)))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("C.albicans Abundance") + ggtitle("Lung")

oralCalbmortality <- oralmbio %>%  filter(!is.na(Mortality60Day)) %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=as.factor(Mortality60Day), y = Candida.albicans, color = as.factor(Mortality60Day)))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("C.albicans Abundance")+ ggtitle("Oral")

gutCalbmortality <- gutmbio %>%  filter(!is.na(Mortality60Day)) %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=as.factor(Mortality60Day), y = Candida.albicans, color = as.factor(Mortality60Day)))  + geom_violin() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("C.albicans Abundance")+ ggtitle("Gut")

calbmortality <- ggarrange(oralCalbmortality, lungCalbmortality, gutCalbmortality, ncol = 3)
ggsave("calbmortality.jpeg", width = 12, height = 6)


```




```{r alir covid}

MetaData <- read_csv("data/MetaData_all.csv")
# covid IDs
covidsampleids <- read_xlsx("data/covidsampleids.xlsx")
covidmbio <- filter(MetaData, sample_id %in% covidsampleids$sample_id)


covidmbio <- filter(MultiCompMbio_all_clusters_allCohorts_2, !is.na(predicted_cluster_dysbiosis_ALIR_COVID))

covidmbio <- merge(covidmbio,MultiCompMbio_all_clusters_allCohorts_2, by = "sample_id", all.x =  TRUE)

covidmbio$SubjectID <-  as.numeric(covidmbio$SubjectID.y)
n_distinct(covidmbio$SubjectID) #
##### 

covidmbio$studydayfollowup <- ordered(covidmbio$studydayfollowup, levels = c("baseline", "middle", "late"))

#### look at shannon and abundance over time
my_comparisons <- list(c("baseline", "middle"), c("middle", "late"), c("baseline", "late"))

covidshannonfollowup <- covidmbio %>% filter(!is.na(studydayfollowup)) %>%  ggplot(aes(x=studydayfollowup, y = ShannonIndex))  + geom_boxplot() + geom_beeswarm(size = 0.3)   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Shannon Index")

covidpathogensfollowup <- covidmbio %>% filter(!is.na(studydayfollowup)) %>%   ggplot(aes(x=studydayfollowup, y = PathogenSum))  + geom_boxplot() + geom_beeswarm(size = 0.3)   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Pathogens Abundance")

covidanaerobesfollowup <- covidmbio %>% filter(!is.na(studydayfollowup)) %>%   ggplot(aes(x=studydayfollowup, y = AnaerobeSum))  + geom_boxplot() + geom_beeswarm(size = 0.3)   + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Anaerobes Abundance")

## export figure
covidfollowup <- ggarrange(covidshannonfollowup, covidanaerobesfollowup, covidpathogensfollowup, nrow = 3)
ggsave("covidfollowup.jpeg", width = 8, height = 12)

#### clusters for covid. 
covidmbio <- covidmbio %>% mutate(cluster = case_when(predicted_cluster_dysbiosis_ALIR_COVID=="LowDiversity" ~ "Low_Diversity", 
                                                         predicted_cluster_dysbiosis_ALIR_COVID=="Intermediate" ~ "Interm_Diversity",
                                                         predicted_cluster_dysbiosis_ALIR_COVID=="HighDiversity" ~ "High_Diversity"))


my_comparisons <- list(c("Low_Diversity", "High_Diversity"))
color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

 # no difference in pathogen or anaerobe abundance

 
covidshannoncluster <- covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%  ggplot(aes(x=cluster, y = ShannonIndex, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Shannon Index") + facet_wrap(~studydayfollowup)
  

covidpathogencluster <-  covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%   ggplot(aes(x=cluster, y = PathogenSum, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3) + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Pathogens Abundance") + facet_wrap(~studydayfollowup)
  
covidanaerobecluster <- covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%   ggplot(aes(x=cluster, y = AnaerobeSum, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Anaerobes Abundance") + facet_wrap(~studydayfollowup)
  
covidclustersabundance <- ggarrange(covidshannoncluster, covidanaerobecluster, covidpathogencluster, nrow = 3)
ggsave("covidclustersabundance.jpeg",  width = 16, height = 12)



covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%   ggplot(aes(x=cluster, y = Nof16scopies, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3) + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("N of 16s qPCR copies")  + scale_y_log10()
  
covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%  ggplot(aes(x=cluster, y = ShannonIndex, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Shannon Index")

### make this a main figure item. 
  covid16scopiesclusters_alldays <-  covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%   ggplot(aes(x=cluster, y = Nof16scopies, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3) + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("N of 16s qPCR copies")  + scale_y_log10() + ggtitle("Bacterial Load")

covidshannonclusters_alldays <-  covidmbio %>% filter(!is.na(cluster)) %>% filter(!is.na(studydayfollowup)) %>%  ggplot(aes(x=cluster, y = ShannonIndex, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("")  + ylab("Shannon Index") + ggtitle("Shannon Index")

covidclustersshannonqpcr_alldays <-ggarrange(covidshannonclusters_alldays,covid16scopiesclusters_alldays, nrow = 2 )
ggsave("covidclustersshannonqpcr_alldays.jpeg", width = 5, height = 10)

 ##### biomarkers
 
covidil6 <- covidmbio %>% filter(!is.na(cluster))    %>% ggplot(aes(x=cluster, y = il6, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma IL6") + ylab("") + theme(plot.title = element_text(size = 8)) + facet_wrap(~studydayfollowup)

covidil6_eta <- covidmbio %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = il6_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA IL6") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidtnfr1 <- covidmbio %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = tnfr1_final, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sTNFR1") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup) 
covidtnfr1_eta <- covidmbio %>% filter(!is.na(cluster))    %>% ggplot(aes(x=cluster, y = tnfr1_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sTNFR1") + ylab("") + theme(plot.title = element_text(size = 8)) + facet_wrap(~studydayfollowup)

covidrage <- covidmbio %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = rage, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sRAGE") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidrage_eta <- covidmbio %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = rage_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sRAGE") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidang2 <- covidmbio %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = ang2, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Ang-2") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidang2_eta <- covidmbio %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = ang2_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Ang-2") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidprocalcitonin <- covidmbio %>% filter(!is.na(cluster))  %>% ggplot(aes(x=cluster, y = procalcitonin, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Procalcitonin") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidprocalcitonin_eta <- covidmbio %>% filter(!is.na(cluster)) %>% ggplot(aes(x=cluster, y = procalcitonin_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Procalcitonin") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidpentraxin3 <- covidmbio %>% filter(!is.na(cluster))  %>% ggplot(aes(x=cluster, y = pentraxin3, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Pentraxin3") + ylab("") + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)

covidpentraxin3_eta <- covidmbio %>% filter(!is.na(cluster))  %>% ggplot(aes(x=cluster, y = pentraxin3_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Pentraxin3") + ylab("")  + theme(plot.title = element_text(size = 8))  + facet_wrap(~studydayfollowup)


covidbiomarkers_alldays <- ggarrange(covidil6_eta, covidtnfr1_eta, covidrage_eta, covidang2_eta, covidprocalcitonin_eta, covidpentraxin3_eta, covidil6, covidtnfr1, covidrage, covidang2, covidprocalcitonin, covidpentraxin3, ncol = 6, nrow = 2)
ggsave("covidbiomarkers_alldays.jpeg", width = 20, height = 8)


#### biomarkers baseline

covidmbio_basl <- filter(covidmbio, studydayfollowup=="baseline")

my_comparisons <- list(c("Low_Diversity", "High_Diversity"))
color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")

covidil6 <- covidmbio_basl %>% filter(!is.na(cluster))    %>% ggplot(aes(x=cluster, y = il6, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma IL6") + ylab("") + theme(plot.title = element_text(size = 8)) +theme(axis.text.x = element_text(size = 6))

covidil6_eta <- covidmbio_basl %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = il6_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA IL6") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidtnfr1 <- covidmbio_basl %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = tnfr1_final, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sTNFR1") + ylab("Plasma sTNFR1 (pg/ml)") + theme(plot.title = element_text(size = 12))    +theme(axis.text.x = element_text(size = 8))

covidtnfr1_eta <- covidmbio_basl %>% filter(!is.na(cluster))    %>% ggplot(aes(x=cluster, y = tnfr1_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sTNFR1") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidrage <- covidmbio_basl %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = rage, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma sRAGE") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidrage_eta <- covidmbio_basl %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = rage_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA sRAGE") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidang2 <- covidmbio_basl %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = ang2, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Ang-2") + ylab("Plasma Ang2 (pg/ml)") + theme(plot.title = element_text(size = 12))  +theme(axis.text.x = element_text(size = 8))

covidang2_eta <- covidmbio_basl %>% filter(!is.na(cluster))   %>% ggplot(aes(x=cluster, y = ang2_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Ang-2") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidprocalcitonin <- covidmbio_basl %>% filter(!is.na(cluster))  %>% ggplot(aes(x=cluster, y = procalcitonin, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Procalcitonin") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidprocalcitonin_eta <- covidmbio_basl %>% filter(!is.na(cluster)) %>% ggplot(aes(x=cluster, y = procalcitonin_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Procalcitonin") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidpentraxin3 <- covidmbio_basl %>% filter(!is.na(cluster))  %>% ggplot(aes(x=cluster, y = pentraxin3, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale)+ stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Plasma Pentraxin3") + ylab("") + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))

covidpentraxin3_eta <- covidmbio_basl %>% filter(!is.na(cluster))  %>% ggplot(aes(x=cluster, y = pentraxin3_eta_pradj, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("ETA Pentraxin3") + ylab("")  + theme(plot.title = element_text(size = 8))  +theme(axis.text.x = element_text(size = 6))


covidbiomarkers_basl <- ggarrange(covidil6_eta, covidtnfr1_eta, covidrage_eta, covidang2_eta, covidprocalcitonin_eta, covidpentraxin3_eta, covidil6, covidtnfr1, covidrage, covidang2, covidprocalcitonin, covidpentraxin3, ncol = 6, nrow = 2)
ggsave("covidbiomarkers_basl.jpeg", width = 20, height = 8)

covidtnfr1Ang2 <- ggarrange(covidtnfr1_eta, covidang2_eta, covidtnfr1, covidang2, nrow = 2, ncol = 2)
ggsave("covidtnfr1Ang2.jpeg", width = 7, height = 10)


######## phenotypes

covidshannonbyphenos <- covidmbio %>% filter(!is.na(phenotype)) %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=phenotype, y = ShannonIndex, color = phenotype))  + geom_boxplot() + geom_jitter(size = 0.3)  + scale_color_jama() + stat_compare_means() + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")

covidanaerobesbyphenos <- covidmbio %>% filter(!is.na(phenotype)) %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=phenotype, y = AnaerobeSum, color = phenotype))  + geom_boxplot() + geom_jitter(size = 0.3)  + scale_color_jama() + stat_compare_means() + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")

covidpathogensbyphenos <- covidmbio %>% filter(!is.na(phenotype))  %>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=phenotype, y = PathogenSum, color = phenotype))  + geom_boxplot() + geom_jitter(size = 0.3)  + scale_color_jama() + stat_compare_means() + scale_y_log10() +  theme_pubr() + theme(legend.position = "none") +xlab("")

covidbyphenosabundance <- ggarrange(covidshannonbyphenos, covidanaerobesbyphenos,covidpathogensbyphenos, nrow = 3 )
ggsave("covidbyphenosabundance.jpeg", width = 6, height =10)

  
 ##### survival
 #### create publishable figure
covidmbio_basl <- covidmbio_basl %>% mutate(survival_status60 = case_when(Mortality60Day==1 ~ 2, 
                                                                                  Mortality60Day==0 ~ 1))

     covidmbio_basl$survival_time60_fromICUAdmission <- covidmbio_basl$survival_time90_fromICUAdmission 
      covidmbio_basl$survival_time60_fromICUAdmission[ covidmbio_basl$survival_time60_fromICUAdmission >60] <- 61       
      

      

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~cluster , data=covidmbio_basl)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") 
 

#### create publishable figure. 

surv_lungclustersCOVID <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE,  legend.labs=c("High_Diversity",  "Interm_Diversity", "Low_Diversity"), palette = c("#4DBBD5FF", "#F39B7FFF", "#7E6148FF"), risk.table = FALSE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("surv_lungclustersCOVID.jpeg", width = 580, height = 418)
print(surv_lungclustersCOVID, newpage = FALSE)
dev.off()

# cox by lung
covidmbio_basl <- covidmbio_basl %>% mutate(lungLowDiv = case_when(cluster=="Low_Diversity" ~ "yes",
                                                                     cluster=="Interm_Diversity" ~ "no"))

survival_coxph_bygroup60 <- coxph(Surv(survival_time60_fromICUAdmission, survival_status60)~lungLowDiv, data=covidmbio_basl)
summary(survival_coxph_bygroup60) # numerically worse but statistically not significant. 


####### liberation

covidmbio_basl$timetoliberation[covidmbio_basl$timetoliberation>60] <- 61

liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~cluster, data=covidmbio_basl)
ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Kaplan-Meier Curve within 60 days") 

liberation_covidclusters <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "", legend.labs=c("High_Diversity",  "Interm_Diversity", "Low_Diversity"), palette = c("#4DBBD5FF", "#F39B7FFF", "#7E6148FF"), risk.table = FALSE, legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("liberation_covidclusters.jpeg", width = 580, height = 418)
print(liberation_covidclusters, newpage = FALSE)
dev.off()



covidmbio%>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=cluster, y = il6, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jco() + stat_compare_means()  +  theme_pubr() + theme(legend.position = "none") +xlab("")

covidmbio%>% filter(studydayfollowup=="baseline") %>% ggplot(aes(x=cluster, y = Nof16scopies, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jco() + stat_compare_means(comparisons = my_comparisons)  +  theme_pubr() + theme(legend.position = "none") +xlab("") + scale_y_log10()

# covidmbio %>% ggscatter(x = "Nof16scopies", y = "rage_eta_pradj", add = "reg.line", conf.int = FALSE,  size = 2, shape = 1,  cor.coef = TRUE, add.params= list(linetype="solid"), legend = "none")  + scale_x_log10() + scale_y_log10() 3### significant

cat_table_covidclusters<- print(CreateCatTable(vars = vars_categorical, strata = "cluster" , data=covidmbio_basl, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_alir_covidclusters <- print(CreateContTable(vars = vars_continuous, strata = "cluster" , data=covidmbio_basl), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_covidclusters, "cat_table_covidclusters.csv")
write.csv(con_table_alir_covidclusters, "con_table_alir_covidclusters.csv")


subset_data <- subset(covidmbio_basl, select = c(SubjectID, cluster, phenotype))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
category_colors <- c("blue", "yellow", "orange" , "red" , "purple", "black")  #CC0033",  "#FFCCCC",  "#0000FF", "#0099CC"") # Adjust the colors as desired
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
grid.col = c(Low_Diversity = "#7E6148FF", Interm_Diversity = "#F39B7FFF", High_Diversity = "#4DBBD5FF", 
    hyperinf = "red", hypoinfl = "blue")
chordDiagram(category_matrix, col = category_colors, grid.col = grid.col)

fisher.test(category_matrix) # 0.234



covidmbio%>%   ggplot(aes(x=as.factor(COVID_clusters), y = Nof16scopies, color = as.factor(COVID_clusters)))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jco() + stat_compare_means()  +  theme_pubr() + theme(legend.position = "none") +xlab("") + scale_y_log10()

covidmbio%>%   ggplot(aes(x=as.factor(Mortality60Day), y = Nof16scopies, color = as.factor(Mortality60Day)))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jco() + stat_compare_means()  +  theme_pubr() + theme(legend.position = "none") +xlab("") + scale_y_log10()


##### create a summary figure for the covid clusters for pub purposes to summarize findings. 

covidshannoncluster <- covidmbio %>% filter(!is.na(cluster)) %>% ggplot(aes(x=cluster, y = ShannonIndex, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jama() + stat_compare_means(comparisons = my_comparisons)  +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Shannon Index") + ylab("") + theme(plot.title = element_text(size = 8))

covidbacterialoadcluster <-  covidmbio %>% filter(!is.na(cluster)) %>% ggplot(aes(x=cluster, y = Nof16scopies, color = cluster))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_jama() + stat_compare_means(comparisons = my_comparisons)  +  theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("N of 16S copies") + ylab("") + theme(plot.title = element_text(size = 8)) + scale_y_log10()

covidSummaryClusters <- ggarrange(covidshannoncluster, covidbacterialoadcluster,covidrage_eta,  covidprocalcitonin_eta, covidpentraxin3_eta, ncol = 5)
ggsave("covidSummaryClusters.jpeg", width = 12, height = 6)


```



```{r MGH covid analysis }

mghstool <- read_csv("mghstool_8.24.23.csv")
mghsputum <- read_csv("mghsputum_8.24.23.csv")

## fix the names to create a single file by appending instead of merging. 
names(mghstool)
names(mghsputum)
names(mghstool)[3] <- "time_sample"
names(mghstool)[4] <- "sampleid_sample"
names(mghstool)[5] <- "date_collect_sample"
names(mghstool)[6] <- "viral_load_sample"

names(mghsputum)[3] <- "time_sample"
names(mghsputum)[4] <- "sampleid_sample"
names(mghsputum)[5] <- "date_collect_sample"
names(mghsputum)[6] <- "viral_load_sample"

mghsubjectid <-rbind(mghstool, mghsputum)
mghsubjectid <- filter(mghsubjectid, !duplicated(subjid)) # 97 subjects

# write this file
write_csv(mghsubjectid, "mghsubjectid_8.24.23.csv")

#### obtain descriptive data for this cohort. 

names(mghsubjectid)

vars_continuous <- c("age","bmi_admit", "WBC",  "los_hospital", "los_icu", "admit_sofa_score", "admit_saps_score", "charlson", "steroid_los", "antibact_los")

vars_categorical <- c( "male", "race",  "covid_severity", "tx_remdesivir", "tx_steroids", "tx_anti_il6", "pmh_immunosuppression", "pmh_malig" , "pmh_pulmonary" , "pmh_cardiac", "pmh_dm", "tx_oxygen_any", "tx_oxygen_mechvent", "dead_hosp90", "dead", "Mortality60Day", "steroid_flag", "antibact_flag") 

############## 

cat_table_mghcovid<- print(CreateCatTable(vars = vars_categorical, data=mghsubjectid,  includeNA = FALSE))
con_table_mghcovid <- print(CreateContTable(vars = vars_continuous, data=mghsubjectid), nonnormal = vars_continuous, digits  = 1)

write.csv(cat_table_mghcovid, "cat_table_mghcovid.csv")
write.csv(con_table_mghcovid, "con_table_mghcovid.csv")


##### incporate cluster assignments and taxa abundance

MultiCompMbio_all_clusters_allCohorts_2$IDT <- MultiCompMbio_all_clusters_allCohorts_2$sample_id
mghsputum <- merge(mghsputum, MultiCompMbio_all_clusters_allCohorts_2, by = "IDT", all.x = TRUE)
table(mghsputum$predicted_cluster_dysbiosis)
mghsputum$ShannonIndex
n_distinct(mghsputum$IDT)
oddsratio.fisher(mghsputum$predicted_cluster_dysbiosis, mghsputum$covid_severity) #### highly significant odds ratio for severity. 

### rename the clusters for easier visualization
color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
# fix the clusters name
mghsputum <- mghsputum %>% mutate(BaslClusters = case_when(predicted_cluster_dysbiosis=="LowDiversity" ~ "Low_Diversity", 
                                                         predicted_cluster_dysbiosis=="Intermediate" ~ "Interm_Diversity",
                                                         predicted_cluster_dysbiosis=="HighDiversity" ~ "High_Diversity"))


my_comparisons <- list( c("Low_Diversity", "High_Diversity"))
##### look at differences in baseline shannon abundance by clusters

mghsputumshannoncluster  <-  mghsputum  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = ShannonIndex, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Shannon index")

mghsputumanaerobescluster <-   mghsputum  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = AnaerobeSum, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe Abundance")

mghsputumpathogenscluster <-   mghsputum  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = PathogenSum, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen Abundance")
  
mghlungclusterabundance <- ggarrange(mghsputumshannoncluster, mghsputumanaerobescluster, mghsputumpathogenscluster, ncol = 3)
ggsave("mghlungclusterabundance.jpeg", width = 10, height = 6)

mghlungclusterabundance_two <- ggarrange(mghsputumshannoncluster, mghsputumanaerobescluster,  nrow = 2)
ggsave("mghlungclusterabundance_two.jpeg", width = 4.5, height = 10)

mghsputum$viral_load_BL
### no differences in viral load. 
mghsputum  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = viral_load_sample, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  

### create chord diagram for severity

library(circlize)

subset_data <- subset(mghsputum, select = c(subjid, BaslClusters, covid_severity))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
category_colors <- c("blue", "yellow", "orange" , "red" , "purple", "black")  #CC0033",  "#FFCCCC",  "#0000FF", "#0099CC"") # Adjust the colors as desired
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
grid.col = c(Low_Diversity = "#7E6148FF", Interm_Diversity = "#F39B7FFF", High_Diversity = "#4DBBD5FF", 
    Severe = "red", Moderate = "blue")
chordDiagram(category_matrix, col = category_colors, grid.col = grid.col)

#### look at mortality
oddsratio.fisher(mghsputum$BaslClusters, mghsputum$covid_severity) 
#                   odds ratio with 95% C.I.
# Predictor          estimate     lower    upper
#   High_Diversity   1.000000        NA       NA
#   Interm_Diversity  3.62376 0.3525873 189.2528
 # Low_Diversity    18.08116 1.9226860 922.5741



oddsratio.fisher(mghsputum$BaslClusters, mghsputum$Mortality60Day) #### few data points. 

survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~ BaslClusters, data=mghsputum)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days") # 



##### stool clusters mgh analyses
##### incporate cluster assignments and taxa abundance


mghstool <- merge(mghstool, MultiCompMbio_all_clusters_allCohorts_2, by = "IDT", all.x = TRUE)
table(mghstool$predicted_cluster_dysbiosis)
mghstool$ShannonIndex
oddsratio.fisher(mghstool$predicted_cluster_dysbiosis, mghstool$covid_severity) #### non-significant


mghstool <- mghstool %>% mutate(BaslClusters = case_when(predicted_cluster_dysbiosis=="LowDiversity" ~ "Low_Diversity", 
                                                         predicted_cluster_dysbiosis=="Intermediate" ~ "Interm_Diversity",
                                                         predicted_cluster_dysbiosis=="HighDiversity" ~ "High_Diversity"))


my_comparisons <- list( c("Low_Diversity", "High_Diversity"))

##### look at differences in baseline shannon abundance by clusters
mghstoolshannoncluster  <-  mghstool  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = ShannonIndex, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut") + theme(axis.text.x = element_text(size = 8))  + ylab("Shannon index")

mghstoolanaerobescluster <-   mghstool  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = AnaerobeSum, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3) + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe Abundance")

mghstoolpathogenscluster <-   mghstool  %>% filter(!is.na(BaslClusters))   %>% ggplot(aes(x=BaslClusters, y = PathogenSum, color = BaslClusters))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_manual(values = color_scale) + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Gut") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen Abundance")
  
mghstoolclusterabundance <- ggarrange(mghstoolshannoncluster, mghstoolanaerobescluster, mghstoolpathogenscluster, ncol = 3)
ggsave("mghstoolclusterabundance.jpeg", width = 10, height = 6)

mghstoolclusterabundance_two <- ggarrange(mghstoolshannoncluster, mghstoolanaerobescluster, nrow = 2)
ggsave("mghstoolclusterabundance_two.jpeg", width = 4.5, height = 10)


### create chord diagram for severity

library(circlize)

subset_data <- subset(mghstool, select = c(subjid, BaslClusters, covid_severity))
subset_data <- na.omit(subset_data)
# Create a matrix with the counts of subjects in each category combination
category_matrix <- table(subset_data[, -1])
chordDiagram(category_matrix)
category_colors <- c("blue",   "red" , "purple", "black")  #CC0033",  "#FFCCCC",  "#0000FF", "#0099CC"") # Adjust the colors as desired
# color_scale <- c("Low_Diversity" = "#7E6148FF", "Interm_Diversity" = "#F39B7FFF", "High_Diversity" = "#4DBBD5FF")
grid.col = c(Low_Diversity = "#7E6148FF", High_Diversity = "#4DBBD5FF",     Severe = "red", Moderate = "blue")
chordDiagram(category_matrix, col = category_colors, grid.col = grid.col)

oddsratio.fisher(mghstool$BaslClusters, mghstool$covid_severity) #### 
#            odds ratio with 95% C.I.
# Predictor          estimate      lower      upper
# Predictor        estimate    lower    upper
#   High_Diversity 1.000000       NA       NA
#   Low_Diversity  4.081573 1.560332 11.20835
#### look at mortality

oddsratio.fisher(mghstool$BaslClusters, mghstool$Mortality60Day) #### few data points. 
### no difference
survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~ BaslClusters, data=mghstool)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "orange"), title = "Kaplan-Meier Curve within 60 days") # 



#### differences by severity
mghsputum$covid_severity

my_comparisons <- list(c("Moderate", "Severe"))

mghstoolshannoncovid_severity  <-  mghstool  %>% filter(!is.na(covid_severity))   %>% ggplot(aes(x=covid_severity, y = ShannonIndex, color = covid_severity))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Stool") + theme(axis.text.x = element_text(size = 8))  + ylab("Shannon index")

mghstoolanaerobescovid_severity <-   mghstool  %>% filter(!is.na(covid_severity))   %>% ggplot(aes(x=covid_severity, y = AnaerobeSum, color = covid_severity))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Stool") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe Abundance")

mghstoolpathogenscovid_severity <-   mghstool  %>% filter(!is.na(covid_severity))   %>% ggplot(aes(x=covid_severity, y = PathogenSum, color = covid_severity))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Stool") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen Abundance")
  
mghstoolcovid_severityabundance <- ggarrange(mghstoolshannoncovid_severity, mghstoolanaerobescovid_severity, mghstoolpathogenscovid_severity, ncol = 3)
ggsave("mghstoolcovid_severityabundance.jpeg", width = 10, height = 6)

mghsputumshannoncovid_severity   <-  mghsputum  %>% filter(!is.na(covid_severity))   %>% ggplot(aes(x=covid_severity, y = ShannonIndex, color = covid_severity))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Shannon index")

mghsputumanaerobescovid_severity  <-   mghsputum  %>% filter(!is.na(covid_severity))   %>% ggplot(aes(x=covid_severity, y = AnaerobeSum, color = covid_severity))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Anaerobe Abundance")

mghsputumpathogenscovid_severity  <-   mghsputum  %>% filter(!is.na(covid_severity))   %>% ggplot(aes(x=covid_severity, y = PathogenSum, color = covid_severity))  + geom_boxplot() + geom_beeswarm(size = 0.3)  + scale_color_lancet() + stat_compare_means(comparisons = my_comparisons) + theme_pubr() + theme(legend.position = "none") +xlab("") + ggtitle("Lung") + theme(axis.text.x = element_text(size = 8))  + ylab("Pathogen Abundance")
  
mghlungcovid_severityabundance <- ggarrange(mghsputumshannoncovid_severity, mghsputumanaerobescovid_severity, mghsputumpathogenscovid_severity, ncol = 3)
ggsave("mghlungcovid_severityabundance.jpeg", width = 10, height = 6)


```

